// @webiny/app-headless-cms/admin/plugins/fieldRenderers/boolean/booleanSwitch.js
import _taggedTemplateLiteral from "@babel/runtime/helpers/taggedTemplateLiteral";

function _templateObject2() {
  var data = _taggedTemplateLiteral(["Renders a simple switch button."]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteral(["Boolean Input"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

import React from "react";
import { I18NValue } from "@webiny/app-i18n/components";
import { Switch } from "@webiny/ui/Switch";
import get from "lodash/get";
import { i18n } from "@webiny/app/i18n";
var t = i18n.ns("app-headless-cms/admin/fields/boolean");
var plugin = {
  type: "cms-editor-field-renderer",
  name: "cms-editor-field-renderer-boolean",
  renderer: {
    rendererName: "boolean-input",
    name: t(_templateObject()),
    description: t(_templateObject2()),
    canUse: function canUse(_ref) {
      var field = _ref.field;
      return field.type === "boolean" && !field.multipleValues && !get(field, "predefinedValues.enabled");
    },
    render: function render(_ref2) {
      var field = _ref2.field,
          getBind = _ref2.getBind;
      var Bind = getBind();
      return /*#__PURE__*/React.createElement(Bind, null, function (bindProps) {
        return /*#__PURE__*/React.createElement(Switch, Object.assign({}, bindProps, {
          label: I18NValue({
            value: field.label
          }),
          description: I18NValue({
            value: field.helpText
          })
        }));
      });
    }
  }
};
export default plugin;
