// @webiny/app-headless-cms/admin/plugins/menus/ContentModelMenuItems.js
import _taggedTemplateLiteral from "@babel/runtime/helpers/taggedTemplateLiteral";

function _templateObject() {
  var data = _taggedTemplateLiteral(["Nothing to show."]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

import React from "react";
import { i18n } from "@webiny/app/i18n";
import { LIST_MENU_CONTENT_GROUPS_MODELS } from "./../../viewsGraphql";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import get from "lodash.get";
var t = i18n.ns("app-headless-cms/admin/menus");
import { useQuery } from "@webiny/app-headless-cms/admin/hooks";

var ContentModelMenuItems = function ContentModelMenuItems(_ref) {
  var Section = _ref.Section,
      Item = _ref.Item;
  var response = useQuery(LIST_MENU_CONTENT_GROUPS_MODELS);

  var _ref2 = get(response, "data.listContentModelGroups") || {},
      data = _ref2.data;

  if (!data) {
    return null;
  }

  return data.map(function (contentModelGroup) {
    return /*#__PURE__*/React.createElement(Section, {
      key: contentModelGroup.id,
      name: "cms-content-models-".concat(contentModelGroup.id),
      label: contentModelGroup.name,
      icon: /*#__PURE__*/React.createElement(FontAwesomeIcon, {
        style: {
          color: "var(--mdc-theme-text-secondary-on-background)"
        },
        icon: contentModelGroup.icon.split("/")
      })
    }, contentModelGroup.contentModels.length === 0 && /*#__PURE__*/React.createElement(Item, {
      style: {
        opacity: 0.4
      },
      key: "empty-item",
      label: t(_templateObject())
    }), contentModelGroup.contentModels.map(function (contentModel) {
      return /*#__PURE__*/React.createElement(Item, {
        key: contentModel.id,
        label: contentModel.name,
        path: "/cms/content-models/manage/".concat(contentModel.modelId)
      });
    }));
  });
};

export default ContentModelMenuItems;
