// @webiny/app-page-builder/editor/plugins/elements/social/twitter/index.js
import React from "react";
import styled from "@emotion/styled";
import { Tab } from "@webiny/ui/Tabs";
import { Input } from "@webiny/ui/Input";
import { Grid, Cell } from "@webiny/ui/Grid";
import { ReactComponent as SocialIcon } from "@svgr/webpack!./../../../elementGroups/social/round-people-24px.svg";
import placeholder from "./placeholder.jpg";
import { createEmbedPlugin, createEmbedSettingsPlugin } from "./../../utils/oembed/createEmbedPlugin";
import { validation } from "@webiny/validation";
import { ReactComponent as LogoIcon } from "@svgr/webpack!./twitter-brands.svg";
var PreviewBox = /*#__PURE__*/styled("div", {
  target: "e1v0vpu10",
  label: "PreviewBox"
})({
  textAlign: "center",
  height: 50,
  svg: {
    height: 50,
    width: 50
  }
});
export default (function () {
  return [createEmbedPlugin({
    type: "twitter",
    toolbar: {
      title: "Tweet",
      group: "pb-editor-element-group-social",
      preview: function preview() {
        return /*#__PURE__*/React.createElement(PreviewBox, null, /*#__PURE__*/React.createElement(LogoIcon, null));
      }
    },
    oembed: {
      global: "twttr",
      sdk: "https://platform.twitter.com/widgets.js",
      init: function init(_ref) {
        var node = _ref.node;
        window.twttr.widgets.load(node);
      }
    },
    renderElementPreview: function renderElementPreview(_ref2) {
      var width = _ref2.width,
          height = _ref2.height;
      return /*#__PURE__*/React.createElement("img", {
        style: {
          width: width,
          height: height
        },
        src: placeholder,
        alt: "Tweet"
      });
    }
  }), createEmbedSettingsPlugin({
    type: "twitter",
    render: function render(_ref3) {
      var Bind = _ref3.Bind;
      return /*#__PURE__*/React.createElement(Tab, {
        icon: /*#__PURE__*/React.createElement(SocialIcon, null),
        label: "Twitter"
      }, /*#__PURE__*/React.createElement(Grid, null, /*#__PURE__*/React.createElement(Cell, {
        span: 12
      }, /*#__PURE__*/React.createElement(Bind, {
        name: "source.url",
        validators: validation.create("required,url")
      }, /*#__PURE__*/React.createElement(Input, {
        label: "Tweet URL",
        description: "Enter a Tweet URL"
      })))));
    }
  })];
});
