// @webiny/app-template-admin/apolloClient/NetworkError.js
import _taggedTemplateLiteral from "@babel/runtime/helpers/taggedTemplateLiteral";

function _templateObject() {
  var data = _taggedTemplateLiteral(["\n    {\n        networkError\n    }\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

import React from "react";
import { useQuery } from "react-apollo";
import gql from "graphql-tag";
import { Alert } from "@webiny/ui/Alert";
import { Elevation } from "@webiny/ui/Elevation";
import { css } from "emotion";
import styled from "@emotion/styled";
var GET_ERROR = gql(_templateObject());
var errorStyle = /*#__PURE__*/css({
  margin: 50,
  padding: "15px 25px 5px 25px",
  boxSizing: "border-box"
}, "label:errorStyle;");
var Code = /*#__PURE__*/styled("span", {
  target: "ezc5eti0",
  label: "Code"
})({
  padding: "2px 5px",
  margin: "0 2px",
  borderRadius: "5px",
  color: "var(--webiny-theme-color-on-primary)",
  backgroundColor: "var(--webiny-theme-color-primary)"
});
var Paragraph = /*#__PURE__*/styled("span", {
  target: "ezc5eti1",
  label: "Paragraph"
})({
  lineHeight: 1.8
});

var NetworkError = function NetworkError(_ref) {
  var children = _ref.children;

  var _useQuery = useQuery(GET_ERROR, {
    fetchPolicy: "cache-only"
  }),
      data = _useQuery.data;

  if (!process.env.REACT_APP_GRAPHQL_API_URL && process.env.NODE_ENV === "production") {
    return /*#__PURE__*/React.createElement(Elevation, {
      className: errorStyle,
      z: 2
    }, /*#__PURE__*/React.createElement(Alert, {
      type: "danger",
      title: "GraphQL API is unavailable"
    }, /*#__PURE__*/React.createElement(Paragraph, null, "Looks like you've deployed your apps before deploying an API for them.", /*#__PURE__*/React.createElement("br", null), "Make sure you deploy a matching API environment before deploying your apps by running", /*#__PURE__*/React.createElement(Code, null, "yarn webiny deploy api --env=your-env"), ". This is necessary for your apps to be built correctly.", /*#__PURE__*/React.createElement("br", null), "Once your API is deployed, deploy your apps by running", /*#__PURE__*/React.createElement(Code, null, "yarn webiny deploy apps --env=your-env"), "."), /*#__PURE__*/React.createElement("br", null), /*#__PURE__*/React.createElement(Paragraph, null, "For step by step instructions on deploying your apps, visit our", " ", /*#__PURE__*/React.createElement("a", {
      target: "_blank",
      rel: "noreferrer",
      href: "https://docs.webiny.com/docs/get-started/going-live"
    }, "Going Live guide"), ".")));
  }

  if (!process.env.REACT_APP_GRAPHQL_API_URL && process.env.NODE_ENV === "development") {
    return /*#__PURE__*/React.createElement(Elevation, {
      className: errorStyle,
      z: 2
    }, /*#__PURE__*/React.createElement(Alert, {
      type: "danger",
      title: "GraphQL API is unavailable"
    }, /*#__PURE__*/React.createElement(Paragraph, null, "Looks like you've started your app before deploying an API for it.", /*#__PURE__*/React.createElement("br", null), "Make sure you deploy a ", /*#__PURE__*/React.createElement(Code, null, "local"), " environment of your API by running", /*#__PURE__*/React.createElement(Code, null, "yarn webiny deploy api --env=local"), ". This is necessary to run local React apps.", /*#__PURE__*/React.createElement("br", null), "Once your API is deployed, restart this app via your terminal by running", " ", /*#__PURE__*/React.createElement(Code, null, "yarn start"), " again."), /*#__PURE__*/React.createElement("br", null), /*#__PURE__*/React.createElement(Paragraph, null, "For step by step instructions on getting started with Webiny, visit our", " ", /*#__PURE__*/React.createElement("a", {
      target: "_blank",
      rel: "noreferrer",
      href: "https://docs.webiny.com/docs/get-started/quick-start"
    }, "Quick Start guide"), ".")));
  }

  return data && data.networkError ? /*#__PURE__*/React.createElement(Elevation, {
    className: errorStyle,
    z: 2
  }, /*#__PURE__*/React.createElement(Alert, {
    type: "danger",
    title: "GraphQL API is unavailable"
  }, /*#__PURE__*/React.createElement(Paragraph, null, "Your API seems to be unavailable!", /*#__PURE__*/React.createElement("br", null), "Make sure your API is working by visiting", " ", /*#__PURE__*/React.createElement("a", {
    href: process.env.REACT_APP_GRAPHQL_API_URL,
    target: "_blank",
    rel: "noreferrer"
  }, process.env.REACT_APP_GRAPHQL_API_URL), ".", /*#__PURE__*/React.createElement("br", null), "If you've recently deployed your API, please allow ", /*#__PURE__*/React.createElement(Code, null, "~10 minutes"), " ", "for your API URL to become available."))) : children;
};

export { NetworkError, GET_ERROR };
