// @webiny/app-page-builder/editor/plugins/elements/code/codesandbox/index.js
import React from "react";
import styled from "@emotion/styled";
import { Tab } from "@webiny/ui/Tabs";
import { Input } from "@webiny/ui/Input";
import { Grid, Cell } from "@webiny/ui/Grid";
import { ReactComponent as CodeIcon } from "@svgr/webpack!./../../../elementGroups/code/code.svg";
import { createEmbedPlugin, createEmbedSettingsPlugin } from "./../../utils/oembed/createEmbedPlugin";
import { ReactComponent as LogoIcon } from "@svgr/webpack!./codesandbox-logo.svg";
import { validation } from "@webiny/validation";
var PreviewBox = /*#__PURE__*/styled("div", {
  target: "e18465i10",
  label: "PreviewBox"
})({
  textAlign: "center",
  height: 50,
  svg: {
    height: 50,
    width: 50
  }
});
export default (function () {
  return [createEmbedPlugin({
    type: "codesandbox",
    toolbar: {
      title: "CodeSandbox",
      group: "pb-editor-element-group-code",
      preview: function preview() {
        return /*#__PURE__*/React.createElement(PreviewBox, null, /*#__PURE__*/React.createElement(LogoIcon, null));
      }
    },
    oembed: {
      onData: function onData(data) {
        data.html = data.html.replace(/1000px/g, "100%").replace(/1000/g, "100%");
        return data;
      }
    }
  }), createEmbedSettingsPlugin({
    type: "codesandbox",
    render: function render(_ref) {
      var Bind = _ref.Bind;
      return /*#__PURE__*/React.createElement(Tab, {
        icon: /*#__PURE__*/React.createElement(CodeIcon, null),
        label: "CodeSandbox"
      }, /*#__PURE__*/React.createElement(Grid, null, /*#__PURE__*/React.createElement(Cell, {
        span: 12
      }, /*#__PURE__*/React.createElement(Bind, {
        name: "source.url",
        validators: validation.create("required,url")
      }, /*#__PURE__*/React.createElement(Input, {
        label: "CodeSandbox URL",
        description: "Enter a CodeSandbox URL"
      })))));
    }
  })];
});
