// @webiny/app-mailchimp/admin/index.js
import _taggedTemplateLiteral from "@babel/runtime/helpers/taggedTemplateLiteral";
import _toConsumableArray from "@babel/runtime/helpers/toConsumableArray";

function _templateObject() {
  var data = _taggedTemplateLiteral(["Mailchimp"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

import React from "react";
import styled from "@emotion/styled";
import { Tab } from "@webiny/ui/Tabs";
import { Route } from "@webiny/react-router";
import Helmet from "react-helmet";
import { AdminLayout } from "@webiny/app-admin/components/AdminLayout";
import MailchimpSettings from "./components/MailchimpSettings";
import MailchimpElementAdvancedSettings from "./components/MailchimpElementAdvancedSettings";
import MailchimpElement from "./components/MailchimpElement";
import { SecureRoute } from "@webiny/app-security/components";
import { ReactComponent as MailchimpLogo } from "@svgr/webpack!./mailchimp-logo.svg";
import render from "./../render";
import { i18n } from "@webiny/app/i18n";
var t = i18n.ns("app-mailchimp/admin");
var PreviewBox = /*#__PURE__*/styled("div", {
  target: "ednexo0",
  label: "PreviewBox"
})({
  textAlign: "center",
  height: 80,
  svg: {
    height: 80,
    width: "auto"
  }
});
var ROLE_PB_SETTINGS = ["pb:page:crud"];
export default (function () {
  return [].concat(_toConsumableArray(render()), [{
    name: "pb-page-element-mailchimp",
    type: "pb-editor-page-element",
    elementType: "mailchimp",
    toolbar: {
      title: "Mailchimp",
      group: "pb-editor-element-group-form",
      preview: function preview() {
        return /*#__PURE__*/React.createElement(PreviewBox, null, /*#__PURE__*/React.createElement(MailchimpLogo, null));
      }
    },
    settings: ["pb-editor-page-element-settings-delete", "", "pb-editor-page-element-settings-height"],
    target: ["column", "row", "list-item"],
    onCreate: "open-settings",
    render: function render(_ref) {
      var element = _ref.element;
      return /*#__PURE__*/React.createElement(MailchimpElement, {
        element: element
      });
    },
    create: function create() {
      return {
        type: "mailchimp",
        elements: [],
        data: {},
        settings: {}
      };
    }
  }, {
    name: "pb-element-advanced-settings-mailchimp",
    type: "pb-editor-page-element-advanced-settings",
    elementType: "mailchimp",
    render: function render(props) {
      return /*#__PURE__*/React.createElement(Tab, {
        label: "Mailchimp"
      }, /*#__PURE__*/React.createElement(MailchimpElementAdvancedSettings, props));
    }
  }, {
    type: "route",
    name: "route-settings-page-builder-mailchimp",
    route: /*#__PURE__*/React.createElement(Route, {
      path: "/settings/page-builder/mailchimp",
      render: function render() {
        return /*#__PURE__*/React.createElement(AdminLayout, null, /*#__PURE__*/React.createElement(Helmet, {
          title: "Page Builder - Mailchimp Settings"
        }), /*#__PURE__*/React.createElement(SecureRoute, {
          scopes: ROLE_PB_SETTINGS
        }, /*#__PURE__*/React.createElement(MailchimpSettings, null)));
      }
    })
  }, {
    type: "menu-settings-page-builder",
    name: "menu-settings-page-builder-mailchimp",
    render: function render(_ref2) {
      var Item = _ref2.Item;
      return /*#__PURE__*/React.createElement(Item, {
        label: t(_templateObject()),
        path: "/settings/page-builder/mailchimp"
      });
    }
  }]);
});
