// @webiny/app-headless-cms/admin/plugins/contentDetails/header/publishRevision/PublishRevisionDialog.js
import React from "react";
import { css } from "emotion";
import { Dialog, DialogTitle, DialogContent, DialogActions, DialogAccept, DialogCancel } from "@webiny/ui/Dialog";
import { Select } from "@webiny/ui/Select";
import { Form } from "@webiny/form";
import { validation } from "@webiny/validation";
import { I18NValue } from "@webiny/app-i18n/components";
var narrowDialog = /*#__PURE__*/css({
  ".mdc-dialog__surface": {
    width: 600,
    minWidth: 600
  }
}, "label:narrowDialog;");

var PublishRevisionDialog = function PublishRevisionDialog(_ref) {
  var open = _ref.open,
      onClose = _ref.onClose,
      _onSubmit = _ref.onSubmit,
      revisions = _ref.revisions,
      selected = _ref.selected;
  return /*#__PURE__*/React.createElement(Dialog, {
    open: open,
    onClose: onClose,
    className: narrowDialog
  }, /*#__PURE__*/React.createElement(Form, {
    data: {
      revision: selected || ""
    },
    onSubmit: function onSubmit(_ref2) {
      var revision = _ref2.revision;
      return _onSubmit(revisions.find(function (r) {
        return r.id === revision;
      }));
    }
  }, function (_ref3) {
    var submit = _ref3.submit,
        Bind = _ref3.Bind;
    return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(DialogTitle, null, "Select a revision to publish"), /*#__PURE__*/React.createElement(DialogContent, null, /*#__PURE__*/React.createElement(Bind, {
      name: "revision",
      validators: validation.create("required")
    }, /*#__PURE__*/React.createElement(Select, {
      label: "Revision to publish"
    }, revisions.filter(function (revision) {
      return !revision.meta.published;
    }).map(function (revision) {
      return /*#__PURE__*/React.createElement("option", {
        key: revision.id,
        value: revision.id
      }, I18NValue({
        value: revision.meta.title
      }), "(#", revision.meta.version, ")");
    })))), /*#__PURE__*/React.createElement(DialogActions, null, /*#__PURE__*/React.createElement(DialogCancel, null, "Cancel"), /*#__PURE__*/React.createElement(DialogAccept, {
      onClick: submit
    }, "Publish")));
  }));
};

export default PublishRevisionDialog;
