// @webiny/app-form-builder/editor/plugins/formSettings/components/TermsOfServiceSettings.js
import _get from "lodash/get";
import * as React from "react";
import { Grid, Cell } from "@webiny/ui/Grid";
import { I18NInput } from "@webiny/app-i18n/admin/components";
import { Switch } from "@webiny/ui/Switch";

var TermsOfServiceSettings = function TermsOfServiceSettings(_ref) {
  var Bind = _ref.Bind,
      formData = _ref.formData;

  var enabled = _get(formData, "termsOfServiceMessage.enabled");

  return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(Grid, null, /*#__PURE__*/React.createElement(Cell, {
    span: 12
  }, /*#__PURE__*/React.createElement(Bind, {
    name: "termsOfServiceMessage.enabled"
  }, /*#__PURE__*/React.createElement(Switch, {
    label: "Enabled"
  }))), /*#__PURE__*/React.createElement(Cell, {
    span: 12
  }, /*#__PURE__*/React.createElement(Bind, {
    name: "termsOfServiceMessage.message"
  }, /*#__PURE__*/React.createElement(I18NInput, {
    disabled: !enabled,
    richText: true,
    label: "Terms of service message",
    description: "Show this message near the submit button or event ask users to accept before submitting."
  }))), /*#__PURE__*/React.createElement(Cell, {
    span: 12
  }, /*#__PURE__*/React.createElement(Bind, {
    name: "termsOfServiceMessage.errorMessage"
  }, /*#__PURE__*/React.createElement(I18NInput, {
    disabled: !enabled,
    label: "Error message",
    description: "Show this message once user submits the form without accepting the terms."
  })))));
};

export default TermsOfServiceSettings;
