// @webiny/app-headless-cms/admin/plugins/fields/text.js
import _taggedTemplateLiteral from "@babel/runtime/helpers/taggedTemplateLiteral";

function _templateObject6() {
  var data = _taggedTemplateLiteral(["Value"]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = _taggedTemplateLiteral(["Placeholder text (optional)"]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _taggedTemplateLiteral(["Placeholder text"]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _taggedTemplateLiteral(["Use as a list of texts"]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteral(["Titles, names, single line values."]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteral(["Text"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

import React from "react";
import { ReactComponent as TextIcon } from "@svgr/webpack!./icons/round-text_fields-24px.svg";
import { Grid, Cell } from "@webiny/ui/Grid";
import { I18NInput } from "@webiny/app-i18n/admin/components";
import { i18n } from "@webiny/app/i18n";
import { Input } from "@webiny/ui/Input";
import PredefinedValuesDynamicFieldset from "./PredefinedValuesDynamicFieldset";
var t = i18n.ns("app-headless-cms/admin/fields");
var plugin = {
  type: "cms-editor-field-type",
  name: "cms-editor-field-type-text",
  field: {
    type: "text",
    validators: ["required", "minLength", "maxLength", "pattern"],
    label: t(_templateObject()),
    description: t(_templateObject2()),
    icon: /*#__PURE__*/React.createElement(TextIcon, null),
    allowMultipleValues: true,
    allowPredefinedValues: true,
    allowIndexes: {
      singleValue: true,
      multipleValues: false
    },
    multipleValuesLabel: t(_templateObject3()),
    createField: function createField() {
      return {
        type: this.type,
        validation: [],
        renderer: {
          name: ""
        }
      };
    },
    renderSettings: function renderSettings(_ref) {
      var Bind = _ref.form.Bind;
      return /*#__PURE__*/React.createElement(Grid, null, /*#__PURE__*/React.createElement(Cell, {
        span: 12
      }, /*#__PURE__*/React.createElement(Bind, {
        name: "placeholderText"
      }, /*#__PURE__*/React.createElement(I18NInput, {
        label: t(_templateObject4()),
        description: t(_templateObject5())
      }))));
    },
    renderPredefinedValues: function renderPredefinedValues(props) {
      return /*#__PURE__*/React.createElement(PredefinedValuesDynamicFieldset, props, /*#__PURE__*/React.createElement(Input, {
        label: t(_templateObject6())
      }));
    }
  }
};
export default plugin;
