// @webiny/app-page-builder/editor/plugins/slate/italic/index.js
import React from "react";
import { ReactComponent as FormatItalicIcon } from "@svgr/webpack!@webiny/app-page-builder/editor/assets/icons/format_italic.svg";
import { isKeyHotkey } from "is-hotkey";
var isItalicHotkey = isKeyHotkey("mod+i");

var hasMark = function hasMark(value, type) {
  return Boolean(value.activeMarks.find(function (mark) {
    return mark.type === type;
  }));
};

var onClickMark = function onClickMark(type, onChange, editor) {
  editor.change(function (change) {
    return onChange(change.toggleMark(type));
  });
};

var mark = "italic";
export default (function () {
  return {
    menu: [{
      name: "pb-editor-slate-menu-item-italic",
      type: "pb-editor-slate-menu-item",
      render: function render(_ref) {
        var MenuButton = _ref.MenuButton,
            editor = _ref.editor,
            onChange = _ref.onChange;
        var isActive = hasMark(editor.value, mark);
        return (
          /*#__PURE__*/
          // eslint-disable-next-line react/jsx-no-bind
          React.createElement(MenuButton, {
            onClick: function onClick() {
              return onClickMark(mark, onChange, editor);
            },
            active: isActive
          }, /*#__PURE__*/React.createElement(FormatItalicIcon, null))
        );
      }
    }],
    editor: [{
      name: "pb-editor-slate-editor-italic",
      type: "pb-editor-slate-editor",
      slate: {
        onKeyDown: function onKeyDown(event, change, next) {
          if (isItalicHotkey(event)) {
            event.preventDefault();
            change.toggleMark(mark);
            return true;
          }

          return next();
        },
        renderMark: function renderMark(props, next) {
          if (props.mark.type === mark) {
            return /*#__PURE__*/React.createElement("em", props.attributes, props.children);
          }

          return next();
        }
      }
    }]
  };
});
