// @webiny/app-plugin-security-cognito/cognito/states/SetNewPassword.js
import _objectWithoutProperties from "@babel/runtime/helpers/objectWithoutProperties";
import _regeneratorRuntime from "@babel/runtime/regenerator";
import _asyncToGenerator from "@babel/runtime/helpers/asyncToGenerator";
import _classCallCheck from "@babel/runtime/helpers/classCallCheck";
import _createClass from "@babel/runtime/helpers/createClass";
import _assertThisInitialized from "@babel/runtime/helpers/assertThisInitialized";
import _inherits from "@babel/runtime/helpers/inherits";
import _createSuper from "@babel/runtime/helpers/createSuper";
import _defineProperty from "@babel/runtime/helpers/defineProperty";
import React from "react";
import Auth from "@aws-amplify/auth";
import debug from "./../debug";

var SetNewPassword = /*#__PURE__*/function (_React$Component) {
  _inherits(SetNewPassword, _React$Component);

  var _super = _createSuper(SetNewPassword);

  function SetNewPassword() {
    var _this;

    _classCallCheck(this, SetNewPassword);

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = _super.call.apply(_super, [this].concat(args));

    _defineProperty(_assertThisInitialized(_this), "authStates", ["setNewPassword"]);

    _defineProperty(_assertThisInitialized(_this), "state", {
      error: null,
      loading: false
    });

    _defineProperty(_assertThisInitialized(_this), "setPassword", /*#__PURE__*/function () {
      var _ref = _asyncToGenerator( /*#__PURE__*/_regeneratorRuntime.mark(function _callee(data) {
        var code, password, authData, res;
        return _regeneratorRuntime.wrap(function _callee$(_context) {
          while (1) {
            switch (_context.prev = _context.next) {
              case 0:
                _this.setState({
                  loading: true
                });

                code = data.code, password = data.password;
                authData = _this.props.authData;
                _context.prev = 3;
                _context.next = 6;
                return Auth.forgotPasswordSubmit(authData.toLowerCase(), code, password);

              case 6:
                res = _context.sent;
                debug("set-new-password:setPassword %O", res);

                _this.props.changeState("signIn", null, {
                  title: "Password updated",
                  text: "You can now login using your new password!",
                  type: "success"
                });

                _this.setState({
                  loading: false,
                  error: null
                });

                _context.next = 15;
                break;

              case 12:
                _context.prev = 12;
                _context.t0 = _context["catch"](3);

                _this.setState({
                  loading: false,
                  error: _context.t0.message
                });

              case 15:
              case "end":
                return _context.stop();
            }
          }
        }, _callee, null, [[3, 12]]);
      }));

      return function (_x) {
        return _ref.apply(this, arguments);
      };
    }());

    return _this;
  }

  _createClass(SetNewPassword, [{
    key: "render",
    value: function render() {
      var _this$props = this.props,
          children = _this$props.children,
          authProps = _objectWithoutProperties(_this$props, ["children"]);

      var authState = authProps.authState;

      if (!this.authStates.includes(authState)) {
        return null;
      }

      return /*#__PURE__*/React.cloneElement(children, {
        authProps: authProps,
        setPassword: this.setPassword,
        error: this.state.error,
        loading: this.state.loading
      });
    }
  }]);

  return SetNewPassword;
}(React.Component);

export default SetNewPassword;
