// @webiny/app-headless-cms/admin/plugins/contentDetails/header/contentFormOptionsMenu/ContentFormOptionsMenu.js
import _regeneratorRuntime from "@babel/runtime/regenerator";
import _asyncToGenerator from "@babel/runtime/helpers/asyncToGenerator";
import _taggedTemplateLiteral from "@babel/runtime/helpers/taggedTemplateLiteral";
import _slicedToArray from "@babel/runtime/helpers/slicedToArray";

function _templateObject4() {
  var data = _taggedTemplateLiteral(["{title} was deleted successfully!"]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _taggedTemplateLiteral(["Could not delete content"]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteral(["You are about to delete this content entry and all of its revisions!\n                    Are you sure you want to permanently delete {title}?"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteral(["Delete content entry"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

import React, { useCallback, useMemo } from "react";
import { css } from "emotion";
import get from "lodash/get";
import { IconButton } from "@webiny/ui/Button";
import I18NValue from "@webiny/app-i18n/components/I18NValue";
import { Menu, MenuItem } from "@webiny/ui/Menu";
import { ListItemGraphic } from "@webiny/ui/List";
import { Icon } from "@webiny/ui/Icon";
import { useRouter } from "@webiny/react-router";
import { useSnackbar } from "@webiny/app-admin/hooks/useSnackbar";
import { useDialog } from "@webiny/app-admin/hooks/useDialog";
import { createDeleteMutation } from "@webiny/app-headless-cms/admin/components/ContentModelForm/graphql";
import { useMutation } from "@webiny/app-headless-cms/admin/hooks";
import { useConfirmationDialog } from "@webiny/app-admin/hooks/useConfirmationDialog";
import { i18n } from "@webiny/app/i18n";
var t = i18n.ns("app-headless-cms/admin/plugins/content-details/header/content-form-options-menu");
import { ReactComponent as MoreVerticalIcon } from "@svgr/webpack!@webiny/app-headless-cms/admin/icons/more_vert.svg";
import { ReactComponent as DeleteIcon } from "@svgr/webpack!@webiny/app-headless-cms/admin/icons/delete.svg";
var menuStyles = /*#__PURE__*/css({
  width: 250,
  right: -105,
  left: "auto !important",
  ".disabled": {
    opacity: 0.5,
    pointerEvents: "none"
  }
}, "label:menuStyles;");

var ContentFormOptionsMenu = function ContentFormOptionsMenu(_ref) {
  var contentModel = _ref.contentModel,
      content = _ref.content,
      dataList = _ref.dataList,
      getLoading = _ref.getLoading,
      setLoading = _ref.setLoading;

  var _useSnackbar = useSnackbar(),
      showSnackbar = _useSnackbar.showSnackbar;

  var _useRouter = useRouter(),
      history = _useRouter.history;

  var _useDialog = useDialog(),
      showDialog = _useDialog.showDialog;

  var DELETE_CONTENT = useMemo(function () {
    return createDeleteMutation(contentModel);
  }, [contentModel.modelId]);

  var _useMutation = useMutation(DELETE_CONTENT),
      _useMutation2 = _slicedToArray(_useMutation, 1),
      deleteContentMutation = _useMutation2[0];

  var title = I18NValue({
    value: get(content, "meta.title")
  });

  var _useConfirmationDialo = useConfirmationDialog({
    title: t(_templateObject()),
    message: /*#__PURE__*/React.createElement("p", null, t(_templateObject2())({
      title: /*#__PURE__*/React.createElement("strong", null, title)
    }))
  }),
      showConfirmation = _useConfirmationDialo.showConfirmation;

  var confirmDelete = useCallback(function () {
    showConfirmation( /*#__PURE__*/_asyncToGenerator( /*#__PURE__*/_regeneratorRuntime.mark(function _callee() {
      var _yield$deleteContentM, res, _get, error, query;

      return _regeneratorRuntime.wrap(function _callee$(_context) {
        while (1) {
          switch (_context.prev = _context.next) {
            case 0:
              setLoading(true);
              _context.next = 3;
              return deleteContentMutation({
                variables: {
                  revision: content.meta.parent
                }
              });

            case 3:
              _yield$deleteContentM = _context.sent;
              res = _yield$deleteContentM.data;
              setLoading(false);
              dataList.refresh();
              _get = get(res, "content"), error = _get.error;

              if (!error) {
                _context.next = 10;
                break;
              }

              return _context.abrupt("return", showDialog(error.message, {
                title: t(_templateObject3())
              }));

            case 10:
              showSnackbar(t(_templateObject4())({
                title: /*#__PURE__*/React.createElement("strong", null, title)
              }));
              query = new URLSearchParams(location.search);
              query.delete("id");
              history.push({
                search: query.toString()
              });

            case 14:
            case "end":
              return _context.stop();
          }
        }
      }, _callee);
    })));
  }, null);
  return /*#__PURE__*/React.createElement(Menu, {
    className: menuStyles,
    handle: /*#__PURE__*/React.createElement(IconButton, {
      icon: /*#__PURE__*/React.createElement(MoreVerticalIcon, null)
    })
  }, /*#__PURE__*/React.createElement(MenuItem, {
    onClick: confirmDelete,
    disabled: !content.id || getLoading()
  }, /*#__PURE__*/React.createElement(ListItemGraphic, null, /*#__PURE__*/React.createElement(Icon, {
    icon: /*#__PURE__*/React.createElement(DeleteIcon, null)
  })), "Delete"));
};

export default ContentFormOptionsMenu;
