// @webiny/app-page-builder/editor/plugins/elements/social/pinterest/index.js
import React from "react";
import styled from "@emotion/styled";
import { Tab } from "@webiny/ui/Tabs";
import { Input } from "@webiny/ui/Input";
import { Select } from "@webiny/ui/Select";
import { Grid, Cell } from "@webiny/ui/Grid";
import { ReactComponent as SocialIcon } from "@svgr/webpack!./../../../elementGroups/social/round-people-24px.svg";
import { createEmbedPlugin, createEmbedSettingsPlugin } from "./../../utils/oembed/createEmbedPlugin";
import PinterestEmbed from "./PinterestEmbed";
import { validation } from "@webiny/validation";
import { ReactComponent as LogoIcon } from "@svgr/webpack!./pinterest-brands.svg";
var PreviewBox = /*#__PURE__*/styled("div", {
  target: "e6sbjxh0",
  label: "PreviewBox"
})({
  textAlign: "center",
  height: 50,
  svg: {
    height: 50,
    width: 50
  }
});
export default (function () {
  return [createEmbedPlugin({
    type: "pinterest",
    toolbar: {
      title: "Pinterest post",
      group: "pb-editor-element-group-social",
      preview: function preview() {
        return /*#__PURE__*/React.createElement(PreviewBox, null, /*#__PURE__*/React.createElement(LogoIcon, null));
      }
    },
    render: function render(_ref) {
      var element = _ref.element;
      return /*#__PURE__*/React.createElement(PinterestEmbed, {
        element: element
      });
    }
  }), createEmbedSettingsPlugin({
    type: "pinterest",
    render: function render(_ref2) {
      var Bind = _ref2.Bind;
      return /*#__PURE__*/React.createElement(Tab, {
        icon: /*#__PURE__*/React.createElement(SocialIcon, null),
        label: "Pinterest"
      }, /*#__PURE__*/React.createElement(Grid, null, /*#__PURE__*/React.createElement(Cell, {
        span: 12
      }, /*#__PURE__*/React.createElement(Bind, {
        name: "source.url",
        validators: validation.create("required,url")
      }, /*#__PURE__*/React.createElement(Input, {
        label: "Pinterest URL",
        description: "Enter a Pinterest URL"
      }))), /*#__PURE__*/React.createElement(Cell, {
        span: 12
      }, /*#__PURE__*/React.createElement(Bind, {
        defaultValue: "small",
        name: "source.size"
      }, /*#__PURE__*/React.createElement(Select, {
        label: "Size"
      }, /*#__PURE__*/React.createElement("option", {
        value: "small"
      }, "Small"), /*#__PURE__*/React.createElement("option", {
        value: "medium"
      }, "Medium"), /*#__PURE__*/React.createElement("option", {
        value: "large"
      }, "Large"))))));
    }
  })];
});
