// @webiny/app-page-builder/editor/plugins/elements/social/instagram/index.js
import React from "react";
import styled from "@emotion/styled";
import { Tab } from "@webiny/ui/Tabs";
import { Input } from "@webiny/ui/Input";
import { Grid, Cell } from "@webiny/ui/Grid";
import { ReactComponent as SocialIcon } from "@svgr/webpack!./../../../elementGroups/social/round-people-24px.svg";
import placeholder from "./placeholder.jpg";
import { createEmbedPlugin, createEmbedSettingsPlugin } from "./../../utils/oembed/createEmbedPlugin";
import { ReactComponent as LogoIcon } from "@svgr/webpack!./instagram-brands.svg";
import { validation } from "@webiny/validation";
var PreviewBox = /*#__PURE__*/styled("div", {
  target: "ed58sfq0",
  label: "PreviewBox"
})({
  textAlign: "center",
  height: 50,
  svg: {
    height: 50,
    width: 50
  }
});
export default (function () {
  return [createEmbedPlugin({
    type: "instagram",
    toolbar: {
      title: "Instagram Post",
      group: "pb-editor-element-group-social",
      preview: function preview() {
        return /*#__PURE__*/React.createElement(PreviewBox, null, /*#__PURE__*/React.createElement(LogoIcon, null));
      }
    },
    oembed: {
      global: "instgrm",
      sdk: "https://www.instagram.com/embed.js",
      init: function init(_ref) {
        var node = _ref.node;
        // @ts-ignore
        window.instgrm.Embeds.process(node.firstChild);
      }
    },
    renderElementPreview: function renderElementPreview(_ref2) {
      var width = _ref2.width,
          height = _ref2.height;
      return /*#__PURE__*/React.createElement("img", {
        style: {
          width: width,
          height: height
        },
        src: placeholder,
        alt: "Instagram"
      });
    }
  }), createEmbedSettingsPlugin({
    type: "instagram",
    render: function render(_ref3) {
      var Bind = _ref3.Bind;
      return /*#__PURE__*/React.createElement(Tab, {
        icon: /*#__PURE__*/React.createElement(SocialIcon, null),
        label: "Instagram"
      }, /*#__PURE__*/React.createElement(Grid, null, /*#__PURE__*/React.createElement(Cell, {
        span: 12
      }, /*#__PURE__*/React.createElement(Bind, {
        name: "source.url",
        validators: validation.create("required,url")
      }, /*#__PURE__*/React.createElement(Input, {
        label: "Instagram URL",
        description: "Enter an Instagram URL"
      })))));
    }
  })];
});
