// @webiny/app-page-builder/editor/plugins/elementSettings/border/Settings.js
import _isEqual from "lodash/isEqual";
import * as React from "react";
import { connect } from "@webiny/app-page-builder/editor/redux";
import { Tabs, Tab } from "@webiny/ui/Tabs";
import { updateElement } from "@webiny/app-page-builder/editor/actions";
import { getActiveElement } from "@webiny/app-page-builder/editor/selectors";
import ColorPicker from "@webiny/app-page-builder/editor/plugins/elementSettings/components/ColorPicker";
import Select from "@webiny/app-page-builder/editor/plugins/elementSettings/components/Select";
import Slider from "@webiny/app-page-builder/editor/plugins/elementSettings/components/Slider";
import Selector from "./Selector";
import useUpdateHandlers from "../useUpdateHandlers";
var options = ["none", "solid", "dashed", "dotted"];
var DATA_NAMESPACE = "data.settings.border";
var EMPTY_OBJECT = {};

var Settings = function Settings(_ref) {
  var element = _ref.element,
      updateElement = _ref.updateElement;

  var _useUpdateHandlers = useUpdateHandlers({
    element: element,
    updateElement: updateElement,
    dataNamespace: DATA_NAMESPACE
  }),
      getUpdateValue = _useUpdateHandlers.getUpdateValue,
      getUpdatePreview = _useUpdateHandlers.getUpdatePreview;

  return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(Tabs, null, /*#__PURE__*/React.createElement(Tab, {
    label: "Border"
  }, /*#__PURE__*/React.createElement(ColorPicker, {
    label: "Color",
    valueKey: DATA_NAMESPACE + ".color",
    defaultValue: "#fff",
    updateValue: getUpdateValue("color"),
    updatePreview: getUpdatePreview("color")
  }), /*#__PURE__*/React.createElement(Slider, {
    label: "Width",
    valueKey: DATA_NAMESPACE + ".width",
    defaultValue: 0,
    updateValue: getUpdateValue("width"),
    updatePreview: getUpdatePreview("width"),
    min: 0,
    max: 20,
    step: 1
  }), /*#__PURE__*/React.createElement(Slider, {
    label: "Radius",
    valueKey: DATA_NAMESPACE + ".radius",
    defaultValue: 0,
    updateValue: getUpdateValue("radius"),
    updatePreview: getUpdatePreview("radius"),
    min: 0,
    max: 100,
    step: 1
  }), /*#__PURE__*/React.createElement(Select, {
    label: "Style",
    valueKey: DATA_NAMESPACE + ".style",
    defaultValue: "none",
    updateValue: getUpdateValue("style"),
    options: options
  }), /*#__PURE__*/React.createElement(Selector, {
    label: "Borders",
    valueKey: DATA_NAMESPACE + ".borders",
    defaultValue: EMPTY_OBJECT,
    updateValue: getUpdateValue("borders")
  }))));
};

export default connect(function (state) {
  var _getActiveElement = getActiveElement(state),
      id = _getActiveElement.id,
      type = _getActiveElement.type,
      path = _getActiveElement.path;

  return {
    element: {
      id: id,
      type: type,
      path: path
    }
  };
}, {
  updateElement: updateElement
}, null, {
  areStatePropsEqual: _isEqual
})(Settings);
