// @webiny/app-i18n/admin/plugins/richTextEditor/underline/index.js
import * as React from "react";
import { ReactComponent as UnderlineIcon } from "@svgr/webpack!./../icons/format_underlined.svg";
import { isKeyHotkey } from "is-hotkey";
var isUnderlineHotkey = isKeyHotkey("mod+u");
var mark = "underline";
var plugin = {
  name: "i18n-input-rich-text-editor-underline",
  type: "i18n-input-rich-text-editor",
  plugin: {
    name: "underline",
    menu: {
      render: function render(_ref) {
        var MenuButton = _ref.MenuButton,
            editor = _ref.editor;
        return /*#__PURE__*/React.createElement(MenuButton, {
          onClick: function onClick() {
            return editor.toggleMark(mark);
          },
          active: editor.hasMark(mark)
        }, /*#__PURE__*/React.createElement(UnderlineIcon, null));
      }
    },
    editor: {
      onKeyDown: function onKeyDown(_ref2, next) {
        var event = _ref2.event,
            editor = _ref2.editor;

        // Decide what to do based on the key code...
        if (isUnderlineHotkey(event)) {
          event.preventDefault();
          editor.toggleMark(mark);
          return true;
        }

        return next();
      },
      renderLeaf: function renderLeaf(_ref3) {
        var leaf = _ref3.leaf,
            attributes = _ref3.attributes,
            children = _ref3.children;

        if (leaf[mark] === true) {
          return /*#__PURE__*/React.createElement("u", attributes, children);
        }

        return children;
      }
    }
  }
};
export default plugin;
