// @webiny/app-page-builder/editor/plugins/blockEditing/AddContent.js
import React from "react";
import styled from "@emotion/styled";
import { keyframes } from "emotion";
import { connect } from "@webiny/app-page-builder/editor/redux";
import { Elevation } from "@webiny/ui/Elevation";
import { ButtonFloating } from "@webiny/ui/Button";
import { togglePlugin } from "@webiny/app-page-builder/editor/actions";
import { getContent } from "@webiny/app-page-builder/editor/selectors";
import { ReactComponent as AddIcon } from "@svgr/webpack!@webiny/app-page-builder/editor/assets/icons/add.svg";
var pulse = /*#__PURE__*/keyframes("0%{box-shadow:0 0 0 0 rgba(0,204,176,0.4);}70%{box-shadow:0 0 0 30px rgba(0,204,176,0);}100%{box-shadow:0 0 0 0 rgba(0,204,176,0);}label:pulse;");
var AddBlockContainer = /*#__PURE__*/styled("div", {
  target: "e1d6k8pe0",
  label: "AddBlockContainer"
})({
  position: "fixed",
  zIndex: 11,
  top: "50%",
  left: "50%",
  transform: "translate(-50%,-50%)",
  display: "flex",
  flexDirection: "column",
  justifyContent: "center",
  alignItems: "center",
  width: "150px",
  height: "100px",
  borderRadius: 2,
  color: "var(--mdc-theme-on-surface)",
  marginLeft: 54
});
var AddBlockContent = /*#__PURE__*/styled("div", {
  target: "e1d6k8pe1",
  label: "AddBlockContent"
})({
  width: 300,
  margin: 5,
  textAlign: "center",
  display: "flex",
  alignItems: "center"
});

var AddContent = function AddContent(_ref) {
  var count = _ref.count,
      togglePlugin = _ref.togglePlugin;

  if (count) {
    return null;
  }

  return /*#__PURE__*/React.createElement(AddBlockContainer, {
    "data-type": "container"
  }, /*#__PURE__*/React.createElement(Elevation, {
    z: 4,
    style: {
      backgroundColor: "var(--mdc-theme-surface)",
      padding: "30px 20px",
      transform: "translateY(-50%)",
      borderRadius: 2
    }
  }, /*#__PURE__*/React.createElement(AddBlockContent, null, "Click the", /*#__PURE__*/React.createElement(ButtonFloating, {
    "data-testid": "pb-content-add-block-button",
    style: {
      animation: pulse + " 3s ease infinite",
      margin: "0 10px"
    },
    small: true,
    icon: /*#__PURE__*/React.createElement(AddIcon, null),
    onClick: function onClick() {
      return togglePlugin({
        name: "pb-editor-search-blocks-bar"
      });
    }
  }), "to start adding content")));
};

export default connect(function (state) {
  return {
    count: getContent(state).elements.length
  };
}, {
  togglePlugin: togglePlugin
})( /*#__PURE__*/React.memo(AddContent));
