// @webiny/app-form-builder/editor/plugins/defaultBar/Name/NameStyled.js
import styled from "@emotion/styled";
import { css } from "emotion";
export var NameInputWrapper = /*#__PURE__*/styled("div", {
  target: "e136jnk10",
  label: "NameInputWrapper"
})({
  width: "100%",
  display: "flex",
  alignItems: "center",
  position: "relative",
  "> .mdc-text-field--upgraded": {
    height: 35,
    marginTop: "0 !important",
    paddingLeft: 10,
    paddingRight: 40
  }
});
export var NameWrapper = /*#__PURE__*/styled("div", {
  target: "e136jnk11",
  label: "NameWrapper"
})({
  height: 50,
  display: "flex",
  alignItems: "baseline",
  justifyContent: "flex-start",
  flexDirection: "column",
  color: "var(--mdc-theme-text-primary-on-background)",
  position: "relative",
  width: "100%",
  marginLeft: 10
});
export var FormName = /*#__PURE__*/styled("div", {
  target: "e136jnk12",
  label: "FormName"
})({
  border: "1px solid transparent",
  fontSize: 20,
  whiteSpace: "nowrap",
  overflow: "hidden",
  textOverflow: "ellipsis",
  width: "100%",
  lineHeight: "120%",
  "&:hover": {
    border: "1px solid var(--mdc-theme-on-background)"
  }
});
export var formNameWrapper = /*#__PURE__*/css({
  maxWidth: "calc(100% - 50px)"
}, "label:formNameWrapper;");
export var FormVersion = /*#__PURE__*/styled("span", {
  target: "e136jnk13",
  label: "FormVersion"
})({
  fontSize: 20,
  color: "var(--mdc-theme-text-secondary-on-background)",
  marginLeft: 5,
  lineHeight: "120%"
});
export var FormMeta = /*#__PURE__*/styled("div", {
  target: "e136jnk14",
  label: "FormMeta"
})({
  height: 20,
  margin: "-2px 2px 2px 2px"
});
