// @webiny/app-form-builder/admin/plugins/formDetails/previewContent/HeaderComponents/EditRevision.js
import React from "react";
import { useRouter } from "@webiny/react-router";
import { IconButton } from "@webiny/ui/Button";
import { Tooltip } from "@webiny/ui/Tooltip";
import { ReactComponent as EditIcon } from "@svgr/webpack!@webiny/app-form-builder/admin/icons/edit.svg";
import { useRevision } from "../../formRevisions/useRevision";

var EditRevision = function EditRevision(_ref) {
  var revision = _ref.revision,
      form = _ref.form;

  var _useRevision = useRevision({
    revision: revision,
    form: form
  }),
      createRevision = _useRevision.createRevision;

  var _useRouter = useRouter(),
      history = _useRouter.history;

  if (revision.status === "draft") {
    return /*#__PURE__*/React.createElement(Tooltip, {
      content: "Edit",
      placement: "top"
    }, /*#__PURE__*/React.createElement(IconButton, {
      icon: /*#__PURE__*/React.createElement(EditIcon, null),
      onClick: function onClick() {
        return history.push("/forms/".concat(revision.id));
      }
    }));
  }

  return /*#__PURE__*/React.createElement(Tooltip, {
    content: "New draft based on this version...",
    placement: "top"
  }, /*#__PURE__*/React.createElement(IconButton, {
    icon: /*#__PURE__*/React.createElement(EditIcon, null),
    onClick: createRevision
  }));
};

export default EditRevision;
