// @webiny/app-form-builder/admin/plugins/formDetails/previewContent/HeaderComponents/RevisionSelector.js
import _get from "lodash/get";
import React from "react";
import { css } from "emotion";
import { ButtonDefault } from "@webiny/ui/Button";
import { Icon } from "@webiny/ui/Icon";
import { ReactComponent as DownButton } from "@svgr/webpack!@webiny/app-form-builder/admin/icons/round-arrow_drop_down-24px.svg";
import { MenuItem } from "@rmwc/menu";
import { Typography } from "@webiny/ui/Typography";
import { Menu } from "@webiny/ui/Menu";
var buttonStyle = /*#__PURE__*/css({
  "&.mdc-button": {
    color: "var(--mdc-theme-text-primary-on-background) !important"
  }
}, "label:buttonStyle;");
var menuList = /*#__PURE__*/css({
  ".mdc-list-item": {
    flexDirection: "column",
    justifyContent: "center",
    alignItems: "baseline",
    textAlign: "left"
  }
}, "label:menuList;");

var RevisionSelector = function RevisionSelector(_ref) {
  var revision = _ref.revision,
      form = _ref.form,
      selectRevision = _ref.selectRevision;
  return /*#__PURE__*/React.createElement(Menu, {
    className: menuList,
    onSelect: function onSelect(evt) {
      return selectRevision(form.revisions[evt.detail.index]);
    },
    handle: /*#__PURE__*/React.createElement(ButtonDefault, {
      className: buttonStyle
    }, "v", revision.version, " ", /*#__PURE__*/React.createElement(Icon, {
      icon: /*#__PURE__*/React.createElement(DownButton, null)
    }))
  }, (_get(form, "revisions") || []).map(function (rev) {
    return /*#__PURE__*/React.createElement(MenuItem, {
      key: rev.id
    }, /*#__PURE__*/React.createElement(Typography, {
      use: "body2"
    }, "v", rev.version), /*#__PURE__*/React.createElement(Typography, {
      use: "caption"
    }, "(", rev.status, ")"));
  }));
};

export default RevisionSelector;
