// @webiny/app-headless-cms/admin/plugins/routes.js
import _taggedTemplateLiteral from "@babel/runtime/helpers/taggedTemplateLiteral";
import _objectWithoutProperties from "@babel/runtime/helpers/objectWithoutProperties";

function _templateObject7() {
  var data = _taggedTemplateLiteral(["Headless CMS - Access Tokens Settings"]);

  _templateObject7 = function _templateObject7() {
    return data;
  };

  return data;
}

function _templateObject6() {
  var data = _taggedTemplateLiteral(["Headless CMS - Environment Aliases Settings"]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = _taggedTemplateLiteral(["Headless CMS - Environments Settings"]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _taggedTemplateLiteral(["Content Models"]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _taggedTemplateLiteral(["Edit Content Model"]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteral(["Content"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteral(["Content Model Groups"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

import React, { Suspense, lazy } from "react";
import Helmet from "react-helmet";
import { SecureRoute } from "@webiny/app-security/components";
import { CircularProgress } from "@webiny/ui/Progress";
import { Route } from "@webiny/react-router";
import { AdminLayout } from "@webiny/app-admin/components/AdminLayout";
import { i18n } from "@webiny/app/i18n";
var t = i18n.ns("app-headless-cms/admin/routes");

var Loader = function Loader(_ref) {
  var children = _ref.children,
      props = _objectWithoutProperties(_ref, ["children"]);

  return /*#__PURE__*/React.createElement(Suspense, {
    fallback: /*#__PURE__*/React.createElement(CircularProgress, null)
  }, /*#__PURE__*/React.cloneElement(children, props));
};

var ContentModelEditor = /*#__PURE__*/lazy(function () {
  return import("../views/Editor");
});
var ContentModelsView = /*#__PURE__*/lazy(function () {
  return import("../views/ContentModels/ContentModels");
});
var ContentModelGroupsView = /*#__PURE__*/lazy(function () {
  return import("../views/ContentModelGroups/ContentModelGroups");
});
var EnvironmentsView = /*#__PURE__*/lazy(function () {
  return import("../views/Environments/Environments");
});
var EnvironmentAliasesView = /*#__PURE__*/lazy(function () {
  return import("../views/EnvironmentAliases/EnvironmentAliases");
});
var AccessTokensView = /*#__PURE__*/lazy(function () {
  return import("../views/AccessTokens");
});
var ContentView = /*#__PURE__*/lazy(function () {
  return import("../views/Content/Content");
});
var ROLE_CMS_CONTENT_GROUPS = ["cms:content-model-group:crud"];
var ROLE_CMS_ENVIRONMENT = ["cms:environment:crud"];
var ROLE_CMS_ENVIRONMENT_ALIAS = ["cms:environment-alias:crud"];
var ROLE_CMS_ACCESS_TOKENS = ["cms:access-tokens:crud"];
var ROLE_CMS_CONTENT_MODELS = ["cms:content-model:crud"];
var plugins = [{
  name: "route-cms-content-models-groups",
  type: "route",
  route: /*#__PURE__*/React.createElement(Route, {
    exact: true,
    path: "/cms/content-model-groups",
    render: function render() {
      return /*#__PURE__*/React.createElement(SecureRoute, {
        scopes: ROLE_CMS_CONTENT_GROUPS
      }, /*#__PURE__*/React.createElement(AdminLayout, null, /*#__PURE__*/React.createElement(Helmet, null, /*#__PURE__*/React.createElement("title", null, t(_templateObject()))), /*#__PURE__*/React.createElement(Loader, null, /*#__PURE__*/React.createElement(ContentModelGroupsView, null))));
    }
  })
}, {
  name: "route-cms-content-models-manage",
  type: "route",
  route: /*#__PURE__*/React.createElement(Route, {
    exact: true,
    path: "/cms/content-models/manage/:modelId",
    render: function render() {
      return /*#__PURE__*/React.createElement(SecureRoute, {
        scopes: ROLE_CMS_CONTENT_MODELS
      }, /*#__PURE__*/React.createElement(AdminLayout, null, /*#__PURE__*/React.createElement(Helmet, null, /*#__PURE__*/React.createElement("title", null, t(_templateObject2()))), /*#__PURE__*/React.createElement(Loader, null, /*#__PURE__*/React.createElement(ContentView, null))));
    }
  })
}, {
  name: "route-cms-content-models-editor",
  type: "route",
  route: /*#__PURE__*/React.createElement(Route, {
    exact: true,
    path: "/cms/content-models/:id",
    render: function render() {
      return /*#__PURE__*/React.createElement(SecureRoute, {
        scopes: ROLE_CMS_CONTENT_MODELS
      }, /*#__PURE__*/React.createElement(Helmet, null, /*#__PURE__*/React.createElement("title", null, t(_templateObject3()))), /*#__PURE__*/React.createElement(Loader, null, /*#__PURE__*/React.createElement(ContentModelEditor, null)));
    }
  })
}, {
  name: "route-cms-content-models",
  type: "route",
  route: /*#__PURE__*/React.createElement(Route, {
    exact: true,
    path: "/cms/content-models",
    render: function render() {
      return /*#__PURE__*/React.createElement(SecureRoute, {
        scopes: ROLE_CMS_CONTENT_MODELS
      }, /*#__PURE__*/React.createElement(AdminLayout, null, /*#__PURE__*/React.createElement(Helmet, {
        title: t(_templateObject4())
      }), /*#__PURE__*/React.createElement(Loader, null, /*#__PURE__*/React.createElement(ContentModelsView, null))));
    }
  })
}, {
  type: "route",
  name: "route-settings-cms-environments",
  route: /*#__PURE__*/React.createElement(Route, {
    exact: true,
    path: "/settings/cms/environments",
    render: function render() {
      return /*#__PURE__*/React.createElement(SecureRoute, {
        scopes: ROLE_CMS_ENVIRONMENT
      }, /*#__PURE__*/React.createElement(AdminLayout, null, /*#__PURE__*/React.createElement(Helmet, {
        title: t(_templateObject5())
      }), /*#__PURE__*/React.createElement(Loader, null, /*#__PURE__*/React.createElement(EnvironmentsView, null))));
    }
  })
}, {
  type: "route",
  name: "route-settings-cms-environment-aliases",
  route: /*#__PURE__*/React.createElement(Route, {
    exact: true,
    path: "/settings/cms/environments/aliases",
    render: function render() {
      return /*#__PURE__*/React.createElement(SecureRoute, {
        scopes: ROLE_CMS_ENVIRONMENT_ALIAS
      }, /*#__PURE__*/React.createElement(AdminLayout, null, /*#__PURE__*/React.createElement(Helmet, {
        title: t(_templateObject6())
      }), /*#__PURE__*/React.createElement(Loader, null, /*#__PURE__*/React.createElement(EnvironmentAliasesView, null))));
    }
  })
}, {
  type: "route",
  name: "route-settings-cms-access-tokens",
  route: /*#__PURE__*/React.createElement(Route, {
    exact: true,
    path: "/settings/cms/accessTokens",
    render: function render() {
      return /*#__PURE__*/React.createElement(SecureRoute, {
        scopes: ROLE_CMS_ACCESS_TOKENS
      }, /*#__PURE__*/React.createElement(AdminLayout, null, /*#__PURE__*/React.createElement(Helmet, {
        title: t(_templateObject7())
      }), /*#__PURE__*/React.createElement(Loader, null, /*#__PURE__*/React.createElement(AccessTokensView, null))));
    }
  })
}];
export default plugins;
