// @webiny/app-form-builder/admin/views/Forms/Forms.js
import _slicedToArray from "@babel/runtime/helpers/slicedToArray";
import React from "react";
import { SplitView, LeftPanel, RightPanel } from "@webiny/app-admin/components/SplitView";
import { FloatingActionButton } from "@webiny/app-admin/components/FloatingActionButton";
import FormsDataList from "./FormsDataList";
import FormDetails from "./FormDetails";
import NewFormDialog from "./NewFormDialog";
import { LIST_FORMS } from "../../viewsGraphql";
import { useDataList } from "@webiny/app/hooks/useDataList";

function Forms() {
  var _React$useState = React.useState(false),
      _React$useState2 = _slicedToArray(_React$useState, 2),
      newFormDialogOpened = _React$useState2[0],
      openNewFormDialog = _React$useState2[1];

  var dataList = useDataList({
    query: LIST_FORMS,
    variables: {
      sort: {
        savedOn: -1
      }
    }
  });
  return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(NewFormDialog, {
    open: newFormDialogOpened,
    onClose: function onClose() {
      return openNewFormDialog(false);
    },
    formsDataList: dataList
  }), /*#__PURE__*/React.createElement(SplitView, null, /*#__PURE__*/React.createElement(LeftPanel, {
    span: 4
  }, /*#__PURE__*/React.createElement(FormsDataList, {
    dataList: dataList
  })), /*#__PURE__*/React.createElement(RightPanel, {
    span: 8
  }, /*#__PURE__*/React.createElement(FormDetails, {
    refreshForms: dataList.refresh
  }))), /*#__PURE__*/React.createElement(FloatingActionButton, {
    "data-testid": "new-record-button",
    onClick: function onClick() {
      return openNewFormDialog(true);
    }
  }));
}

export default Forms;
