// @webiny/app-form-builder/admin/plugins/settings/index.js
import _taggedTemplateLiteral from "@babel/runtime/helpers/taggedTemplateLiteral";

function _templateObject3() {
  var data = _taggedTemplateLiteral(["reCAPTCHA"]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteral(["Form Builder"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteral(["Form Builder - reCAPTCHA Settings"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

import * as React from "react";
import { Route } from "@webiny/react-router";
import { AdminLayout } from "@webiny/app-admin/components/AdminLayout";
import FormsSettings from "./components/FormsSettings";
import { SecureRoute } from "@webiny/app-security/components";
import Helmet from "react-helmet";
import { i18n } from "@webiny/app/i18n";
var t = i18n.ns("app-form-builder/admin/menus");
var ROLE_FORMS_SETTINGS = ["forms:settings"];
var plugins = [{
  type: "route",
  name: "route-settings-form-builder",
  route: /*#__PURE__*/React.createElement(Route, {
    path: "/settings/form-builder/recaptcha",
    render: function render() {
      return /*#__PURE__*/React.createElement(AdminLayout, null, /*#__PURE__*/React.createElement(Helmet, {
        title: t(_templateObject())
      }), /*#__PURE__*/React.createElement(SecureRoute, {
        scopes: ROLE_FORMS_SETTINGS
      }, /*#__PURE__*/React.createElement(FormsSettings, null)));
    }
  })
}, {
  type: "admin-menu-settings",
  name: "menu-settings-form-builder",
  render: function render(_ref) {
    var Item = _ref.Item,
        Section = _ref.Section;
    return /*#__PURE__*/React.createElement(Section, {
      label: t(_templateObject2())
    }, /*#__PURE__*/React.createElement(Item, {
      label: t(_templateObject3()),
      path: "/settings/form-builder/recaptcha"
    }));
  }
}];
export default plugins;
