// @webiny/app-headless-cms/admin/components/ContentModelEditor/Tabs/PreviewTab/PreviewTab.js
import _taggedTemplateLiteral from "@babel/runtime/helpers/taggedTemplateLiteral";

function _templateObject() {
  var data = _taggedTemplateLiteral(["Before previewing the form, please add at least one field to the content model."]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

import React from "react";
import { css } from "emotion";
import { useContentModelEditor } from "@webiny/app-headless-cms/admin/components/ContentModelEditor/Context";
import { Elevation } from "@webiny/ui/Elevation";
import { ContentModelForm } from "@webiny/app-headless-cms/admin/components/ContentModelForm";
import { i18n } from "@webiny/app/i18n";
import { useI18N } from "@webiny/app-i18n/hooks/useI18N";
var t = i18n.ns("app-headless-cms/admin/components/editor/tabs/preview");
var formPreviewWrapper = /*#__PURE__*/css({
  padding: 40,
  backgroundColor: "var(--webiny-theme-color-surface, #fff) !important",
  margin: 40,
  boxSizing: "border-box"
}, "label:formPreviewWrapper;");
var style = {
  noFieldsMessage: /*#__PURE__*/css({
    textAlign: "center"
  }, "label:noFieldsMessage;")
};
export var PreviewTab = function PreviewTab(_ref) {
  var activeTab = _ref.activeTab;

  var _useContentModelEdito = useContentModelEditor(),
      data = _useContentModelEdito.data;

  var i18n = useI18N();
  var fields = data.fields;
  return /*#__PURE__*/React.createElement(Elevation, {
    z: 1,
    className: formPreviewWrapper
  }, fields && fields.length && activeTab ? /*#__PURE__*/React.createElement(ContentModelForm, {
    contentModel: data,
    locale: i18n.getLocale().id
  }) : /*#__PURE__*/React.createElement("div", {
    className: style.noFieldsMessage
  }, t(_templateObject())));
};
