// @webiny/app-page-builder/editor/plugins/elements/pagesList/index.js
import _objectSpread from "@babel/runtime/helpers/objectSpread2";
import React from "react";
import { Tab } from "@webiny/ui/Tabs";
import { ReactComponent as DesignIcon } from "@svgr/webpack!./icons/round-style-24px.svg";
import { ReactComponent as FilterIcon } from "@svgr/webpack!./icons/round-filter_list-24px.svg";
import { ReactComponent as PageListIcon } from "@svgr/webpack!./page-list-icon.svg";
import PagesList from "./PagesList";
import PagesListFilterSettings from "./PagesListFilterSettings";
import PagesListDesignSettings from "./PagesListDesignSettings";
import styled from "@emotion/styled";
export default (function () {
  var PreviewBox = /*#__PURE__*/styled("div", {
    target: "er2nsm10",
    label: "PreviewBox"
  })({
    textAlign: "center",
    margin: "0 auto",
    width: 100,
    svg: {
      width: 100
    }
  });
  return [{
    name: "pb-editor-page-element-pages-list",
    type: "pb-editor-page-element",
    elementType: "pages-list",
    toolbar: {
      title: "List of pages",
      group: "pb-editor-element-group-basic",
      preview: function preview() {
        return /*#__PURE__*/React.createElement(PreviewBox, null, /*#__PURE__*/React.createElement(PageListIcon, null));
      }
    },
    settings: ["pb-editor-page-element-settings-delete"],
    target: ["row", "column"],
    onCreate: "open-settings",
    create: function create() {
      var options = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {};
      return _objectSpread({
        type: "pages-list",
        data: {
          resultsPerPage: 10,
          component: "default",
          settings: {
            margin: {
              desktop: {
                all: 0
              },
              mobile: {
                all: 0
              }
            },
            padding: {
              desktop: {
                all: 0
              },
              mobile: {
                all: 0
              }
            }
          }
        }
      }, options);
    },
    render: function render(_ref) {
      var element = _ref.element;
      return /*#__PURE__*/React.createElement(PagesList, {
        data: element.data
      });
    }
  }, {
    name: "pb-editor-page-element-advanced-settings-pages-list-filter",
    type: "pb-editor-page-element-advanced-settings",
    elementType: "pages-list",
    render: function render(props) {
      return /*#__PURE__*/React.createElement(Tab, {
        icon: /*#__PURE__*/React.createElement(FilterIcon, null),
        label: "Filter"
      }, /*#__PURE__*/React.createElement(PagesListFilterSettings, props));
    }
  }, {
    name: "pb-editor-page-element-advanced-settings-pages-list-design",
    type: "pb-editor-page-element-advanced-settings",
    elementType: "pages-list",
    render: function render(props) {
      return /*#__PURE__*/React.createElement(Tab, {
        icon: /*#__PURE__*/React.createElement(DesignIcon, null),
        label: "Design"
      }, /*#__PURE__*/React.createElement(PagesListDesignSettings, props));
    }
  }];
});
