// @webiny/app-page-builder/editor/plugins/elements/pagesList/PagesListDesignSettings.js
import * as React from "react";
import { Grid, Cell } from "@webiny/ui/Grid";
import { Input } from "@webiny/ui/Input";
import { Select } from "@webiny/ui/Select";
import PagesList from "./PagesList";
import { getPlugins } from "@webiny/plugins";
import { validation } from "@webiny/validation";

var PagesListDesignSettings = function PagesListDesignSettings(_ref) {
  var Bind = _ref.Bind,
      data = _ref.data;
  var components = getPlugins("pb-page-element-pages-list-component");
  return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(Grid, null, /*#__PURE__*/React.createElement(Cell, {
    span: 6
  }, /*#__PURE__*/React.createElement(Bind, {
    name: "component",
    defaultValue: components[0] ? components[0].componentName : null
  }, /*#__PURE__*/React.createElement(Select, {
    label: "Design",
    description: "Select a component to render the list"
  }, components.map(function (cmp) {
    return /*#__PURE__*/React.createElement("option", {
      key: cmp.name,
      value: cmp.componentName
    }, cmp.title);
  })))), /*#__PURE__*/React.createElement(Cell, {
    span: 6
  }, /*#__PURE__*/React.createElement(Bind, {
    name: "resultsPerPage",
    validators: validation.create("numeric")
  }, /*#__PURE__*/React.createElement(Input, {
    label: "Results per page"
  })))), /*#__PURE__*/React.createElement(Grid, null, /*#__PURE__*/React.createElement(Cell, {
    span: 12,
    style: {
      overflowY: "scroll"
    }
  }, /*#__PURE__*/React.createElement(PagesList, {
    data: data
  }))));
};

export default PagesListDesignSettings;
