// @webiny/app-headless-cms/admin/plugins/fieldRenderers/richText/richTextInputs.js
import _taggedTemplateLiteral from "@babel/runtime/helpers/taggedTemplateLiteral";

function _templateObject2() {
  var data = _taggedTemplateLiteral(["Renders a simple list of rich text editors."]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteral(["Rich Text Inputs"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

import React from "react";
import { I18NValue } from "@webiny/app-i18n/components";
import { i18n } from "@webiny/app/i18n";
import I18NRichTextEditor from "@webiny/app-i18n/admin/components/I18NRichTextEditor";
import { ReactComponent as DeleteIcon } from "@svgr/webpack!@webiny/app-headless-cms/admin/icons/close.svg";
import get from "lodash/get";
import DynamicListMultipleValues from "@webiny/app-headless-cms/admin/plugins/fieldRenderers/DynamicListMultipleValues";
var t = i18n.ns("app-headless-cms/admin/fields/rich-text");
var plugin = {
  type: "cms-editor-field-renderer",
  name: "cms-editor-field-renderer-rich-text-inputs",
  renderer: {
    rendererName: "rich-text-inputs",
    name: t(_templateObject()),
    description: t(_templateObject2()),
    canUse: function canUse(_ref) {
      var field = _ref.field;
      return field.type === "rich-text" && field.multipleValues && !get(field, "predefinedValues.enabled");
    },
    render: function render(props) {
      var field = props.field;
      return /*#__PURE__*/React.createElement(DynamicListMultipleValues, props, function (_ref2) {
        var bind = _ref2.bind,
            index = _ref2.index;
        return /*#__PURE__*/React.createElement(I18NRichTextEditor, Object.assign({}, bind.index, {
          label: I18NValue({
            value: "Value ".concat(index + 1)
          }),
          placeholder: I18NValue({
            value: field.placeholderText
          }),
          description: I18NValue({
            value: field.helpText
          }),
          trailingIcon: index > 0 && {
            icon: /*#__PURE__*/React.createElement(DeleteIcon, null),
            onClick: bind.index.removeValue
          }
        }));
      });
    }
  }
};
export default plugin;
