// @webiny/app-google-tag-manager/admin/index.js
import _taggedTemplateLiteral from "@babel/runtime/helpers/taggedTemplateLiteral";

function _templateObject() {
  var data = _taggedTemplateLiteral(["Google Tag Manager"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

import * as React from "react";
import { Route } from "@webiny/react-router";
import { Helmet } from "react-helmet";
import { AdminLayout } from "@webiny/app-admin/components/AdminLayout";
import GoogleTagManagerSettings from "./components/GoogleTagManagerSettings";
import { SecureRoute } from "@webiny/app-security/components";
import { i18n } from "@webiny/app/i18n";
var t = i18n.ns("app-google-tag-manager/admin");
var ROLE_PB_SETTINGS = ["pb:page:crud"];
export default (function () {
  return [{
    type: "route",
    name: "route-settings-page-builder-google-tag-manager",
    route: /*#__PURE__*/React.createElement(Route, {
      path: "/settings/page-builder/google-tag-manager",
      render: function render() {
        return /*#__PURE__*/React.createElement(AdminLayout, null, /*#__PURE__*/React.createElement(Helmet, {
          title: "Page Builder - Google Tag Manager Settings"
        }), /*#__PURE__*/React.createElement(SecureRoute, {
          scopes: ROLE_PB_SETTINGS
        }, /*#__PURE__*/React.createElement(GoogleTagManagerSettings, null)));
      }
    })
  }, {
    type: "menu-settings-page-builder",
    name: "menu-settings-page-builder-google-tag-manager",
    render: function render(_ref) {
      var Item = _ref.Item;
      return /*#__PURE__*/React.createElement(Item, {
        label: t(_templateObject()),
        path: "/settings/page-builder/google-tag-manager"
      });
    }
  }];
});
