// @webiny/app-page-builder/render/plugins/elements/block/Block.js
import _toConsumableArray from "@babel/runtime/helpers/toConsumableArray";
import _objectWithoutProperties from "@babel/runtime/helpers/objectWithoutProperties";
import React from "react";
import { css } from "emotion";
import Element from "@webiny/app-page-builder/render/components/Element";
import { ElementRoot } from "@webiny/app-page-builder/render/components/ElementRoot";
import ElementAnimation from "@webiny/app-page-builder/render/components/ElementAnimation";

var Block = function Block(_ref) {
  var element = _ref.element;
  return /*#__PURE__*/React.createElement(ElementAnimation, null, /*#__PURE__*/React.createElement(ElementRoot, {
    element: element
  }, function (_ref2) {
    var elementStyle = _ref2.elementStyle,
        elementAttributes = _ref2.elementAttributes,
        customClasses = _ref2.customClasses,
        combineClassNames = _ref2.combineClassNames;

    var width = elementStyle.width,
        alignItems = elementStyle.alignItems,
        justifyContent = elementStyle.justifyContent,
        containerStyle = _objectWithoutProperties(elementStyle, ["width", "alignItems", "justifyContent"]);

    return /*#__PURE__*/React.createElement("div", Object.assign({
      style: {
        width: "100%",
        display: "flex",
        justifyContent: "center"
      },
      className: "webiny-pb-layout-block-container " + /*#__PURE__*/css(containerStyle, "label:Block;")
    }, elementAttributes), /*#__PURE__*/React.createElement("div", {
      style: {
        width: width ? width : "100%",
        alignSelf: justifyContent,
        alignItems: alignItems
      },
      className: combineClassNames.apply(void 0, ["webiny-pb-layout-block webiny-pb-base-page-element-style"].concat(_toConsumableArray(customClasses)))
    }, element.elements.map(function (element) {
      return /*#__PURE__*/React.createElement(Element, {
        key: element.id,
        element: element
      });
    })));
  }));
};

export default Block;
