// @webiny/app-page-builder/editor/plugins/toolbar/addElement/AddElement.js
import _objectSpread from "@babel/runtime/helpers/objectSpread2";
import _slicedToArray from "@babel/runtime/helpers/slicedToArray";
import React, { useCallback, useState } from "react";
import { connect } from "@webiny/app-page-builder/editor/redux";
import Draggable from "@webiny/app-page-builder/editor/components/Draggable";
import { dragStart, dragEnd, deactivatePlugin, dropElement } from "@webiny/app-page-builder/editor/actions";
import { getPlugins } from "@webiny/plugins";
import { getActivePluginParams } from "@webiny/app-page-builder/editor/selectors";
import { usePageBuilder } from "@webiny/app-page-builder/hooks/usePageBuilder";
import * as Styled from "./StyledComponents";
import { css } from "emotion";
import { List, ListItem, ListItemMeta } from "@webiny/ui/List";
import { Icon } from "@webiny/ui/Icon";
import { Typography } from "@webiny/ui/Typography";
import { ButtonFloating } from "@webiny/ui/Button";
import { ReactComponent as AddIcon } from "@svgr/webpack!@webiny/app-page-builder/editor/assets/icons/add.svg";
var ADD_ELEMENT = "pb-editor-toolbar-add-element"; // @ts-ignore

var categoriesList = /*#__PURE__*/css({
  backgroundColor: "var(--mdc-theme-surface)",
  boxShadow: "inset 1px 0px 5px 0px var(--mdc-theme-background)",
  borderTop: "1px solid var(--mdc-theme-background)",
  ".mdc-list-item": {
    width: 150,
    fontWeight: "600 !important",
    borderBottom: "1px solid var(--mdc-theme-background)",
    "&.active": {
      backgroundColor: "var(--mdc-theme-background)",
      color: "var(--mdc-theme-primary)",
      ".mdc-list-item__meta": {
        color: "var(--mdc-theme-primary)"
      }
    }
  }
}, "label:categoriesList;");

var AddElement = function AddElement(_ref) {
  var params = _ref.params,
      dropElement = _ref.dropElement,
      dragStart = _ref.dragStart,
      deactivatePlugin = _ref.deactivatePlugin,
      dragEnd = _ref.dragEnd;
  var getGroups = useCallback(function () {
    return getPlugins("pb-editor-page-element-group");
  }, []);
  var getGroupElements = useCallback(function (group) {
    return getPlugins("pb-editor-page-element").filter(function (el) {
      return el.toolbar && el.toolbar.group === group;
    });
  }, []);

  var _useState = useState(getGroups()[0].name),
      _useState2 = _slicedToArray(_useState, 2),
      group = _useState2[0],
      setGroup = _useState2[1];

  var _usePageBuilder = usePageBuilder(),
      theme = _usePageBuilder.theme;

  var refresh = useCallback(function () {
    setGroup(group);
  }, []);
  var enableDragOverlay = useCallback(function () {
    var el = document.querySelector(".pb-editor");

    if (el) {
      el.classList.add("pb-editor-dragging");
    }
  }, []);
  var disableDragOverlay = useCallback(function () {
    var el = document.querySelector(".pb-editor");

    if (el) {
      el.classList.remove("pb-editor-dragging");
    }
  }, []);
  var renderDraggable = useCallback(function (element, plugin) {
    var elementType = plugin.elementType;
    return /*#__PURE__*/React.createElement(Draggable, {
      key: plugin.name,
      target: plugin.target,
      beginDrag: function beginDrag(props) {
        dragStart({
          element: {
            type: elementType
          }
        });
        setTimeout(function () {
          return deactivatePlugin({
            name: ADD_ELEMENT
          });
        }, 20);
        return {
          type: elementType,
          target: props.target
        };
      },
      endDrag: function endDrag(props, monitor) {
        dragEnd({
          element: monitor.getItem()
        });
      }
    }, function (_ref2) {
      var drag = _ref2.drag;
      return /*#__PURE__*/React.createElement("div", {
        ref: drag
      }, renderOverlay(element, null, "Drag to Add", plugin));
    });
  }, [dragStart, deactivatePlugin, dragEnd]);
  var renderOverlay = useCallback(function (element) {
    var onClick = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : null;
    var label = arguments.length > 2 ? arguments[2] : undefined;
    var plugin = arguments.length > 3 ? arguments[3] : undefined;
    return /*#__PURE__*/React.createElement(Styled.ElementPreview, null, /*#__PURE__*/React.createElement(Styled.Overlay, null, /*#__PURE__*/React.createElement(Styled.Backdrop, {
      className: "backdrop"
    }), /*#__PURE__*/React.createElement(Styled.AddBlock, {
      className: "add-block"
    }, /*#__PURE__*/React.createElement(ButtonFloating, {
      "data-testid": "pb-editor-add-element-button-".concat(plugin.elementType),
      onClick: onClick,
      label: label,
      icon: /*#__PURE__*/React.createElement(AddIcon, null),
      onMouseDown: enableDragOverlay,
      onMouseUp: disableDragOverlay
    }))), element);
  }, [enableDragOverlay, disableDragOverlay]);
  var renderClickable = useCallback(function (element, plugin) {
    var item = renderOverlay(element, function () {
      dropElement({
        source: {
          type: plugin.elementType
        },
        target: _objectSpread({}, params)
      });
      deactivatePlugin({
        name: ADD_ELEMENT
      });
    }, "Click to Add", plugin);
    return /*#__PURE__*/React.cloneElement(item, {
      key: plugin.name
    });
  }, [params, deactivatePlugin, dropElement, renderOverlay]);
  return /*#__PURE__*/React.createElement(Styled.Flex, null, /*#__PURE__*/React.createElement(List, {
    className: categoriesList
  }, getGroups().map(function (plugin) {
    return /*#__PURE__*/React.createElement(ListItem, {
      onClick: function onClick() {
        return setGroup(plugin.name);
      },
      key: plugin.name,
      className: plugin.name === group && "active"
    }, plugin.group.title, plugin.group.icon && /*#__PURE__*/React.createElement(ListItemMeta, null, /*#__PURE__*/React.createElement(Icon, {
      icon: plugin.group.icon
    })));
  })), /*#__PURE__*/React.createElement(Styled.Elements, null, group && getGroupElements(group).map(function (plugin) {
    return (params ? renderClickable : renderDraggable)( /*#__PURE__*/React.createElement("div", {
      "data-role": "draggable"
    }, /*#__PURE__*/React.createElement(Styled.ElementBox, null, /*#__PURE__*/React.createElement(Styled.ElementTitle, null, typeof plugin.toolbar.title === "function" ? plugin.toolbar.title({
      refresh: refresh
    }) : /*#__PURE__*/React.createElement(Typography, {
      use: "overline"
    }, plugin.toolbar.title)), /*#__PURE__*/React.createElement(Styled.ElementPreviewCanvas, null, plugin.toolbar.preview({
      theme: theme
    })))), plugin);
  })));
};

export default connect(function (state) {
  var getParams = getActivePluginParams("pb-editor-toolbar-add-element");
  return {
    params: getParams ? getParams(state) : null
  };
}, {
  dragStart: dragStart,
  dragEnd: dragEnd,
  deactivatePlugin: deactivatePlugin,
  dropElement: dropElement
})(AddElement);
