// @webiny/app-i18n/admin/plugins/richTextEditor/bold/index.js
import * as React from "react";
import { css } from "emotion";
import { isKeyHotkey } from "is-hotkey";
import { ReactComponent as FormatBoldIcon } from "@svgr/webpack!@webiny/app-i18n/admin/assets/icons/format_bold.svg";
var isBoldHotkey = isKeyHotkey("mod+b");
var mark = "bold";
/* prettier-ignore */

var strongStyle = /*#__PURE__*/css("font-weight:bold !important;label:strongStyle;");
var plugin = {
  name: "i18n-input-rich-text-editor-bold",
  type: "i18n-input-rich-text-editor",
  plugin: {
    name: "bold",
    menu: {
      render: function render(_ref) {
        var MenuButton = _ref.MenuButton,
            editor = _ref.editor;
        return /*#__PURE__*/React.createElement(MenuButton, {
          onClick: function onClick() {
            return editor.toggleMark(mark);
          },
          active: editor.hasMark(mark)
        }, /*#__PURE__*/React.createElement(FormatBoldIcon, null));
      }
    },
    editor: {
      onKeyDown: function onKeyDown(_ref2, next) {
        var event = _ref2.event,
            editor = _ref2.editor;

        // Decide what to do based on the key code...
        if (isBoldHotkey(event)) {
          event.preventDefault();
          editor.toggleMark(mark);
          return true;
        }

        return next();
      },
      renderLeaf: function renderLeaf(_ref3) {
        var leaf = _ref3.leaf,
            attributes = _ref3.attributes,
            children = _ref3.children;

        if (leaf[mark] === true) {
          return /*#__PURE__*/React.createElement("strong", Object.assign({
            className: strongStyle
          }, attributes), children);
        }

        return children;
      }
    }
  }
};
export default plugin;
