// @webiny/app-form-builder/admin/plugins/fields/contact/email.js
import _objectSpread from "@babel/runtime/helpers/objectSpread2";
import textFieldPlugin from "./../text";
var plugin = {
  type: "form-editor-field-type",
  name: "form-editor-field-type-email",
  field: _objectSpread(_objectSpread({}, textFieldPlugin.field), {}, {
    unique: true,
    group: "form-editor-field-group-contact",
    name: "email",
    label: "Email   ",
    description: "Email address",
    // TODO: validators: validation.create("required"), // Editable validators.
    createField: function createField(props) {
      var i18n = props.i18n;
      return _objectSpread(_objectSpread({}, textFieldPlugin.field.createField(props)), {}, {
        name: this.name,
        fieldId: "email",
        label: {
          values: [{
            locale: i18n.getDefaultLocale().id,
            value: "Email"
          }]
        },
        validation: [{
          name: "pattern",
          message: {
            values: [{
              locale: i18n.getDefaultLocale().id,
              value: "Please enter a valid e-mail."
            }]
          },
          settings: {
            preset: "email",
            regex: null,
            flags: null
          }
        }]
      });
    }
  })
};
export default plugin;
