// @webiny/app-form-builder/admin/plugins/fields/contact/website.js
import _objectSpread from "@babel/runtime/helpers/objectSpread2";
import textFieldPlugin from "./../text";
var plugin = {
  type: "form-editor-field-type",
  name: "form-editor-field-type-website",
  field: _objectSpread(_objectSpread({}, textFieldPlugin.field), {}, {
    unique: true,
    group: "form-editor-field-group-contact",
    name: "website",
    label: "Website",
    description: "Link to a website",
    createField: function createField(props) {
      var i18n = props.i18n;
      return _objectSpread(_objectSpread({}, textFieldPlugin.field.createField(props)), {}, {
        name: this.name,
        fieldId: "website",
        label: {
          values: [{
            locale: i18n.getDefaultLocale().id,
            value: "Website"
          }]
        },
        validation: [{
          name: "pattern",
          message: {
            values: [{
              locale: i18n.getDefaultLocale().id,
              value: "Please enter a valid URL."
            }]
          },
          settings: {
            preset: "url",
            regex: null,
            flags: null
          }
        }]
      });
    }
  })
};
export default plugin;
