// @webiny/app-page-builder/editor/plugins/pageSettings/index.js
import * as React from "react";
import GeneralSettings from "./components/GeneralSettings";
import SeoSettings from "./components/SeoSettings";
import SocialSettings from "./components/SocialSettings";
import { ReactComponent as SettingsIcon } from "@svgr/webpack!./icons/round-settings-24px.svg";
import { ReactComponent as SocialIcon } from "@svgr/webpack!./icons/round-thumb_up-24px.svg";
import { ReactComponent as SeoIcon } from "@svgr/webpack!./icons/round-search-24px.svg";
var plugins = [{
  name: "pb-editor-page-settings-general",
  type: "pb-editor-page-settings",
  title: "General settings",
  description: "Manage things like title, page status,url and more.",
  icon: /*#__PURE__*/React.createElement(SettingsIcon, null),
  fields: "\n            general {\n                image {\n                    id\n                    src\n                }\n                tags\n                layout\n            }\n    ",
  render: function render(props) {
    return /*#__PURE__*/React.createElement(GeneralSettings, props);
  }
}, {
  name: "pb-editor-page-settings-seo",
  type: "pb-editor-page-settings",
  title: "SEO",
  description: "Control SEO settings like description and keywords.",
  icon: /*#__PURE__*/React.createElement(SeoIcon, null),
  fields: "\n            seo {\n                title\n                description\n                meta {\n                    name\n                    content\n                }\n            }\n    ",
  render: function render(props) {
    return /*#__PURE__*/React.createElement(SeoSettings, props);
  }
}, {
  name: "pb-editor-page-settings-social",
  type: "pb-editor-page-settings",
  title: "Social media",
  description: "Set share images and settings for social media sites.",
  icon: /*#__PURE__*/React.createElement(SocialIcon, null),
  fields: "\n            social {\n                image {\n                    id\n                    src\n                }\n                title\n                description\n                meta {\n                    property\n                    content\n                }\n            }\n    ",
  render: function render(props) {
    return /*#__PURE__*/React.createElement(SocialSettings, props);
  }
}];
export default plugins;
