// @webiny/app-page-builder/admin/views/Categories/CategoriesForm.js
import _taggedTemplateLiteral from "@babel/runtime/helpers/taggedTemplateLiteral";

function _templateObject6() {
  var data = _taggedTemplateLiteral(["Save category"]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = _taggedTemplateLiteral(["Layout"]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _taggedTemplateLiteral(["URL"]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _taggedTemplateLiteral(["Slug"]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteral(["Name"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteral(["New Category"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

import * as React from "react";
import { i18n } from "@webiny/app/i18n";
import { getPlugins } from "@webiny/plugins";
import { Form } from "@webiny/form";
import { Grid, Cell } from "@webiny/ui/Grid";
import { Input } from "@webiny/ui/Input";
import { Select } from "@webiny/ui/Select";
import { ButtonPrimary } from "@webiny/ui/Button";
import { SimpleForm, SimpleFormHeader, SimpleFormFooter, SimpleFormContent } from "@webiny/app-admin/components/SimpleForm";
import { categoryUrlValidator } from "./validators";
import { CircularProgress } from "@webiny/ui/Progress";
import { validation } from "@webiny/validation";
import { useCrud } from "@webiny/app-admin/hooks/useCrud";
var t = i18n.ns("app-page-builder/admin/categories/form");

var CategoriesForm = function CategoriesForm() {
  var _useCrud = useCrud(),
      crudForm = _useCrud.form;

  var layouts = React.useMemo(function () {
    return getPlugins("pb-page-layout").map(function (pl) {
      return pl.layout;
    });
  }, []);
  return /*#__PURE__*/React.createElement(Form, crudForm, function (_ref) {
    var data = _ref.data,
        form = _ref.form,
        Bind = _ref.Bind;
    return /*#__PURE__*/React.createElement(SimpleForm, null, crudForm.loading && /*#__PURE__*/React.createElement(CircularProgress, null), /*#__PURE__*/React.createElement(SimpleFormHeader, {
      title: data.name || t(_templateObject())
    }), /*#__PURE__*/React.createElement(SimpleFormContent, null, /*#__PURE__*/React.createElement(Grid, null, /*#__PURE__*/React.createElement(Cell, {
      span: 6
    }, /*#__PURE__*/React.createElement(Bind, {
      name: "name",
      validators: validation.create("required")
    }, /*#__PURE__*/React.createElement(Input, {
      label: t(_templateObject2())
    }))), /*#__PURE__*/React.createElement(Cell, {
      span: 6
    }, /*#__PURE__*/React.createElement(Bind, {
      name: "slug",
      validators: validation.create("required")
    }, /*#__PURE__*/React.createElement(Input, {
      disabled: data.id,
      label: t(_templateObject3())
    }))), /*#__PURE__*/React.createElement(Cell, {
      span: 12
    }, /*#__PURE__*/React.createElement(Bind, {
      name: "url",
      validators: [validation.create("required"), categoryUrlValidator]
    }, /*#__PURE__*/React.createElement(Input, {
      disabled: data.id,
      label: t(_templateObject4())
    }))), /*#__PURE__*/React.createElement(Cell, {
      span: 6
    }, /*#__PURE__*/React.createElement(Bind, {
      name: "layout",
      defaultValue: layouts[0].name
    }, /*#__PURE__*/React.createElement(Select, {
      label: t(_templateObject5())
    }, layouts.map(function (_ref2) {
      var name = _ref2.name,
          title = _ref2.title;
      return /*#__PURE__*/React.createElement("option", {
        key: name,
        value: name
      }, title);
    })))))), /*#__PURE__*/React.createElement(SimpleFormFooter, null, /*#__PURE__*/React.createElement(ButtonPrimary, {
      onClick: form.submit
    }, t(_templateObject6()))));
  });
};

export default CategoriesForm;
