// @webiny/app-page-builder/admin/views/Categories/graphql.js
import _taggedTemplateLiteral from "@babel/runtime/helpers/taggedTemplateLiteral";

function _templateObject6() {
  var data = _taggedTemplateLiteral(["\n    mutation PbDeleteCategory($id: ID!) {\n        pageBuilder {\n            deleteCategory(id: $id) {\n                data\n                error {\n                    code\n                    message\n                }\n            }\n        }\n    }\n"]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = _taggedTemplateLiteral(["\n    mutation PbUpdateCategory($id: ID!, $data: PbCategoryInput!){\n        pageBuilder {\n            category: updateCategory(id: $id, data: $data) {\n                data {\n                    ", "\n                }\n                error {\n                    code\n                    message\n                    data\n                }\n            }\n        }\n    }\n"]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _taggedTemplateLiteral(["\n    mutation PbCreateCategory($data: PbCategoryInput!){\n        pageBuilder {\n            category: createCategory(data: $data) {\n                data {\n                    ", "\n                }\n                error {\n                    code\n                    message\n                    data\n                }\n            }\n        }\n    }\n"]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _taggedTemplateLiteral(["\n    query PbLoadCategory($id: ID!) {\n        pageBuilder {\n            category: getCategory(id: $id){\n                data {\n                    ", "\n                }\n                error {\n                    code\n                    message\n                }\n            }\n        }\n    }\n"]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteral(["\n    query PbListCategoriesByName {\n        pageBuilder {\n            categories: listCategories(sort: { name: 1 }, limit: 100) {\n                data {\n                    id\n                    name\n                    slug\n                    url\n                }\n            }\n        }\n    }\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteral(["\n    query PbLoadCategories(\n        $where: JSON\n        $sort: JSON\n        $search: PbSearchInput\n        $limit: Int\n        $after: String\n        $before: String\n    ) {\n        pageBuilder {\n            categories: listCategories(\n                where: $where\n                sort: $sort\n                search: $search\n                limit: $limit\n                after: $after\n                before: $before\n            ) {\n                data {\n                    id\n                    name\n                    slug\n                    url\n                    createdOn\n                }\n                meta {\n                    cursors {\n                        next\n                        previous\n                    }\n                    hasNextPage\n                    hasPreviousPage\n                    totalCount\n                }\n            }\n        }\n    }\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

import gql from "graphql-tag";
var fields = "\n    id\n    name\n    slug\n    url\n    layout\n";
export var LIST_CATEGORIES = gql(_templateObject());
export var LIST_CATEGORIES_BY_NAME = gql(_templateObject2());
export var READ_CATEGORY = gql(_templateObject3(), fields);
export var CREATE_CATEGORY = gql(_templateObject4(), fields);
export var UPDATE_CATEGORY = gql(_templateObject5(), fields);
export var DELETE_CATEGORY = gql(_templateObject6());
