// @webiny/app-form-builder/components/Form/FormLoad.js
import _get from "lodash/get";
import React from "react";
import { GET_PUBLISHED_FORM } from "./graphql";
import { Query } from "react-apollo";
import FormRender from "./FormRender";

var FormLoad = function FormLoad(props) {
  var variables = {
    slug: null,
    version: null,
    parent: null,
    id: null
  };

  if (props.slug) {
    variables.slug = props.slug;

    if (props.version) {
      variables.version = props.version;
    }
  } else if (props.parentId) {
    variables.parent = props.parentId;
  } else {
    variables.id = props.revisionId;
  }

  return /*#__PURE__*/React.createElement(Query, {
    query: GET_PUBLISHED_FORM,
    variables: variables
  }, function (_ref) {
    var data = _ref.data,
        loading = _ref.loading;

    if (loading) {
      // TODO: handle loading
      return null;
    }

    var formData = _get(data, "forms.getPublishedForm.data");

    if (!formData) {
      // TODO: handle cannot load form
      return /*#__PURE__*/React.createElement("span", null, "Form not found.");
    }

    return /*#__PURE__*/React.createElement(FormRender, Object.assign({}, props, {
      data: formData
    }));
  });
};

export default FormLoad;
