// @webiny/app-page-builder/admin/views/Menus/MenusForm.js
import _taggedTemplateLiteral from "@babel/runtime/helpers/taggedTemplateLiteral";

function _templateObject4() {
  var data = _taggedTemplateLiteral(["Save menu"]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _taggedTemplateLiteral(["Description"]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteral(["Slug"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteral(["Name"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

import React from "react";
import { Form } from "@webiny/form";
import { Grid, Cell } from "@webiny/ui/Grid";
import { Input } from "@webiny/ui/Input";
import { ButtonPrimary } from "@webiny/ui/Button";
import MenuItems from "./MenusForm/MenuItems";
import { CircularProgress } from "@webiny/ui/Progress";
import { useCrud } from "@webiny/app-admin/hooks/useCrud";
import { i18n } from "@webiny/app/i18n";
import { validation } from "@webiny/validation";
import { SimpleForm, SimpleFormFooter, SimpleFormContent } from "@webiny/app-admin/components/SimpleForm";
var t = i18n.ns("app-page-builder/admin/menus/form");

function MenusForm() {
  var _useCrud = useCrud(),
      crudForm = _useCrud.form;

  return /*#__PURE__*/React.createElement(Form, Object.assign({}, crudForm, {
    data: crudForm.id ? crudForm.data : {
      items: []
    }
  }), function (_ref) {
    var data = _ref.data,
        form = _ref.form,
        Bind = _ref.Bind;
    return /*#__PURE__*/React.createElement(SimpleForm, {
      "data-testid": "pb-menus-form"
    }, crudForm.loading && /*#__PURE__*/React.createElement(CircularProgress, null), /*#__PURE__*/React.createElement(SimpleFormContent, null, /*#__PURE__*/React.createElement(Grid, null, /*#__PURE__*/React.createElement(Cell, {
      span: 6
    }, /*#__PURE__*/React.createElement(Bind, {
      name: "title",
      validators: validation.create("required")
    }, /*#__PURE__*/React.createElement(Input, {
      label: t(_templateObject())
    }))), /*#__PURE__*/React.createElement(Cell, {
      span: 6
    }, /*#__PURE__*/React.createElement(Bind, {
      name: "slug",
      validators: validation.create("required")
    }, /*#__PURE__*/React.createElement(Input, {
      disabled: data.id,
      label: t(_templateObject2())
    }))), /*#__PURE__*/React.createElement(Cell, {
      span: 12
    }, /*#__PURE__*/React.createElement(Bind, {
      name: "description"
    }, /*#__PURE__*/React.createElement(Input, {
      rows: 5,
      label: t(_templateObject3())
    })))), /*#__PURE__*/React.createElement(Bind, {
      name: "items"
    }, function (props) {
      return /*#__PURE__*/React.createElement(MenuItems, Object.assign({
        menuForm: form
      }, props));
    })), /*#__PURE__*/React.createElement(SimpleFormFooter, null, /*#__PURE__*/React.createElement(ButtonPrimary, {
      onClick: form.submit
    }, t(_templateObject4()))));
  });
}

export default MenusForm;
