// @webiny/app-page-builder/admin/views/Menus/Menus.js
import React from "react";
import { SplitView, LeftPanel, RightPanel } from "@webiny/app-admin/components/SplitView";
import { FloatingActionButton } from "@webiny/app-admin/components/FloatingActionButton";
import MenusDataList from "./MenusDataList";
import MenusForm from "./MenusForm";
import { CrudProvider } from "@webiny/app-admin/contexts/Crud";
import { READ_MENU, LIST_MENUS, CREATE_MENU, UPDATE_MENU, DELETE_MENU } from "./graphql";

function Menus() {
  return /*#__PURE__*/React.createElement(CrudProvider, {
    delete: DELETE_MENU,
    read: READ_MENU,
    create: CREATE_MENU,
    update: UPDATE_MENU,
    list: {
      query: LIST_MENUS,
      variables: {
        sort: {
          savedOn: -1
        }
      }
    }
  }, function (_ref) {
    var actions = _ref.actions;
    return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(SplitView, null, /*#__PURE__*/React.createElement(LeftPanel, {
      span: 3
    }, /*#__PURE__*/React.createElement(MenusDataList, null)), /*#__PURE__*/React.createElement(RightPanel, {
      span: 9
    }, /*#__PURE__*/React.createElement(MenusForm, null))), /*#__PURE__*/React.createElement(FloatingActionButton, {
      "data-testid": "new-record-button",
      onClick: actions.resetForm
    }));
  });
}

export default Menus;
