// @webiny/app-page-builder/editor/plugins/elements/button/index.js
import _objectSpread from "@babel/runtime/helpers/objectSpread2";
import React from "react";
import { css } from "emotion";
import { createValue } from "@webiny/app-page-builder/editor/components/Slate";
import { ReactComponent as ButtonIcon } from "@svgr/webpack!./round-toggle_on-24px.svg";
import ButtonSettings from "./ButtonSettings";
import Button from "./Button";
import Action from "../../elementSettings/components/Action";
var buttonWrapper = /*#__PURE__*/css({
  display: "flex",
  justifyContent: "center"
}, "label:buttonWrapper;");
export default (function () {
  return [{
    name: "pb-editor-page-element-button",
    type: "pb-editor-page-element",
    elementType: "button",
    toolbar: {
      title: "Button",
      group: "pb-editor-element-group-basic",
      preview: function preview() {
        return /*#__PURE__*/React.createElement("div", {
          className: buttonWrapper
        }, /*#__PURE__*/React.createElement("button", {
          className: "webiny-pb-page-element-button"
        }, "Click me"));
      }
    },
    settings: ["pb-editor-page-element-settings-button", "pb-editor-page-element-settings-link", "", "pb-editor-page-element-settings-horizontal-align-flex", "", "pb-editor-page-element-settings-clone", "pb-editor-page-element-settings-delete", ""],
    target: ["column", "row"],
    create: function create(options) {
      return _objectSpread({
        type: "button",
        elements: [],
        data: {
          text: createValue("Click me", "button"),
          settings: {
            margin: {
              desktop: {
                all: 0
              },
              mobile: {
                all: 0
              }
            },
            padding: {
              desktop: {
                all: 0
              },
              mobile: {
                all: 0
              }
            }
          }
        }
      }, options);
    },
    render: function render(_ref) {
      var element = _ref.element;
      return /*#__PURE__*/React.createElement(Button, {
        element: element
      });
    }
  }, {
    name: "pb-editor-page-element-settings-button",
    type: "pb-editor-page-element-settings",
    renderAction: function renderAction() {
      return /*#__PURE__*/React.createElement(Action, {
        plugin: this.name,
        tooltip: "Button",
        icon: /*#__PURE__*/React.createElement(ButtonIcon, null)
      });
    },
    renderMenu: function renderMenu() {
      return /*#__PURE__*/React.createElement(ButtonSettings, null);
    }
  }];
});
