// @webiny/app-page-builder/editor/plugins/elements/row/ResizeHandle.js
import _objectWithoutProperties from "@babel/runtime/helpers/objectWithoutProperties";
import React from "react";
import styled from "@emotion/styled";
var Handle = /*#__PURE__*/React.memo( /*#__PURE__*/styled("div", {
  target: "ey1sh8q0",
  label: "Handle"
})(function (_ref) {
  var isResizing = _ref.isResizing;
  return {
    position: "absolute",
    left: -5,
    top: "10%",
    height: "80%",
    minHeight: 25,
    translate: "width 0.5s",
    zIndex: 30,
    width: 10,
    borderRadius: 10,
    cursor: "col-resize",
    backgroundColor: isResizing ? "var(--mdc-theme-primary)" : "var(--mdc-theme-secondary)",
    display: "none",
    transition: "background-color 0.5s, opacity 0.2s",
    opacity: 0.2,
    "&:hover": {
      opacity: 1
    }
  };
}));
var RatioContainer = /*#__PURE__*/React.memo( /*#__PURE__*/styled("div", {
  target: "ey1sh8q1",
  label: "RatioContainer"
})({
  position: "absolute",
  top: -12,
  transform: "translateX(-50%)",
  color: "#fff",
  fontSize: 12,
  backgroundColor: "var(--mdc-theme-secondary)",
  padding: 5,
  borderRadius: 5,
  zIndex: 10
}));
var Ratio = /*#__PURE__*/React.memo(function (_ref2) {
  var left = _ref2.left,
      right = _ref2.right;
  return /*#__PURE__*/React.createElement(RatioContainer, null, left.toFixed(1), /*#__PURE__*/React.createElement("small", null, "%"), " / ", right.toFixed(1), /*#__PURE__*/React.createElement("small", null, "%"));
});
Ratio.displayName = "Ratio";

var ResizeHandle = function ResizeHandle(_ref3) {
  var leftWidth = _ref3.leftWidth,
      rightWidth = _ref3.rightWidth,
      props = _objectWithoutProperties(_ref3, ["leftWidth", "rightWidth"]);

  return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(Handle, Object.assign({
    className: "resize-handle"
  }, props)), props.isResizing && /*#__PURE__*/React.createElement(Ratio, {
    left: leftWidth,
    right: rightWidth
  }));
};

export default /*#__PURE__*/React.memo(ResizeHandle);
