// @webiny/app-page-builder/editor/plugins/elements/row/Row.js
import React from "react";
import ConnectedElement from "@webiny/app-page-builder/editor/components/ConnectedElement";
import { ElementRoot } from "@webiny/app-page-builder/render/components/ElementRoot";
import RowContainer from "./RowContainer";
import ElementAnimation from "@webiny/app-page-builder/render/components/ElementAnimation";
/**
 * TODO: this entire component can be further optimized (see the Chrome Profiler) to avoid unnecessary elements re-renders.
 */

var Row = function Row(_ref) {
  var element = _ref.element;
  return /*#__PURE__*/React.createElement(ElementAnimation, null, /*#__PURE__*/React.createElement(ElementRoot, {
    element: element,
    className: "webiny-pb-base-page-element-style webiny-pb-layout-row",
    style: {
      zIndex: 20,
      position: "relative"
    }
  }, /*#__PURE__*/React.createElement(ConnectedElement, {
    elementId: element.id,
    withChildElements: true
  }, function (element) {
    return /*#__PURE__*/React.createElement(RowContainer, {
      elementId: element.id
    });
  })));
};

export default Row;
