// @webiny/app-admin/components/FileManager/FileDetails.js
import _regeneratorRuntime from "@babel/runtime/regenerator";
import _asyncToGenerator from "@babel/runtime/helpers/asyncToGenerator";
import _taggedTemplateLiteral from "@babel/runtime/helpers/taggedTemplateLiteral";
import _slicedToArray from "@babel/runtime/helpers/slicedToArray";

function _templateObject10() {
  var data = _taggedTemplateLiteral(["Created:"]);

  _templateObject10 = function _templateObject10() {
    return data;
  };

  return data;
}

function _templateObject9() {
  var data = _taggedTemplateLiteral(["Tags:"]);

  _templateObject9 = function _templateObject9() {
    return data;
  };

  return data;
}

function _templateObject8() {
  var data = _taggedTemplateLiteral(["Type:"]);

  _templateObject8 = function _templateObject8() {
    return data;
  };

  return data;
}

function _templateObject7() {
  var data = _taggedTemplateLiteral(["Size:"]);

  _templateObject7 = function _templateObject7() {
    return data;
  };

  return data;
}

function _templateObject6() {
  var data = _taggedTemplateLiteral(["Name:"]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = _taggedTemplateLiteral(["File deleted successfully."]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _taggedTemplateLiteral(["Delete image"]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _taggedTemplateLiteral(["Download file"]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteral(["You're about to delete file {name}. Are you sure you want to continue?"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteral(["Delete file"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

/* eslint-disable */
import React from "react";
import bytes from "bytes";
import { css } from "emotion";
import { Drawer, DrawerContent } from "@webiny/ui/Drawer";
import { IconButton } from "@webiny/ui/Button";
import getFileTypePlugin from "./getFileTypePlugin";
import get from "lodash.get";
import Tags from "./FileDetails/Tags";
import Name from "./FileDetails/Name";
import { Tooltip } from "@webiny/ui/Tooltip";
import { useHotkeys } from "react-hotkeyz";
import { ReactComponent as DownloadIcon } from "@svgr/webpack!./icons/round-cloud_download-24px.svg";
import { ReactComponent as DeleteIcon } from "@svgr/webpack!./icons/delete.svg";
import TimeAgo from "timeago-react";
import { useFileManager } from "./FileManagerContext";
import { useMutation } from "react-apollo";
import { useSnackbar } from "@webiny/app-admin/hooks/useSnackbar";
import { ConfirmationDialog } from "@webiny/ui/ConfirmationDialog";
import { DELETE_FILE } from "./graphql";
import { i18n } from "@webiny/app/i18n";
var t = i18n.ns("app-admin/file-manager/file-details");
var style = {
  wrapper: /*#__PURE__*/css({
    padding: 10,
    height: "100%",
    overflow: "auto"
  }, "label:wrapper;"),
  header: /*#__PURE__*/css({
    fontSize: 18,
    textAlign: "center",
    padding: 10,
    fontWeight: 600,
    color: "var(--mdc-theme-on-surface)"
  }, "label:header;"),
  preview: /*#__PURE__*/css({
    backgroundColor: "var(--mdc-theme-background)",
    padding: 10,
    position: "relative",
    width: 200,
    height: 200,
    margin: "0 auto",
    img: {
      maxHeight: 200,
      maxWidth: 200,
      width: "auto",
      position: "absolute",
      top: "50%",
      left: "50%",
      transform: "translateX(-50%) translateY(-50%)",
      backgroundColor: "#fff"
    }
  }, "label:preview;"),
  download: /*#__PURE__*/css({
    textAlign: "center",
    margin: "0 auto",
    width: "100%"
  }, "label:download;"),
  list: /*#__PURE__*/css({
    textAlign: "left",
    color: "var(--mdc-theme-on-surface)",
    li: {
      padding: 10,
      lineHeight: "22px",
      "li-title": {
        fontWeight: 600,
        display: "block"
      },
      "li-content": {
        whiteSpace: "nowrap",
        overflow: "hidden",
        textOverflow: "ellipsis",
        width: "100%",
        display: "block"
      }
    }
  }, "label:list;")
};
export default function FileDetails(props) {
  var file = props.file,
      uploadFile = props.uploadFile,
      validateFiles = props.validateFiles;
  var filePlugin = getFileTypePlugin(file);
  var actions = get(filePlugin, "fileDetails.actions") || [];

  var _useFileManager = useFileManager(),
      hideFileDetails = _useFileManager.hideFileDetails;

  useHotkeys({
    zIndex: 55,
    disabled: !file,
    keys: {
      esc: hideFileDetails
    }
  });

  var _useMutation = useMutation(DELETE_FILE, {
    refetchQueries: ["ListFiles"]
  }),
      _useMutation2 = _slicedToArray(_useMutation, 1),
      deleteFile = _useMutation2[0];

  var _useSnackbar = useSnackbar(),
      showSnackbar = _useSnackbar.showSnackbar;

  var fileDeleteConfirmationProps = {
    title: t(_templateObject()),
    message: file && /*#__PURE__*/React.createElement("span", null, t(_templateObject2())({
      name: file.name
    }))
  };
  return /*#__PURE__*/React.createElement(Drawer, {
    dir: "rtl",
    modal: true,
    open: file,
    onClose: hideFileDetails
  }, file && /*#__PURE__*/React.createElement("div", {
    className: style.wrapper,
    dir: "ltr"
  }, /*#__PURE__*/React.createElement("div", {
    className: style.header
  }, "File details"), /*#__PURE__*/React.createElement("div", {
    className: style.preview
  }, filePlugin.render({
    file: file,
    uploadFile: uploadFile,
    validateFiles: validateFiles
  })), /*#__PURE__*/React.createElement("div", {
    className: style.download
  }, /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(Tooltip, {
    content: /*#__PURE__*/React.createElement("span", null, t(_templateObject3())),
    placement: "bottom"
  }, /*#__PURE__*/React.createElement(IconButton, {
    onClick: function onClick() {
      return window.open(file.src, "_blank");
    },
    icon: /*#__PURE__*/React.createElement(DownloadIcon, {
      style: {
        margin: "0 8px 0 0"
      }
    })
  })), actions.map(function (Component, index) {
    return /*#__PURE__*/React.createElement(Component, Object.assign({
      key: index
    }, props));
  }), /*#__PURE__*/React.createElement(ConfirmationDialog, fileDeleteConfirmationProps, function (_ref) {
    var showConfirmation = _ref.showConfirmation;
    return /*#__PURE__*/React.createElement(Tooltip, {
      content: /*#__PURE__*/React.createElement("span", null, t(_templateObject4())),
      placement: "bottom"
    }, /*#__PURE__*/React.createElement(IconButton, {
      icon: /*#__PURE__*/React.createElement(DeleteIcon, {
        style: {
          margin: "0 8px 0 0"
        }
      }),
      onClick: function onClick() {
        return showConfirmation( /*#__PURE__*/_asyncToGenerator( /*#__PURE__*/_regeneratorRuntime.mark(function _callee() {
          return _regeneratorRuntime.wrap(function _callee$(_context) {
            while (1) {
              switch (_context.prev = _context.next) {
                case 0:
                  _context.next = 2;
                  return deleteFile({
                    variables: {
                      id: file.id
                    }
                  });

                case 2:
                  showSnackbar(t(_templateObject5()));

                case 3:
                case "end":
                  return _context.stop();
              }
            }
          }, _callee);
        })));
      }
    }));
  }))), /*#__PURE__*/React.createElement(DrawerContent, {
    dir: "ltr"
  }, /*#__PURE__*/React.createElement("ul", {
    className: style.list
  }, /*#__PURE__*/React.createElement("li", null, /*#__PURE__*/React.createElement("li-title", null, t(_templateObject6())), /*#__PURE__*/React.createElement(Name, props)), /*#__PURE__*/React.createElement("li", null, /*#__PURE__*/React.createElement("li-title", null, t(_templateObject7())), /*#__PURE__*/React.createElement("li-content", null, bytes.format(file.size))), /*#__PURE__*/React.createElement("li", null, /*#__PURE__*/React.createElement("li-title", null, t(_templateObject8())), /*#__PURE__*/React.createElement("li-content", null, file.type)), /*#__PURE__*/React.createElement("li", null, /*#__PURE__*/React.createElement("li-title", null, t(_templateObject9())), /*#__PURE__*/React.createElement(Tags, props)), /*#__PURE__*/React.createElement("li", null, /*#__PURE__*/React.createElement("li-title", null, t(_templateObject10())), /*#__PURE__*/React.createElement("li-content", null, /*#__PURE__*/React.createElement(TimeAgo, {
    datetime: file.createdOn
  })))))));
}
