// @webiny/app-admin/components/FileManager/File.js
import React from "react";
import { css } from "emotion";
import { ReactComponent as Checked } from "@svgr/webpack!./icons/round-check_box-24px.svg";
import { ReactComponent as InfoIcon } from "@svgr/webpack!./icons/round-info-24px.svg";
import { Ripple } from "@webiny/ui/Ripple";
import LazyLoad from "react-lazy-load";
var COMPONENT_WIDTH = 200;
var COMPONENT_HEIGHT = 200;
var styles = /*#__PURE__*/css({
  display: "inline-block",
  float: "left",
  position: "relative",
  zIndex: 1,
  margin: 10,
  cursor: "pointer",
  width: "100%",
  maxWidth: COMPONENT_WIDTH,
  border: "1px solid var(--mdc-theme-on-background)",
  borderRadius: 2,
  "> .body": {
    width: COMPONENT_WIDTH,
    height: COMPONENT_HEIGHT,
    overflow: "hidden",
    "--icon-color": "var(--mdc-theme-on-background)",
    ".checkedIcon": {
      color: "var(--mdc-theme-secondary)",
      position: "absolute",
      top: 4,
      left: 4,
      zIndex: 11
    },
    ".infoIcon": {
      opacity: 0,
      color: "var(--icon-color)",
      position: "absolute",
      top: 4,
      right: 4,
      zIndex: 10,
      transition: "all 150ms ease-in"
    },
    ".filePreview": {
      textAlign: "center",
      position: "relative",
      backgroundColor: "#fff",
      width: "100%",
      height: "100%",
      ".clickableArea": {
        position: "absolute",
        top: 30,
        left: 0,
        width: "100%",
        height: 170,
        zIndex: 2
      }
    },
    "&:hover .infoIcon": {
      opacity: 1,
      "--icon-color": "var(--mdc-theme-secondary)"
    }
  },
  "> .label": {
    padding: "15px 10px",
    whiteSpace: "nowrap",
    overflow: "hidden",
    textOverflow: "ellipsis",
    fontSize: "0.8rem",
    color: "var(--mdc-theme-on-surface)",
    backgroundColor: "var(--mdc-theme-on-background)"
  }
}, "label:styles;");
export default /*#__PURE__*/React.memo(function File(props) {
  var file = props.file,
      selected = props.selected,
      onSelect = props.onSelect,
      children = props.children,
      showFileDetails = props.showFileDetails;
  return /*#__PURE__*/React.createElement("div", {
    className: styles
  }, /*#__PURE__*/React.createElement("div", {
    className: "body"
  }, /*#__PURE__*/React.createElement("div", {
    className: "checkedIcon",
    onClick: onSelect
  }, selected ? /*#__PURE__*/React.createElement(Checked, null) : null), /*#__PURE__*/React.createElement("div", {
    className: "infoIcon"
  }, /*#__PURE__*/React.createElement(InfoIcon, {
    onClick: showFileDetails
  })), /*#__PURE__*/React.createElement(LazyLoad, {
    height: 200,
    offsetVertical: 300
  }, /*#__PURE__*/React.createElement(Ripple, null, /*#__PURE__*/React.createElement("div", {
    className: "filePreview"
  }, /*#__PURE__*/React.createElement("div", {
    className: "clickableArea",
    onClick: onSelect
  }), children)))), /*#__PURE__*/React.createElement("div", {
    className: "label",
    onClick: onSelect
  }, file.name));
}, function (prev, next) {
  if (prev.selected !== next.selected) {
    return false;
  }

  if (prev.file.name !== next.file.name) {
    return false;
  }

  return true;
});
