// @webiny/app-admin/components/FileManager/LeftSidebar.js
import _get from "lodash/get";
import React from "react";
import { Query } from "react-apollo";
import { LIST_TAGS } from "./graphql";
import { css } from "emotion";
import { Icon } from "@webiny/ui/Icon";
import styled from "@emotion/styled";
import { ReactComponent as TagIcon } from "@svgr/webpack!./icons/round-label-24px.svg";
var style = {
  leftDrawer: /*#__PURE__*/css({
    float: "left",
    display: "inline-block",
    width: 250,
    height: "100%",
    backgroundColor: "var(--mdc-theme-surface)",
    padding: 10
  }, "label:leftDrawer;"),
  noTagged: /*#__PURE__*/css({
    paddingTop: 100,
    textAlign: "center",
    color: "var(--mdc-theme-on-surface)"
  }, "label:noTagged;")
};
var TagList = /*#__PURE__*/styled("div", {
  target: "e1tgxgqn0",
  label: "TagList"
})({
  display: "flex",
  flexDirection: "column"
});
var Tag = /*#__PURE__*/styled("div", {
  target: "e1tgxgqn1",
  label: "Tag"
})({
  display: "flex",
  flexDirection: "row",
  height: 40,
  alignItems: "center",
  cursor: "pointer",
  paddingLeft: 15,
  color: "var(--mdc-theme-on-surface)",
  svg: {
    color: "var(--mdc-theme-on-surface)",
    marginRight: 10
  },
  "&:hover": {
    backgroundColor: "var(--mdc-theme-background)"
  },
  "&.active": {
    svg: {
      color: "var(--mdc-theme-secondary)"
    }
  }
});

function LeftSidebar(_ref) {
  var toggleTag = _ref.toggleTag,
      tags = _ref.queryParams.tags;
  var activeTags = Array.isArray(tags) ? tags : [];
  return /*#__PURE__*/React.createElement("div", {
    className: style.leftDrawer
  }, /*#__PURE__*/React.createElement(Query, {
    query: LIST_TAGS
  }, function (_ref2) {
    var data = _ref2.data;
    var list = _get(data, "files.listTags") || [];

    if (!list.length) {
      return /*#__PURE__*/React.createElement("div", {
        className: style.noTagged
      }, "Once you tag an image, the tag will be displayed here.");
    }

    return /*#__PURE__*/React.createElement(TagList, null, list.map(function (item, index) {
      return /*#__PURE__*/React.createElement(Tag, {
        className: activeTags.includes(item) && "active",
        key: item + index,
        onClick: function onClick() {
          return toggleTag(item);
        }
      }, /*#__PURE__*/React.createElement(Icon, {
        icon: /*#__PURE__*/React.createElement(TagIcon, null)
      }), " ", item);
    }));
  }));
}

export default LeftSidebar;
