// @webiny/app-page-builder/render/plugins/slate/typography/index.js
import _objectSpread from "@babel/runtime/helpers/objectSpread2";
import React from "react";

/**
 * TODO: figure out why slate typings are not really working here (see all the @ts-ignore lines)
 * Maybe I'm doing something wrong; if anybody knows - please help!
 */
export default (function () {
  return {
    name: "pb-render-slate-editor-typography",
    type: "pb-render-slate-editor",
    slate: {
      renderNode: function renderNode(props, next) {
        var attributes = props.attributes,
            children = props.children,
            node = props.node,
            editor = props.editor; // @ts-ignore

        var type = node.type; // @ts-ignore

        var typography = editor.props.theme.typography;

        if (typography.hasOwnProperty(type) && typography[type].component) {
          var _typography$type = typography[type],
              Node = _typography$type.component,
              _typography$type$clas = _typography$type.className,
              className = _typography$type$clas === void 0 ? null : _typography$type$clas;

          var nodeProps = _objectSpread(_objectSpread({}, attributes), {}, {
            className: className,
            // @ts-ignore
            style: {
              textAlign: "".concat(node.data.get("align"))
            }
          });

          if (typeof Node !== "string") {
            // @ts-ignore
            nodeProps = props;
          }

          return /*#__PURE__*/React.createElement(Node, nodeProps, children);
        }

        return next();
      }
    }
  };
});
