// @webiny/app-page-builder-theme/components/DesktopHeader.js
import React from "react";
import { Link } from "@webiny/react-router";
import { Menu } from "@webiny/app-page-builder/render/components";
import Navigation from "./Navigation";

var DesktopHeader = function DesktopHeader(_ref) {
  var menuName = _ref.menuName,
      logo = _ref.logo,
      name = _ref.name;
  return /*#__PURE__*/React.createElement("div", {
    className: "webiny-pb-section-header__wrapper hide-on-mobile",
    "data-testid": "pb-desktop-header"
  }, /*#__PURE__*/React.createElement("div", {
    className: "webiny-pb-section-header__logo"
  }, /*#__PURE__*/React.createElement(Link, {
    to: "/"
  }, logo && logo.src && /*#__PURE__*/React.createElement("img", {
    src: logo.src,
    alt: name
  }), " ", (!logo || !logo.src) && /*#__PURE__*/React.createElement("span", {
    className: "webiny-pb-section-header__site-name"
  }, name))), /*#__PURE__*/React.createElement("nav", {
    className: "webiny-pb-section-header__navigation"
  }, /*#__PURE__*/React.createElement(Menu, {
    slug: menuName,
    component: Navigation
  })));
};

export default DesktopHeader;
