// @webiny/app-page-builder-theme/components/Footer.js
import * as React from "react";
import { Link } from "@webiny/react-router";
import { Query } from "react-apollo";
import get from "lodash.get";
import { getFooterData } from "./graphql";
import { ReactComponent as FacebookIcon } from "@svgr/webpack!./assets/facebook-square-brands.svg";
import { ReactComponent as TwitterIcon } from "@svgr/webpack!./assets/twitter-square-brands.svg";
import { ReactComponent as InstagramIcon } from "@svgr/webpack!./assets/instagram-brands.svg";

var Footer = function Footer() {
  return /*#__PURE__*/React.createElement(Query, {
    query: getFooterData
  }, function (_ref) {
    var response = _ref.data;

    var _ref2 = get(response, "pageBuilder.getSettings.data") || {},
        name = _ref2.name,
        logo = _ref2.logo,
        social = _ref2.social;

    return /*#__PURE__*/React.createElement("div", {
      className: "webiny-pb-section-footer",
      "data-testid": "pb-footer"
    }, /*#__PURE__*/React.createElement("div", {
      className: "webiny-pb-section-footer__wrapper"
    }, /*#__PURE__*/React.createElement("div", {
      className: "webiny-pb-section-footer__logo"
    }, /*#__PURE__*/React.createElement(Link, {
      to: "/"
    }, logo && logo.src && /*#__PURE__*/React.createElement("img", {
      src: logo.src,
      alt: name
    })), /*#__PURE__*/React.createElement("div", {
      className: "webiny-pb-section-footer__copy webiny-pb-typography-description"
    }, name, " \xA9 ", new Date().getFullYear())), social && /*#__PURE__*/React.createElement("div", {
      className: "webiny-pb-section-footer__social"
    }, social.facebook && /*#__PURE__*/React.createElement("a", {
      href: social.facebook
    }, /*#__PURE__*/React.createElement(FacebookIcon, null)), social.twitter && /*#__PURE__*/React.createElement("a", {
      href: social.twitter
    }, /*#__PURE__*/React.createElement(TwitterIcon, null)), social.instagram && /*#__PURE__*/React.createElement("a", {
      href: social.instagram
    }, /*#__PURE__*/React.createElement(InstagramIcon, null)))));
  });
};

export default Footer;
