// @webiny/app-page-builder-theme/components/Header.js
import _slicedToArray from "@babel/runtime/helpers/slicedToArray";
import React, { useState, useCallback } from "react";
import get from "lodash.get";
import { Query } from "react-apollo";
import { getHeaderData } from "./graphql";
import DesktopHeader from "./DesktopHeader";
import MobileHeader from "./MobileHeader";
var menuName = "main-menu";

var Header = function Header() {
  var _useState = useState(false),
      _useState2 = _slicedToArray(_useState, 2),
      mobileMenu = _useState2[0],
      showMobileMenu = _useState2[1];

  var toggleMobileMenu = useCallback(function () {
    showMobileMenu(!mobileMenu);
  }, [mobileMenu]);
  return /*#__PURE__*/React.createElement(Query, {
    query: getHeaderData
  }, function (_ref) {
    var response = _ref.data;

    var _ref2 = get(response, "pageBuilder.getSettings.data") || {},
        name = _ref2.name,
        logo = _ref2.logo;

    return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement("div", {
      className: "webiny-pb-section-header",
      "data-testid": "pb-desktop-mobile-headers"
    }, /*#__PURE__*/React.createElement(DesktopHeader, {
      menuName: menuName,
      name: name,
      logo: logo
    }), /*#__PURE__*/React.createElement(MobileHeader, {
      menuName: menuName,
      name: name,
      logo: logo,
      active: mobileMenu,
      toggleMenu: toggleMobileMenu
    })), /*#__PURE__*/React.createElement("div", {
      className: "webiny-pb-section-header-spacer"
    }));
  });
};

export default Header;
