// @webiny/app-page-builder-theme/components/MobileHeader.js
import React from "react";
import { Link } from "@webiny/react-router";
import HamburgerMenu from "react-hamburger-menu";
import classNames from "classnames";
import { Menu } from "@webiny/app-page-builder/render/components";
import Navigation from "./Navigation";

var MobileHeader = function MobileHeader(_ref) {
  var menuName = _ref.menuName,
      logo = _ref.logo,
      name = _ref.name,
      active = _ref.active,
      toggleMenu = _ref.toggleMenu;
  return /*#__PURE__*/React.createElement("div", {
    className: "webiny-pb-section-header__wrapper hide-on-desktop-and-tablet",
    "data-testid": "pb-mobile-header"
  }, /*#__PURE__*/React.createElement("div", {
    className: "webiny-pb-section-header__logo"
  }, /*#__PURE__*/React.createElement(Link, {
    to: "/"
  }, logo && logo.src && /*#__PURE__*/React.createElement("img", {
    src: logo.src,
    alt: name
  }), " ", (!logo || !logo.src) && /*#__PURE__*/React.createElement("span", {
    className: "webiny-pb-section-header__site-name"
  }, name))), /*#__PURE__*/React.createElement("nav", {
    className: classNames("webiny-pb-section-header__navigation", {
      "webiny-pb-section-header__navigation--mobile-active": active
    })
  }, /*#__PURE__*/React.createElement(Menu, {
    slug: menuName,
    component: Navigation
  }), /*#__PURE__*/React.createElement("div", {
    className: "webiny-pb-section-header__mobile-site-name"
  }, /*#__PURE__*/React.createElement("a", {
    href: "/"
  }, name))), /*#__PURE__*/React.createElement("div", {
    onClick: toggleMenu,
    className: "webiny-pb-section-header__mobile-icon"
  }, /*#__PURE__*/React.createElement(HamburgerMenu, {
    isOpen: active,
    menuClicked: toggleMenu,
    width: 18,
    height: 15,
    strokeWidth: 1,
    rotate: 0,
    color: "black",
    borderRadius: 0,
    animationDuration: 0.5
  })), /*#__PURE__*/React.createElement("div", {
    onClick: toggleMenu,
    className: classNames("webiny-pb-section-header__mobile-overlay", {
      "webiny-pb-section-header__mobile-overlay--active": active
    })
  }));
};

export default MobileHeader;
