// @webiny/app-i18n/site/plugins/richTextEditor/lists/index.js
import * as React from "react";
var plugins = [{
  name: "i18n-value-rich-text-editor-unordered-list",
  type: "i18n-value-rich-text-editor",
  plugin: {
    name: "unordered-list",
    editor: {
      renderElement: function renderElement(_ref, next) {
        var attributes = _ref.attributes,
            children = _ref.children,
            element = _ref.element;

        switch (element.type) {
          case "unordered-list":
            return /*#__PURE__*/React.createElement("ul", Object.assign({
              className: "webiny-rich-typography-unordered-list"
            }, attributes), children);

          case "list-item":
            return /*#__PURE__*/React.createElement("li", attributes, children);

          default:
            return next();
        }
      }
    }
  }
}, {
  name: "i18n-value-rich-text-editor-ordered-list",
  type: "i18n-value-rich-text-editor",
  plugin: {
    name: "ordered-list",
    editor: {
      renderElement: function renderElement(_ref2, next) {
        var attributes = _ref2.attributes,
            children = _ref2.children,
            element = _ref2.element;

        switch (element.type) {
          case "ordered-list":
            return /*#__PURE__*/React.createElement("ol", Object.assign({
              className: "webiny-rich-typography-ordered-list"
            }, attributes), children);

          case "list-item":
            return /*#__PURE__*/React.createElement("li", attributes, children);

          default:
            return next();
        }
      }
    }
  }
}];
export default plugins;
