// @webiny/app-page-builder/editor/plugins/elements/media/soundcloud/index.js
import React from "react";
import styled from "@emotion/styled";
import { Tab } from "@webiny/ui/Tabs";
import { Input } from "@webiny/ui/Input";
import { Grid, Cell } from "@webiny/ui/Grid";
import { createEmbedPlugin, createEmbedSettingsPlugin } from "./../../utils/oembed/createEmbedPlugin";
import { ReactComponent as MediaIcon } from "@svgr/webpack!./../../../elementGroups/media/round-music_video-24px.svg";
import { ReactComponent as LogoIcon } from "@svgr/webpack!./soundcloud-brands.svg";
import { validation } from "@webiny/validation";
var PreviewBox = /*#__PURE__*/styled("div", {
  target: "elfmwaw0",
  label: "PreviewBox"
})({
  textAlign: "center",
  height: 50,
  svg: {
    height: 50,
    width: 50
  }
});
export default (function () {
  return [createEmbedPlugin({
    type: "soundcloud",
    toolbar: {
      title: "Soundcloud",
      group: "pb-editor-element-group-media",
      preview: function preview() {
        return /*#__PURE__*/React.createElement(PreviewBox, null, /*#__PURE__*/React.createElement(LogoIcon, null));
      }
    }
  }), createEmbedSettingsPlugin({
    type: "soundcloud",
    render: function render(_ref) {
      var Bind = _ref.Bind;
      return /*#__PURE__*/React.createElement(Tab, {
        icon: /*#__PURE__*/React.createElement(MediaIcon, null),
        label: "Soundcloud"
      }, /*#__PURE__*/React.createElement(Grid, null, /*#__PURE__*/React.createElement(Cell, {
        span: 12
      }, /*#__PURE__*/React.createElement(Bind, {
        name: "source.url",
        validators: validation.create("required,url")
      }, /*#__PURE__*/React.createElement(Input, {
        label: "Song URL",
        description: "Enter a song URL"
      })))));
    }
  })];
});
