// @webiny/app-headless-cms/admin/views/AccessTokens/index.js
import React from "react";
import { SplitView, LeftPanel, RightPanel } from "@webiny/app-admin/components/SplitView";
import { FloatingActionButton } from "@webiny/app-admin/components/FloatingActionButton";
import AccessTokensDataList from "./AccessTokensDataList";
import AccessTokensForm from "./AccessTokensForm";
import { CrudProvider } from "@webiny/app-admin/contexts/Crud";
import { GET_ACCESS_TOKEN, LIST_ACCESS_TOKENS, CREATE_ACCESS_TOKEN, UPDATE_ACCESS_TOKEN, DELETE_ACCESS_TOKEN } from "./graphql";

var getMutationVariables = function getMutationVariables(formData) {
  delete formData.token;
  formData.environments = formData.selectedEnvironmentIds;
  delete formData.selectedEnvironmentIds;
  return {
    data: formData
  };
};

function EnvironmentAliases() {
  return /*#__PURE__*/React.createElement(CrudProvider, {
    delete: {
      mutation: DELETE_ACCESS_TOKEN
    },
    read: GET_ACCESS_TOKEN,
    create: {
      mutation: CREATE_ACCESS_TOKEN,
      variables: getMutationVariables
    },
    update: {
      mutation: UPDATE_ACCESS_TOKEN,
      variables: getMutationVariables
    },
    list: {
      query: LIST_ACCESS_TOKENS,
      variables: {
        sort: {
          savedOn: -1
        }
      }
    }
  }, function (_ref) {
    var actions = _ref.actions;
    return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(SplitView, null, /*#__PURE__*/React.createElement(LeftPanel, {
      span: 4
    }, /*#__PURE__*/React.createElement(AccessTokensDataList, null)), /*#__PURE__*/React.createElement(RightPanel, {
      span: 8
    }, /*#__PURE__*/React.createElement(AccessTokensForm, null))), /*#__PURE__*/React.createElement(FloatingActionButton, {
      "data-testid": "new-record-button",
      onClick: actions.resetForm
    }));
  });
}

export default EnvironmentAliases;
