// @webiny/app-admin/components/Install/AppInstaller.js
import _slicedToArray from "@babel/runtime/helpers/slicedToArray";
import React, { useState } from "react";
import styled from "@emotion/styled";
import { css } from "emotion";
import { CircularProgress } from "@webiny/ui/Progress";
import { ButtonPrimary } from "@webiny/ui/Button";
import { SplitView, LeftPanel, RightPanel } from "@webiny/app-admin/components/SplitView";
import { useInstaller } from "./useInstaller";
import Sidebar from "./Sidebar";
import { Elevation } from "@webiny/ui/Elevation";
export var Wrapper = /*#__PURE__*/styled("section", {
  target: "e280ipe0",
  label: "Wrapper"
})({
  display: "flex",
  flexDirection: "column",
  justifyContent: "center",
  minHeight: "100vh",
  color: "var(--mdc-theme-on-surface)"
});
export var InstallContent = /*#__PURE__*/styled("div", {
  target: "e280ipe1",
  label: "InstallContent"
})({
  maxWidth: 800,
  margin: "0 auto 25px auto",
  ".mdc-elevation--z2": {
    borderRadius: 4,
    boxShadow: "0 1px 3px 0 rgba(0,0,0,0.15)"
  }
});
var installerSplitView = /*#__PURE__*/css({
  ".webiny-split-view__inner": {
    height: "100vh",
    ".webiny-split-view__right-panel-wrapper": {
      height: "100vh"
    }
  }
}, "label:installerSplitView;");
var SuccessDialog = /*#__PURE__*/styled("div", {
  target: "e280ipe2",
  label: "SuccessDialog"
})({
  padding: 40,
  textAlign: "center",
  p: {
    paddingBottom: 40
  }
});
export var InnerContent = /*#__PURE__*/styled("div", {
  target: "e280ipe3",
  label: "InnerContent"
})({
  padding: 25,
  position: "relative"
});
export var AppInstaller = function AppInstaller(_ref) {
  var children = _ref.children,
      security = _ref.security;

  var _useState = useState(false),
      _useState2 = _slicedToArray(_useState, 2),
      finished = _useState2[0],
      setFinished = _useState2[1];

  var _useInstaller = useInstaller(),
      loading = _useInstaller.loading,
      installers = _useInstaller.installers,
      installer = _useInstaller.installer,
      showNextInstaller = _useInstaller.showNextInstaller,
      showLogin = _useInstaller.showLogin,
      onUser = _useInstaller.onUser;

  var renderLayout = function renderLayout(content) {
    var secure = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : false;
    return /*#__PURE__*/React.createElement(SplitView, {
      className: installerSplitView
    }, /*#__PURE__*/React.createElement(LeftPanel, {
      span: 2
    }, /*#__PURE__*/React.createElement(Sidebar, {
      allInstallers: installers,
      installer: installer,
      showLogin: showLogin
    })), /*#__PURE__*/React.createElement(RightPanel, {
      span: 10
    }, !showLogin && !secure && content, (showLogin || secure) && /*#__PURE__*/React.cloneElement(security, {
      onUser: onUser
    }, content)));
  };

  var renderBody = function renderBody(content) {
    return /*#__PURE__*/React.createElement(Wrapper, null, /*#__PURE__*/React.createElement(InstallContent, null, /*#__PURE__*/React.createElement(InnerContent, null, content)));
  }; // Loading installers data


  if (loading) {
    return /*#__PURE__*/React.createElement(CircularProgress, {
      label: "Checking apps..."
    });
  } // This means there are no installers to run or installation was finished


  if (!loading && (installers.length === 0 || finished)) {
    return /*#__PURE__*/React.cloneElement(security, null, children);
  }

  if (installer) {
    return renderLayout(renderBody(installer.plugin.render({
      onInstalled: showNextInstaller
    })), installer.plugin.secure);
  }

  return renderLayout(renderBody( /*#__PURE__*/React.createElement(Elevation, {
    z: 1
  }, /*#__PURE__*/React.createElement(SuccessDialog, null, /*#__PURE__*/React.createElement("p", null, "You have successfully installed all new applications!"), /*#__PURE__*/React.createElement(ButtonPrimary, {
    "data-testid": "open-webiny-cms-admin-button",
    onClick: function onClick() {
      return setFinished(true);
    }
  }, "Open Webiny CMS Administration")))), true);
};
