// @webiny/app-admin/components/Install/useInstaller.js
import _regeneratorRuntime from "@babel/runtime/regenerator";
import _asyncToGenerator from "@babel/runtime/helpers/asyncToGenerator";
import _objectSpread from "@babel/runtime/helpers/objectSpread2";
import _slicedToArray from "@babel/runtime/helpers/slicedToArray";
import { useCallback, useReducer, useEffect } from "react";
import { Graph, alg } from "graphlib";
import { useApolloClient } from "react-apollo";
import { getPlugins } from "@webiny/plugins";
export var useInstaller = function useInstaller() {
  var _useReducer = useReducer(function (prev, next) {
    return _objectSpread(_objectSpread({}, prev), next);
  }, {
    loading: true,
    installers: [],
    installerIndex: -1,
    showLogin: false
  }),
      _useReducer2 = _slicedToArray(_useReducer, 2),
      _useReducer2$ = _useReducer2[0],
      loading = _useReducer2$.loading,
      installers = _useReducer2$.installers,
      installerIndex = _useReducer2$.installerIndex,
      showLogin = _useReducer2$.showLogin,
      setState = _useReducer2[1];

  var client = useApolloClient();

  var validateGraph = function validateGraph(graph) {
    var isAcyclic = alg.isAcyclic(graph);

    if (!isAcyclic) {
      var cycles = alg.findCycles(graph);
      var msg = ["Your installers have circular dependencies:"];
      cycles.forEach(function (cycle, index) {
        var fromAToB = cycle.join(" --> ");
        fromAToB = "".concat(index + 1, ". ").concat(fromAToB);
        var fromBToA = cycle.reverse().join(" <-- ");
        var padLength = fromAToB.length + 4;
        msg.push(fromAToB.padStart(padLength));
        msg.push(fromBToA.padStart(padLength));
      }, cycles);
      throw new Error(msg.join("\n"));
    }
  };

  var createGraph = function createGraph(installers) {
    var graph = new Graph();
    installers.forEach(function (_ref) {
      var plugin = _ref.plugin;
      graph.setNode(plugin.name, plugin);
    });
    installers.forEach(function (_ref2) {
      var pl = _ref2.plugin;

      if (Array.isArray(pl.dependencies)) {
        pl.dependencies.forEach(function (dep) {
          graph.setEdge(pl.name, dep);
        });
      }
    });
    validateGraph(graph);
    return graph;
  };

  var getInstallers = useCallback(function (installers, graph) {
    var list = arguments.length > 2 && arguments[2] !== undefined ? arguments[2] : [];
    var leaf = graph.sinks()[0];

    if (leaf) {
      var installer = installers.find(function (inst) {
        return inst.plugin.name === leaf;
      });
      graph.removeNode(leaf);

      if (!installer.installed) {
        list.push(installer);
      }

      return getInstallers(installers, graph, list);
    }

    return list;
  }, []);

  var onUser = function onUser() {
    setState({
      showLogin: false
    });
  };

  var showNextInstaller = function showNextInstaller() {
    var prevInstaller = installers[installerIndex];
    installers[installerIndex].installed = true;
    setState({
      installers: installers
    });

    if (installers.length < installerIndex + 1) {
      setState({
        installerIndex: null
      });
      return;
    }

    var nextIndex = installerIndex + 1;
    var showLogin = false;
    var nextInstaller = installers[nextIndex];
    var prevSecure = prevInstaller && prevInstaller.plugin.secure;
    var nextSecure = nextInstaller && nextInstaller.plugin.secure;

    if (!prevSecure && nextSecure) {
      showLogin = true;
    }

    setState({
      installerIndex: nextIndex,
      showLogin: showLogin
    });
  };

  useEffect(function () {
    _asyncToGenerator( /*#__PURE__*/_regeneratorRuntime.mark(function _callee2() {
      var allInstallers, graph, installers;
      return _regeneratorRuntime.wrap(function _callee2$(_context2) {
        while (1) {
          switch (_context2.prev = _context2.next) {
            case 0:
              allInstallers = [];
              _context2.next = 3;
              return Promise.all(getPlugins("admin-installation").map( /*#__PURE__*/function () {
                var _ref4 = _asyncToGenerator( /*#__PURE__*/_regeneratorRuntime.mark(function _callee(pl) {
                  var installed;
                  return _regeneratorRuntime.wrap(function _callee$(_context) {
                    while (1) {
                      switch (_context.prev = _context.next) {
                        case 0:
                          _context.next = 2;
                          return pl.isInstalled({
                            client: client
                          });

                        case 2:
                          installed = _context.sent;
                          allInstallers.push({
                            plugin: pl,
                            installed: installed
                          });

                        case 4:
                        case "end":
                          return _context.stop();
                      }
                    }
                  }, _callee);
                }));

                return function (_x) {
                  return _ref4.apply(this, arguments);
                };
              }()));

            case 3:
              graph = createGraph(allInstallers);
              installers = getInstallers(allInstallers, graph);
              setState({
                installers: installers,
                installerIndex: 0,
                loading: false
              });

            case 6:
            case "end":
              return _context2.stop();
          }
        }
      }, _callee2);
    }))();
  }, []);
  return {
    loading: loading,
    installers: installers,
    installer: installers[installerIndex],
    showNextInstaller: showNextInstaller,
    showLogin: showLogin,
    onUser: onUser
  };
};
