// @webiny/app-headless-cms/admin/components/ContentModelEditor/Tabs/EditTab/Field.js
import _regeneratorRuntime from "@babel/runtime/regenerator";
import _asyncToGenerator from "@babel/runtime/helpers/asyncToGenerator";
import _taggedTemplateLiteral from "@babel/runtime/helpers/taggedTemplateLiteral";

function _templateObject6() {
  var data = _taggedTemplateLiteral(["Delete"]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = _taggedTemplateLiteral(["Cannot delete"]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _taggedTemplateLiteral(["Use as title"]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _taggedTemplateLiteral(["Title field set successfully."]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteral(["entry title"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteral(["multiple values"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

import React from "react";
import styled from "@emotion/styled";
import { IconButton } from "@webiny/ui/Button";
import { Typography } from "@webiny/ui/Typography";
import { ReactComponent as EditIcon } from "@svgr/webpack!@webiny/app-headless-cms/admin/icons/edit.svg";
import { ReactComponent as DeleteIcon } from "@svgr/webpack!@webiny/app-headless-cms/admin/icons/delete.svg";
import { ReactComponent as TitleIcon } from "@svgr/webpack!@webiny/app-headless-cms/admin/icons/title-24px.svg";
import { useI18N } from "@webiny/app-i18n/hooks/useI18N";
import { useContentModelEditor } from "@webiny/app-headless-cms/admin/components/ContentModelEditor/Context";
import { ReactComponent as MoreVerticalIcon } from "@svgr/webpack!@webiny/app-headless-cms/admin/icons/more_vert.svg";
import { Menu, MenuItem } from "@webiny/ui/Menu";
import { getPlugins } from "@webiny/plugins";
import { css } from "emotion";
import { ListItemGraphic } from "@webiny/ui/List";
import { Icon } from "@webiny/ui/Icon";
import { i18n } from "@webiny/app/i18n";
import { useSnackbar } from "@webiny/app-admin/hooks/useSnackbar";
var t = i18n.ns("app-headless-cms/admin/components/editor/field");
var FieldContainer = /*#__PURE__*/styled("div", {
  target: "e4t4mqa0",
  label: "FieldContainer"
})({
  display: "flex",
  flexDirection: "row",
  justifyContent: "space-between",
  alignItems: "center"
});
var Info = /*#__PURE__*/styled("div", {
  target: "e4t4mqa1",
  label: "Info"
})({
  display: "flex",
  flexDirection: "column",
  "> *": {
    flex: "1 100%",
    lineHeight: "150%"
  }
});
var Actions = /*#__PURE__*/styled("div", {
  target: "e4t4mqa2",
  label: "Actions"
})({
  display: "flex",
  flexDirection: "row",
  alignItems: "right",
  "> *": {
    flex: "1 100%"
  }
});
var menuStyles = /*#__PURE__*/css({
  width: 220,
  right: -105,
  left: "auto !important",
  ".disabled": {
    opacity: 0.5,
    pointerEvents: "none"
  }
}, "label:menuStyles;");

var Field = function Field(props) {
  var field = props.field,
      onEdit = props.onEdit,
      onDelete = props.onDelete;

  var _useI18N = useI18N(),
      getValue = _useI18N.getValue;

  var _useSnackbar = useSnackbar(),
      showSnackbar = _useSnackbar.showSnackbar;

  var _useContentModelEdito = useContentModelEditor(),
      getFieldPlugin = _useContentModelEdito.getFieldPlugin,
      setData = _useContentModelEdito.setData,
      data = _useContentModelEdito.data;

  var fieldPlugin = getFieldPlugin({
    type: field.type
  });
  var plugins = getPlugins("cms-editor-field-option");
  var lockedFields = data.lockedFields || [];
  return /*#__PURE__*/React.createElement(FieldContainer, null, /*#__PURE__*/React.createElement(Info, null, /*#__PURE__*/React.createElement(Typography, {
    use: "subtitle1"
  }, getValue(field.label), " "), /*#__PURE__*/React.createElement(Typography, {
    use: "caption"
  }, fieldPlugin && fieldPlugin.field.label, " ", field.multipleValues && /*#__PURE__*/React.createElement(React.Fragment, null, "(", t(_templateObject()), ")"), field.fieldId === data.titleFieldId && /*#__PURE__*/React.createElement(React.Fragment, null, "(", t(_templateObject2()), ")"))), /*#__PURE__*/React.createElement(Actions, null, /*#__PURE__*/React.createElement(IconButton, {
    icon: /*#__PURE__*/React.createElement(EditIcon, null),
    onClick: function onClick() {
      return onEdit(field);
    }
  }), /*#__PURE__*/React.createElement(Menu, {
    className: menuStyles,
    handle: /*#__PURE__*/React.createElement(IconButton, {
      icon: /*#__PURE__*/React.createElement(MoreVerticalIcon, null)
    })
  }, plugins.map(function (pl) {
    return /*#__PURE__*/React.cloneElement(pl.render(), {
      key: pl.name
    });
  }), /*#__PURE__*/React.createElement(MenuItem, {
    disabled: field.multipleValues || field.type !== "text",
    onClick: /*#__PURE__*/_asyncToGenerator( /*#__PURE__*/_regeneratorRuntime.mark(function _callee() {
      var response;
      return _regeneratorRuntime.wrap(function _callee$(_context) {
        while (1) {
          switch (_context.prev = _context.next) {
            case 0:
              _context.next = 2;
              return setData(function (data) {
                data.titleFieldId = field.fieldId;
                return data;
              });

            case 2:
              response = _context.sent;

              if (!response.error) {
                _context.next = 5;
                break;
              }

              return _context.abrupt("return", showSnackbar(response.error.message));

            case 5:
              showSnackbar(t(_templateObject3()));

            case 6:
            case "end":
              return _context.stop();
          }
        }
      }, _callee);
    }))
  }, /*#__PURE__*/React.createElement(ListItemGraphic, null, /*#__PURE__*/React.createElement(Icon, {
    icon: /*#__PURE__*/React.createElement(TitleIcon, null)
  })), t(_templateObject4())), /*#__PURE__*/React.createElement(MenuItem, {
    disabled: lockedFields.find(function (lockedField) {
      return lockedField.fieldId === field.fieldId;
    }),
    onClick: function onClick() {
      return onDelete(field);
    }
  }, /*#__PURE__*/React.createElement(ListItemGraphic, null, /*#__PURE__*/React.createElement(Icon, {
    icon: /*#__PURE__*/React.createElement(DeleteIcon, null)
  })), lockedFields.find(function (lockedField) {
    return lockedField.fieldId === field.fieldId;
  }) ? t(_templateObject5()) : t(_templateObject6())))));
};

export default Field;
