// @webiny/app-page-builder/editor/plugins/elements/spacer/index.js
import _objectSpread from "@babel/runtime/helpers/objectSpread2";
import React from "react";
import styled from "@emotion/styled";
import Spacer, { INIT_HEIGHT } from "./Spacer";
import { ReactComponent as SpacerIcon } from "@svgr/webpack!@webiny/app-page-builder/editor/assets/icons/spacer-icon.svg";
import "./actions";
export default (function () {
  var PreviewBox = /*#__PURE__*/styled("div", {
    target: "e1037d0p0",
    label: "PreviewBox"
  })({
    textAlign: "center",
    height: 50,
    svg: {
      height: 50,
      width: 50
    }
  });
  return {
    name: "pb-editor-page-element-spacer",
    type: "pb-editor-page-element",
    elementType: "spacer",
    toolbar: {
      title: "Spacer",
      group: "pb-editor-element-group-layout",
      preview: function preview() {
        return /*#__PURE__*/React.createElement(PreviewBox, null, /*#__PURE__*/React.createElement(SpacerIcon, null));
      }
    },
    settings: ["pb-editor-page-element-settings-delete"],
    target: ["block", "column"],
    create: function create() {
      var options = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {};
      return _objectSpread({
        type: "spacer",
        data: {
          settings: {
            height: INIT_HEIGHT,
            margin: {
              desktop: {
                all: 0
              }
            },
            padding: {
              desktop: {
                all: 0
              }
            }
          }
        }
      }, options);
    },
    render: function render(props) {
      return /*#__PURE__*/React.createElement(Spacer, props);
    }
  };
});
