// @webiny/app-page-builder/editor/plugins/elements/spacer/Spacer.js
import React from "react";
import styled from "@emotion/styled";
import { ElementRoot } from "@webiny/app-page-builder/render/components/ElementRoot";
import SpacerContainer from "./SpacerContainer";
export { MIN_HEIGHT, INIT_HEIGHT } from "./SpacerContainer";
var SpacerElement = /*#__PURE__*/styled("div", {
  target: "e1bts6jt0",
  label: "SpacerElement"
})({
  position: "relative",
  width: "100%",
  border: "2px dotted var(--mdc-theme-secondary)",
  textAlign: "center",
  opacity: 0,
  cursor: "ns-resize",
  transition: "opacity 0.2s",
  backgroundColor: "var(--mdc-theme-secondary)",
  minHeight: 20,
  "&:hover": {
    opacity: 1
  }
});

var Spacer = function Spacer(_ref) {
  var element = _ref.element;
  return /*#__PURE__*/React.createElement(SpacerElement, null, /*#__PURE__*/React.createElement(ElementRoot, {
    element: element
  }, function (_ref2) {
    var elementStyle = _ref2.elementStyle,
        customClasses = _ref2.customClasses,
        combineClassNames = _ref2.combineClassNames;
    return /*#__PURE__*/React.createElement(SpacerContainer, {
      elementId: element.id,
      elementStyle: elementStyle,
      customClasses: customClasses,
      combineClassNames: combineClassNames
    });
  }));
};

export default Spacer;
