// @webiny/app-security/admin/plugins/routes.js
import React from "react";
import Helmet from "react-helmet";
import { Route } from "@webiny/react-router";
import { AdminLayout } from "@webiny/app-admin/components/AdminLayout";
import { Roles } from "@webiny/app-security/admin/views/Roles";
import { Users } from "@webiny/app-security/admin/views/Users";
import { Groups } from "@webiny/app-security/admin/views/Groups";
import Account from "@webiny/app-security/admin/views/Account";
import { SecureRoute } from "@webiny/app-security/components";
var ROLE_SECURITY_ROLES = ["security:role:crud"];
var ROLE_SECURITY_GROUPS = ["security:group:crud"];
var ROLE_SECURITY_USERS = ["security:user:crud"];
var plugins = [{
  name: "route-roles",
  type: "route",
  route: /*#__PURE__*/React.createElement(Route, {
    exact: true,
    path: "/roles",
    render: function render() {
      return /*#__PURE__*/React.createElement(SecureRoute, {
        scopes: ROLE_SECURITY_ROLES
      }, /*#__PURE__*/React.createElement(AdminLayout, null, /*#__PURE__*/React.createElement(Helmet, {
        title: "Security - Roles"
      }), /*#__PURE__*/React.createElement(Roles, null)));
    }
  })
}, {
  name: "route-groups",
  type: "route",
  route: /*#__PURE__*/React.createElement(Route, {
    exact: true,
    path: "/groups",
    render: function render() {
      return /*#__PURE__*/React.createElement(SecureRoute, {
        scopes: ROLE_SECURITY_GROUPS
      }, /*#__PURE__*/React.createElement(AdminLayout, null, /*#__PURE__*/React.createElement(Helmet, {
        title: "Security - Groups"
      }), /*#__PURE__*/React.createElement(Groups, null)));
    }
  })
}, {
  name: "route-users",
  type: "route",
  route: /*#__PURE__*/React.createElement(Route, {
    exact: true,
    path: "/users",
    render: function render() {
      return /*#__PURE__*/React.createElement(SecureRoute, {
        scopes: ROLE_SECURITY_USERS
      }, /*#__PURE__*/React.createElement(AdminLayout, null, /*#__PURE__*/React.createElement(Helmet, {
        title: "Security - Users"
      }), /*#__PURE__*/React.createElement(Users, null)));
    }
  })
}, {
  name: "route-account",
  type: "route",
  route: /*#__PURE__*/React.createElement(Route, {
    exact: true,
    path: "/account",
    render: function render() {
      return /*#__PURE__*/React.createElement(AdminLayout, null, /*#__PURE__*/React.createElement(Helmet, {
        title: "User Account"
      }), /*#__PURE__*/React.createElement(Account, null));
    }
  })
}];
export default plugins;
