// @webiny/app-security/admin/plugins/scopesList.js
import _taggedTemplateLiteral from "@babel/runtime/helpers/taggedTemplateLiteral";

function _templateObject6() {
  var data = _taggedTemplateLiteral(["Allows CRUD operations on all users."]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = _taggedTemplateLiteral(["Security users CRUD"]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _taggedTemplateLiteral(["Allows CRUD operations on all roles."]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _taggedTemplateLiteral(["Security roles CRUD"]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteral(["Allows CRUD operations on all groups."]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteral(["Security groups CRUD"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

import { i18n } from "@webiny/app/i18n";
var t = i18n.ns("app-security/admin/scopesList");
export default [{
  name: "security-scopes-list-security",
  type: "security-scopes-list",
  scopes: [{
    scope: "security:group:crud",
    title: t(_templateObject()),
    description: t(_templateObject2())
  }, {
    scope: "security:role:crud",
    title: t(_templateObject3()),
    description: t(_templateObject4())
  }, {
    scope: "security:user:crud",
    title: t(_templateObject5()),
    description: t(_templateObject6())
  }]
}];
