// @webiny/app-page-builder/admin/components/EditorPluginsLoader.js
import _regeneratorRuntime from "@babel/runtime/regenerator";
import _asyncToGenerator from "@babel/runtime/helpers/asyncToGenerator";
import _objectSpread from "@babel/runtime/helpers/objectSpread2";
import _slicedToArray from "@babel/runtime/helpers/slicedToArray";
import React, { useEffect, useReducer } from "react";
import { registerPlugins } from "@webiny/plugins";
import { CircularProgress } from "@webiny/ui/Progress";
var globalState = {
  render: false,
  editor: false
};
export function EditorPluginsLoader(_ref) {
  var children = _ref.children,
      location = _ref.location;

  var _useReducer = useReducer(function (state, newState) {
    return _objectSpread(_objectSpread({}, state), newState);
  }, globalState),
      _useReducer2 = _slicedToArray(_useReducer, 2),
      loaded = _useReducer2[0],
      setLoaded = _useReducer2[1];

  function loadPlugins() {
    return _loadPlugins.apply(this, arguments);
  }

  function _loadPlugins() {
    _loadPlugins = _asyncToGenerator( /*#__PURE__*/_regeneratorRuntime.mark(function _callee() {
      var renderPlugins, plugins;
      return _regeneratorRuntime.wrap(function _callee$(_context) {
        while (1) {
          switch (_context.prev = _context.next) {
            case 0:
              if (!(location.pathname.startsWith("/page-builder/pages") && !loaded.render)) {
                _context.next = 7;
                break;
              }

              _context.next = 3;
              return import("@webiny/app-page-builder/render/presets/default");

            case 3:
              renderPlugins = _context.sent;
              registerPlugins(renderPlugins.default());
              globalState.render = true;
              setLoaded({
                render: true
              });

            case 7:
              if (!(location.pathname.startsWith("/page-builder/editor") && !loaded.editor)) {
                _context.next = 15;
                break;
              }

              _context.next = 10;
              return Promise.all([import("../../editor/presets/default"), !loaded.render ? import("@webiny/app-page-builder/render/presets/default") : null].filter(Boolean));

            case 10:
              plugins = _context.sent;
              registerPlugins(plugins.map(function (p) {
                return p.default();
              }));
              globalState.editor = true;
              globalState.render = true;
              setLoaded({
                editor: true,
                render: true
              });

            case 15:
            case "end":
              return _context.stop();
          }
        }
      }, _callee);
    }));
    return _loadPlugins.apply(this, arguments);
  }

  useEffect(function () {
    loadPlugins();
  }, []);

  if (location.pathname.startsWith("/page-builder/pages") && loaded.render) {
    return children;
  }

  if (location.pathname.startsWith("/page-builder/editor") && loaded.editor) {
    return children;
  }

  return /*#__PURE__*/React.createElement(CircularProgress, null);
}
