// @webiny/app-plugin-security-cognito-theme/site/index.js
import _objectWithoutProperties from "@babel/runtime/helpers/objectWithoutProperties";
import React from "react";
import SignInView from "./views/SignIn";
import ForgotPasswordView from "./views/ForgotPassword";
import RequireNewPasswordView from "./views/RequireNewPassword";
import SetNewPasswordView from "./views/SetNewPassword";
import { Loading, ViewContainer } from "./components/Layout";
export default (function (_ref) {
  var logo = _ref.logo;
  return [{
    name: "cognito-view-sign-in",
    type: "cognito-view",
    viewType: "signIn",
    view: function SignIn(_ref2) {
      var type = _ref2.type,
          props = _objectWithoutProperties(_ref2, ["type"]);

      if (type === "compact") {
        return /*#__PURE__*/React.createElement(SignInView, props);
      }

      return /*#__PURE__*/React.createElement(ViewContainer, {
        logo: logo
      }, /*#__PURE__*/React.createElement(SignInView, props));
    }
  }, {
    name: "cognito-view-forgot-password",
    type: "cognito-view",
    viewType: "forgotPassword",
    view: function ForgotPassword(_ref3) {
      var type = _ref3.type,
          props = _objectWithoutProperties(_ref3, ["type"]);

      if (type === "compact") {
        return /*#__PURE__*/React.createElement(ForgotPasswordView, props);
      }

      return /*#__PURE__*/React.createElement(ViewContainer, {
        logo: logo
      }, /*#__PURE__*/React.createElement(ForgotPasswordView, props));
    }
  }, {
    name: "cognito-view-require-new-password",
    type: "cognito-view",
    viewType: "requireNewPassword",
    view: function RequireNewPassword(_ref4) {
      var type = _ref4.type,
          props = _objectWithoutProperties(_ref4, ["type"]);

      if (type === "compact") {
        return /*#__PURE__*/React.createElement(RequireNewPasswordView, props);
      }

      return /*#__PURE__*/React.createElement(ViewContainer, {
        logo: logo
      }, /*#__PURE__*/React.createElement(RequireNewPasswordView, props));
    }
  }, {
    name: "cognito-view-set-new-password",
    type: "cognito-view",
    viewType: "setNewPassword",
    view: function SetNewPassword(_ref5) {
      var type = _ref5.type,
          props = _objectWithoutProperties(_ref5, ["type"]);

      if (type === "compact") {
        return /*#__PURE__*/React.createElement(SetNewPasswordView, props);
      }

      return /*#__PURE__*/React.createElement(ViewContainer, {
        logo: logo
      }, /*#__PURE__*/React.createElement(SetNewPasswordView, props));
    }
  }, {
    name: "cognito-view-loading",
    type: "cognito-view",
    viewType: "loading",
    view: Loading
  }];
});
