// @webiny/app-page-builder/site/components/Page/PageRender.js
import _objectSpread from "@babel/runtime/helpers/objectSpread2";
import _get from "lodash/get";
import React, { useMemo } from "react";
import { Helmet } from "react-helmet";
import Layout from "./Layout";
import PageContent from "./PageContent";
import { getPlugin, getPlugins } from "@webiny/plugins";

function PageRender(_ref) {
  var _data$settings, _data$settings$genera;

  var loading = _ref.loading,
      data = _ref.data,
      error = _ref.error;
  var Loader = useMemo(function () {
    var plugins = getPlugins("pb-layout-component");
    var pl = plugins.find(function (pl) {
      return pl.componentType === "loader";
    });
    return pl ? pl.component : null;
  }, []);

  var _useMemo = useMemo(function () {
    var defaultErrorPagePlugin = getPlugin("pb-default-page-error");
    var defaultNotFoundPagePlugin = getPlugin("pb-default-page-not-found");
    return {
      DefaultNotFoundPage: defaultNotFoundPagePlugin === null || defaultNotFoundPagePlugin === void 0 ? void 0 : defaultNotFoundPagePlugin.component,
      DefaultErrorPage: defaultErrorPagePlugin === null || defaultErrorPagePlugin === void 0 ? void 0 : defaultErrorPagePlugin.component
    };
  }, []),
      DefaultErrorPage = _useMemo.DefaultErrorPage,
      DefaultNotFoundPage = _useMemo.DefaultNotFoundPage;

  if (loading) {
    return Loader ? /*#__PURE__*/React.createElement(Loader, null) : null;
  }

  if (!data) {
    if ((error === null || error === void 0 ? void 0 : error.code) === "NOT_FOUND") {
      return DefaultNotFoundPage ? /*#__PURE__*/React.createElement(DefaultNotFoundPage, {
        error: error
      }) : null;
    }

    return DefaultErrorPage ? /*#__PURE__*/React.createElement(DefaultErrorPage, {
      error: error
    }) : null;
  }

  var head = {
    title: data.title,
    seo: _objectSpread({
      title: "",
      description: "",
      meta: []
    }, _get(data, "settings.seo")),
    social: _objectSpread({
      title: "",
      description: "",
      image: null
    }, _get(data, "settings.social"))
  };
  return /*#__PURE__*/React.createElement("div", {
    className: "webiny-pb-page"
  }, /*#__PURE__*/React.createElement(Helmet, null, head.title && /*#__PURE__*/React.createElement("title", null, head.title), head.seo.title && /*#__PURE__*/React.createElement("meta", {
    name: "title",
    content: head.seo.title
  }), head.seo.description && /*#__PURE__*/React.createElement("meta", {
    name: "description",
    content: head.seo.description
  }), head.seo.meta.map(function (_ref2, index) {
    var name = _ref2.name,
        content = _ref2.content;
    return /*#__PURE__*/React.createElement("meta", {
      key: index,
      name: name,
      content: content
    });
  }), head.social.image && /*#__PURE__*/React.createElement("meta", {
    property: "og:image",
    content: head.social.image.src + "?width=1596"
  }), head.social.title && /*#__PURE__*/React.createElement("meta", {
    property: "og:title",
    content: head.social.title
  }), head.social.description && /*#__PURE__*/React.createElement("meta", {
    property: "og:description",
    content: head.social.description
  }), head.social.meta.map(function (_ref3, index) {
    var property = _ref3.property,
        content = _ref3.content;
    // Replace duplicate "og:og:" with single "og:".
    var preparedProperty = "og:".concat(property).replace("og:og:", "og:");
    return /*#__PURE__*/React.createElement("meta", {
      key: index,
      property: preparedProperty,
      content: content
    });
  })), /*#__PURE__*/React.createElement(Layout, {
    layout: data === null || data === void 0 ? void 0 : (_data$settings = data.settings) === null || _data$settings === void 0 ? void 0 : (_data$settings$genera = _data$settings.general) === null || _data$settings$genera === void 0 ? void 0 : _data$settings$genera.layout
  }, /*#__PURE__*/React.createElement(PageContent, {
    data: data.content
  })));
}

export default PageRender;
