// @webiny/app-page-builder/render/plugins/elementSettings/align/index.js
import _objectSpread from "@babel/runtime/helpers/objectSpread2";
import _get from "lodash/get";
var vertical = {
  start: "flex-start",
  center: "center",
  end: "flex-end"
};
var plugins = [{
  name: "pb-render-page-element-style-horizontal-align",
  type: "pb-render-page-element-style",
  renderStyle: function renderStyle(_ref) {
    var element = _ref.element,
        style = _ref.style;

    var _get2 = _get(element, "data.settings", {}),
        horizontalAlign = _get2.horizontalAlign;

    if (!horizontalAlign) {
      return style;
    }

    return _objectSpread(_objectSpread({}, style), {}, {
      textAlign: horizontalAlign
    });
  }
}, {
  name: "pb-render-page-element-style-horizontal-align-flex",
  type: "pb-render-page-element-style",
  renderStyle: function renderStyle(_ref2) {
    var element = _ref2.element,
        style = _ref2.style;

    var _get3 = _get(element, "data.settings", {}),
        horizontalAlignFlex = _get3.horizontalAlignFlex;

    if (!horizontalAlignFlex) {
      return style;
    }

    return _objectSpread(_objectSpread({}, style), {}, {
      alignItems: horizontalAlignFlex
    });
  }
}, {
  name: "pb-render-page-element-style-vertical-align",
  type: "pb-render-page-element-style",
  renderStyle: function renderStyle(_ref3) {
    var element = _ref3.element,
        style = _ref3.style;

    var _get4 = _get(element, "data.settings", {}),
        verticalAlign = _get4.verticalAlign;

    if (!verticalAlign) {
      return style;
    }

    return _objectSpread(_objectSpread({}, style), {}, {
      justifyContent: vertical[verticalAlign]
    });
  }
}];
export default plugins;
