// @webiny/app-headless-cms/admin/plugins/contentForm/header/revisionSelector/RevisionSelector.js
import _get from "lodash/get";
import React from "react";
import { css } from "emotion";
import { useRouter } from "@webiny/react-router";
import { ButtonDefault } from "@webiny/ui/Button";
import { Typography } from "@webiny/ui/Typography";
import { Menu, MenuItem } from "@webiny/ui/Menu";
var buttonStyle = /*#__PURE__*/css({
  "&.mdc-button": {
    color: "var(--mdc-theme-text-primary-on-background) !important"
  }
}, "label:buttonStyle;");
var menuList = /*#__PURE__*/css({
  ".mdc-list-item": {
    flexDirection: "column",
    justifyContent: "center",
    alignItems: "baseline",
    textAlign: "left"
  }
}, "label:menuList;");

var RevisionSelector = function RevisionSelector(_ref) {
  var entry = _ref.entry;

  var _useRouter = useRouter(),
      location = _useRouter.location,
      history = _useRouter.history;

  var query = new URLSearchParams(location.search);
  return /*#__PURE__*/React.createElement(Menu, {
    className: menuList,
    onSelect: function onSelect(evt) {
      query.set("id", entry.revisions[evt.detail.index].id);
      history.push({
        search: query.toString()
      });
    },
    handle: /*#__PURE__*/React.createElement(ButtonDefault, {
      className: buttonStyle
    }, "v", _get(entry, "meta.version"))
  }, (_get(entry, "meta.revisions") || []).map(function (rev) {
    return /*#__PURE__*/React.createElement(MenuItem, {
      key: rev.id
    }, /*#__PURE__*/React.createElement(Typography, {
      use: "body2"
    }, "v", _get(rev, "meta.version")), /*#__PURE__*/React.createElement(Typography, {
      use: "caption"
    }, "(", _get(rev, "meta.status"), ")"));
  }));
};

export default RevisionSelector;
