// @webiny/app-form-builder/admin/viewsGraphql.js
import _taggedTemplateLiteral from "@babel/runtime/helpers/taggedTemplateLiteral";

function _templateObject10() {
  var data = _taggedTemplateLiteral(["\n    mutation DeleteForm($id: ID!) {\n        forms {\n            deleteForm(id: $id) {\n                data\n                error {\n                    ", "\n                }\n            }\n        }\n    }\n"]);

  _templateObject10 = function _templateObject10() {
    return data;
  };

  return data;
}

function _templateObject9() {
  var data = _taggedTemplateLiteral(["\n    mutation FormsDeleteRevision($id: ID!) {\n        forms {\n            deleteRevision(id: $id) {\n                data\n                error {\n                    ", "\n                }\n            }\n        }\n    }\n"]);

  _templateObject9 = function _templateObject9() {
    return data;
  };

  return data;
}

function _templateObject8() {
  var data = _taggedTemplateLiteral(["\n    mutation FormsUnpublishRevision($id: ID!) {\n        forms {\n            unpublishRevision(id: $id) {\n                data {\n                    id\n                    status\n                    published\n                }\n                error {\n                    ", "\n                }\n            }\n        }\n    }\n"]);

  _templateObject8 = function _templateObject8() {
    return data;
  };

  return data;
}

function _templateObject7() {
  var data = _taggedTemplateLiteral(["\n    mutation FormsPublishRevision($id: ID!) {\n        forms {\n            publishRevision(id: $id) {\n                data {\n                    id\n                    status\n                    published\n                }\n                error {\n                    ", "\n                }\n            }\n        }\n    }\n"]);

  _templateObject7 = function _templateObject7() {
    return data;
  };

  return data;
}

function _templateObject6() {
  var data = _taggedTemplateLiteral(["\n    mutation FormsCreateRevisionFrom($revision: ID!) {\n        forms {\n            revision: createRevisionFrom(revision: $revision) {\n                data {\n                    id\n                }\n                error {\n                    ", "\n                }\n            }\n        }\n    }\n"]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = _taggedTemplateLiteral(["\n    mutation FormsExportFormSubmissions($ids: [ID], $parent: ID, $form: ID) {\n        forms {\n            exportFormSubmissions(ids: $ids, parent: $parent, form: $form) {\n                data {\n                    src\n                }\n                error {\n                    message\n                }\n            }\n        }\n    }\n"]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _taggedTemplateLiteral(["\n    query FormsListFormSubmissions(\n        $sort: JSON\n        $search: String\n        $where: JSON\n        $limit: Int\n        $after: String\n        $before: String\n    ) {\n        forms {\n            listFormSubmissions(\n                sort: $sort\n                search: $search\n                where: $where\n                limit: $limit\n                after: $after\n                before: $before\n            ) {\n                data {\n                    id\n                    data\n                    meta {\n                        ip\n                        submittedOn\n                    }\n                    form {\n                        revision {\n                            id\n                            name\n                            version\n                            layout\n                            fields {\n                                _id\n                                fieldId\n                                type\n                                label {\n                                    value\n                                }\n                                options {\n                                    label {\n                                        value\n                                    }\n                                    value\n                                }\n                            }\n                        }\n                    }\n                }\n                meta {\n                    cursors {\n                        next\n                        previous\n                    }\n                    hasNextPage\n                    hasPreviousPage\n                    totalCount\n                }\n            }\n        }\n    }\n"]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _taggedTemplateLiteral(["\n    query FormsGetForm($id: ID!) {\n        forms {\n            form: getForm(id: $id) {\n                data {\n                    ", "\n                    overallStats {\n                        views\n                        submissions\n                        conversionRate\n                    }\n                    revisions {\n                        id\n                        savedOn\n                        parent\n                        name\n                        version\n                        status\n                        stats {\n                            views\n                            submissions\n                            conversionRate\n                        }\n                    }\n                }\n                error {\n                    ", "\n                }\n            }\n        }\n    }\n"]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteral(["\n    mutation FormsCreateForm($name: String!) {\n        forms {\n            form: createForm(data: { name: $name }) {\n                data {\n                    id\n                }\n                error {\n                    ", "\n                }\n            }\n        }\n    }\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteral(["\n    query FormsListForms($sort: JSON, $limit: Int, $search: String, $after: String, $before: String) {\n        forms {\n            listForms(sort: $sort, limit: $limit, search: $search, after: $after, before: $before) {\n                data {  \n                    ", "\n                }\n                meta {\n                    cursors {\n                        next\n                        previous\n                    }\n                    hasNextPage\n                    hasPreviousPage\n                    totalCount\n                }\n            }\n        }\n    }\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

import gql from "graphql-tag";
var ERROR_FIELDS = "\n    code\n    message\n    data\n";
var BASE_FORM_FIELDS = "  \n    id\n    name\n    version\n    parent\n    published\n    status\n    savedOn\n    createdBy {\n        firstName\n        lastName\n    }\n";
export var LIST_FORMS = gql(_templateObject(), BASE_FORM_FIELDS);
export var CREATE_FORM = gql(_templateObject2(), ERROR_FIELDS);
export var GET_FORM = gql(_templateObject3(), BASE_FORM_FIELDS, ERROR_FIELDS);
export var LIST_FORM_SUBMISSIONS = gql(_templateObject4());
export var EXPORT_FORM_SUBMISSIONS = gql(_templateObject5());
export var CREATE_REVISION_FROM = gql(_templateObject6(), ERROR_FIELDS);
export var PUBLISH_REVISION = gql(_templateObject7(), ERROR_FIELDS);
export var UNPUBLISH_REVISION = gql(_templateObject8(), ERROR_FIELDS);
export var DELETE_REVISION = gql(_templateObject9(), ERROR_FIELDS);
export var DELETE_FORM = gql(_templateObject10(), ERROR_FIELDS);
