// @webiny/app-security/admin/views/Roles/Roles.js
import * as React from "react";
import { SplitView, LeftPanel, RightPanel } from "@webiny/app-admin/components/SplitView";
import { FloatingActionButton } from "@webiny/app-admin/components/FloatingActionButton";
import { CrudProvider } from "@webiny/app-admin/contexts/Crud";
import RolesDataList from "./RolesDataList";
import RolesForm from "./RolesForm";
import { DELETE_ROLE, GET_ROLE, LIST_ROLES, CREATE_ROLE, UPDATE_ROLE } from "./graphql";

function Roles() {
  return /*#__PURE__*/React.createElement(CrudProvider, {
    read: GET_ROLE,
    delete: DELETE_ROLE,
    create: CREATE_ROLE,
    update: UPDATE_ROLE,
    list: {
      query: LIST_ROLES,
      variables: {
        sort: {
          savedOn: -1
        }
      }
    }
  }, function (_ref) {
    var actions = _ref.actions;
    return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(SplitView, null, /*#__PURE__*/React.createElement(LeftPanel, null, /*#__PURE__*/React.createElement(RolesDataList, null)), /*#__PURE__*/React.createElement(RightPanel, null, /*#__PURE__*/React.createElement(RolesForm, null))), /*#__PURE__*/React.createElement(FloatingActionButton, {
      "data-testid": "new-record-button",
      onClick: actions.resetForm
    }));
  });
}

export default Roles;
