// @webiny/app-page-builder/editor/plugins/elements/media/vimeo/index.js
import React from "react";
import styled from "@emotion/styled";
import { Tab } from "@webiny/ui/Tabs";
import { Input } from "@webiny/ui/Input";
import { Grid, Cell } from "@webiny/ui/Grid";
import { createEmbedPlugin, createEmbedSettingsPlugin } from "./../../utils/oembed/createEmbedPlugin";
import { ReactComponent as MediaIcon } from "@svgr/webpack!./../../../elementGroups/media/round-music_video-24px.svg";
import VimeoEmbed from "./VimeoEmbed";
import { validation } from "@webiny/validation";
import { ReactComponent as LogoIcon } from "@svgr/webpack!./vimeo-v-brands.svg";
var PreviewBox = /*#__PURE__*/styled("div", {
  target: "e10gswqi0",
  label: "PreviewBox"
})({
  textAlign: "center",
  height: 50,
  svg: {
    height: 50,
    width: 50
  }
});
export default (function () {
  return [createEmbedPlugin({
    type: "vimeo",
    toolbar: {
      title: "Vimeo",
      group: "pb-editor-element-group-media",
      preview: function preview() {
        return /*#__PURE__*/React.createElement(PreviewBox, null, /*#__PURE__*/React.createElement(LogoIcon, null));
      }
    },
    oembed: {
      renderEmbed: function renderEmbed(props) {
        return /*#__PURE__*/React.createElement(VimeoEmbed, props);
      }
    },
    renderElementPreview: function renderElementPreview() {
      return /*#__PURE__*/React.createElement("div", null, "Vimeo video");
    }
  }), createEmbedSettingsPlugin({
    type: "vimeo",
    render: function render(_ref) {
      var Bind = _ref.Bind;
      return /*#__PURE__*/React.createElement(Tab, {
        icon: /*#__PURE__*/React.createElement(MediaIcon, null),
        label: "Vimeo"
      }, /*#__PURE__*/React.createElement(Grid, null, /*#__PURE__*/React.createElement(Cell, {
        span: 12
      }, /*#__PURE__*/React.createElement(Bind, {
        name: "source.url",
        validators: validation.create("required,url")
      }, /*#__PURE__*/React.createElement(Input, {
        label: "Video URL",
        description: "Enter a video URL"
      })))));
    }
  })];
});
