// @webiny/app-template-admin/index.js
import _toConsumableArray from "@babel/runtime/helpers/toConsumableArray";
import React from "react";
import { createTemplate } from "@webiny/app-template";
import { ApolloProvider } from "react-apollo"; // App structure imports

import { UiProvider } from "@webiny/app/contexts/Ui";
import { I18NProvider } from "@webiny/app-i18n/contexts/I18N";
import { SecurityProvider } from "@webiny/app-security/contexts/Security";
import { CircularProgress } from "@webiny/ui/Progress";
import { AppInstaller } from "@webiny/app-admin/components/Install/AppInstaller";
import { ThemeProvider } from "@webiny/app-admin/contexts/Theme"; // Other plugins

import { fileUploadPlugin, imagePlugin } from "@webiny/app/plugins";
import adminPlugins from "@webiny/app-admin/plugins";
import welcomeScreenPlugins from "@webiny/app-plugin-admin-welcome-screen";
import i18nPlugins from "@webiny/app-i18n/admin/plugins";
import securityPlugins from "@webiny/app-security/admin/plugins";
import cognito from "@webiny/app-plugin-security-cognito";
import cognitoTheme from "@webiny/app-plugin-security-cognito-theme/admin";
import fileManagerPlugins from "@webiny/app-file-manager/admin"; // ApolloClient

import { createApolloClient } from "./apolloClient";
import { NetworkError } from "./apolloClient/NetworkError"; // Router

import { BrowserRouter, Route, Redirect } from "@webiny/react-router";
export default createTemplate(function (opts) {
  var appStructure = [{
    type: "app-template-renderer",
    name: "app-template-renderer-apollo",
    render: function render(children) {
      return /*#__PURE__*/React.createElement(ApolloProvider, {
        client: createApolloClient()
      }, /*#__PURE__*/React.createElement(NetworkError, null, children));
    }
  }, {
    type: "app-template-renderer",
    name: "app-template-renderer-router",
    render: function render(children) {
      return /*#__PURE__*/React.createElement(BrowserRouter, {
        basename: process.env.PUBLIC_URL
      }, children, /*#__PURE__*/React.createElement(Route, {
        exact: true,
        path: "/",
        render: function render() {
          return /*#__PURE__*/React.createElement(Redirect, {
            to: opts.defaultRoute || "/"
          });
        }
      }));
    }
  }, {
    type: "app-template-renderer",
    name: "app-template-renderer-ui",
    render: function render(children) {
      return /*#__PURE__*/React.createElement(UiProvider, null, children);
    }
  }, {
    type: "app-template-renderer",
    name: "app-template-renderer-i18n",
    render: function render(children) {
      return /*#__PURE__*/React.createElement(I18NProvider, {
        loader: /*#__PURE__*/React.createElement(CircularProgress, {
          label: "Loading locales..."
        })
      }, children);
    }
  }, {
    type: "app-template-renderer",
    name: "app-template-renderer-app-installer",
    render: function render(children) {
      var securityProvider = /*#__PURE__*/React.createElement(SecurityProvider, {
        loader: /*#__PURE__*/React.createElement(CircularProgress, {
          label: "Checking user..."
        })
      });
      return /*#__PURE__*/React.createElement(AppInstaller, {
        security: securityProvider
      }, children);
    }
  }, {
    type: "app-template-renderer",
    name: "app-template-renderer-admin-theme",
    render: function render(children) {
      return /*#__PURE__*/React.createElement(ThemeProvider, null, children);
    }
  }];
  var otherPlugins = [welcomeScreenPlugins(), fileUploadPlugin(), imagePlugin(), adminPlugins(), i18nPlugins(), securityPlugins(), cognito(opts.cognito), cognitoTheme(), fileManagerPlugins()].concat(_toConsumableArray(opts.plugins || []));
  return [].concat(appStructure, _toConsumableArray(otherPlugins));
});
