// @webiny/app-page-builder/editor/plugins/elementSettings/align/index.js
import React from "react";
import Action from "../components/Action";
import HorizontalAlignAction from "./HorizontalAlignAction";
import HorizontalAlignFlexAction from "./HorizontalAlignFlex";
import VerticalAlignAction from "./VerticalAlignAction";
export default [{
  name: "pb-editor-page-element-settings-horizontal-align",
  type: "pb-editor-page-element-settings",
  renderAction: function renderAction(_ref) {
    var options = _ref.options;
    return /*#__PURE__*/React.createElement(HorizontalAlignAction, {
      options: options
    }, /*#__PURE__*/React.createElement(Action, {
      plugin: this.name,
      tooltip: "Horizontal align"
    }));
  }
}, {
  name: "pb-editor-page-element-settings-horizontal-align-flex",
  type: "pb-editor-page-element-settings",
  renderAction: function renderAction() {
    return /*#__PURE__*/React.createElement(HorizontalAlignFlexAction, null, /*#__PURE__*/React.createElement(Action, {
      plugin: this.name,
      tooltip: "Horizontal align"
    }));
  }
}, {
  name: "pb-editor-page-element-settings-vertical-align",
  type: "pb-editor-page-element-settings",
  renderAction: function renderAction() {
    return /*#__PURE__*/React.createElement(VerticalAlignAction, null, /*#__PURE__*/React.createElement(Action, {
      plugin: this.name,
      tooltip: "Vertical align"
    }));
  }
}];
