// @webiny/app-page-builder/admin/views/Pages/PagesDataList.js
import _taggedTemplateLiteral from "@babel/runtime/helpers/taggedTemplateLiteral";

function _templateObject() {
  var data = _taggedTemplateLiteral(["Pages"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

import * as React from "react";
import TimeAgo from "timeago-react";
import { useRouter } from "@webiny/react-router";
import { i18n } from "@webiny/app/i18n";
import { css } from "emotion";
import { Typography } from "@webiny/ui/Typography";
import { DataList, List, ListItem, ListItemText, ListItemTextSecondary, ListTextOverline, ListItemMeta } from "@webiny/ui/List";
var t = i18n.namespace("Pb.PagesDataList");
var rightAlign = /*#__PURE__*/css({
  alignItems: "flex-end !important"
}, "label:rightAlign;");

var PagesDataList = function PagesDataList(props) {
  var _useRouter = useRouter(),
      location = _useRouter.location,
      history = _useRouter.history;

  var dataList = props.dataList;
  var query = new URLSearchParams(location.search);
  return /*#__PURE__*/React.createElement(DataList, Object.assign({}, dataList, {
    title: t(_templateObject()),
    sorters: [{
      label: "Newest to oldest",
      sorters: {
        createdOn: -1
      }
    }, {
      label: "Oldest to newest",
      sorters: {
        createdOn: 1
      }
    }, {
      label: "Title A-Z",
      sorters: {
        title: 1
      }
    }, {
      label: "Title Z-A",
      sorters: {
        title: -1
      }
    }]
  }), function (_ref) {
    var _ref$data = _ref.data,
        data = _ref$data === void 0 ? [] : _ref$data;
    return /*#__PURE__*/React.createElement(List, {
      "data-testid": "default-data-list"
    }, data.map(function (page) {
      return /*#__PURE__*/React.createElement(ListItem, {
        key: page.id
      }, /*#__PURE__*/React.createElement(ListItemText, {
        onClick: function onClick() {
          query.set("id", page.id);
          history.push({
            search: query.toString()
          });
        }
      }, page.title, /*#__PURE__*/React.createElement(ListTextOverline, null, page.category.name), page.createdBy && /*#__PURE__*/React.createElement(ListItemTextSecondary, null, "Created by: ", page.createdBy.firstName || "N/A", ". Last modified: ", /*#__PURE__*/React.createElement(TimeAgo, {
        datetime: page.savedOn
      }), ".")), /*#__PURE__*/React.createElement(ListItemMeta, {
        className: rightAlign
      }, /*#__PURE__*/React.createElement(Typography, {
        use: "subtitle2"
      }, page.locked ? "Published" : "Draft", " (v", page.version, ")")));
    }));
  });
};

export default PagesDataList;
