// @webiny/app-page-builder/editor/plugins/slate/block/Styled.js
import styled from "@emotion/styled";
import { css } from "emotion";
export var Item = /*#__PURE__*/styled("div", {
  target: "e17cyl550",
  label: "Item"
})(function (_ref) {
  var isActive = _ref.isActive,
      isSelected = _ref.isSelected;
  return {
    cursor: "pointer",
    display: "block",
    border: "none",
    height: "auto",
    textAlign: "left",
    borderTop: "none",
    lineHeight: "1em",
    color: "var(--mdc-theme-text-primary-on-background)",
    fontSize: "1rem",
    textTransform: "none",
    boxShadow: "none",
    padding: ".8rem 1.1rem",
    boxSizing: "border-box",
    whiteSpace: "nowrap",
    wordWrap: "normal",
    backgroundColor: isActive ? "var(--mdc-theme-on-background)" : "var(--mdc-theme-surface)",
    fontWeight: isSelected ? "bold" : "normal",
    "&:hover, &:focus": {
      borderColor: "#96c8da",
      boxShadow: "0 2px 3px 0 rgba(34,36,38,.15)"
    }
  };
});
export var List = /*#__PURE__*/styled("div", {
  target: "e17cyl551",
  label: "List"
})({
  border: "var(--mdc-theme-on-background)",
  backgroundColor: "var(--mdc-theme-surface)",
  position: "relative",
  left: 0,
  top: 0,
  height: "200px",
  width: "100%",
  overflow: "scroll"
});
export var dropDownDialog = /*#__PURE__*/css({
  position: "absolute",
  zIndex: 2,
  top: 27,
  left: -5,
  width: 200
}, "label:dropDownDialog;");
export var Button = /*#__PURE__*/styled("div", {
  target: "e17cyl552",
  label: "Button"
})({
  padding: "1px 3px 3px 3px",
  backgroundColor: "var(--mdc-theme-on-background)",
  fontSize: "0.8em",
  whiteSpace: "nowrap",
  lineHeight: "110%"
});
