// @webiny/app-headless-cms/admin/views/ContentModelGroups/ContentModelGroups.js
import React from "react";
import { SplitView, LeftPanel, RightPanel } from "@webiny/app-admin/components/SplitView";
import { FloatingActionButton } from "@webiny/app-admin/components/FloatingActionButton";
import ContentModelGroupsDataList from "./ContentModelGroupsDataList";
import ContentModelGroupsForm from "./ContentModelGroupsForm";
import { CrudProvider } from "@webiny/app-admin/contexts/Crud";
import { GET_CONTENT_MODEL_GROUP, LIST_CONTENT_MODEL_GROUPS, CREATE_CONTENT_MODEL_GROUP, UPDATE_CONTENT_MODEL_GROUP, DELETE_CONTENT_MODEL_GROUP } from "./graphql";
import { useCms } from "@webiny/app-headless-cms/admin/hooks";

function ContentModelGroups() {
  var _useCms = useCms(),
      apolloClient = _useCms.environments.apolloClient;

  return /*#__PURE__*/React.createElement(CrudProvider, {
    delete: {
      mutation: DELETE_CONTENT_MODEL_GROUP,
      options: {
        client: apolloClient,
        refetchQueries: ["HeadlessCmsListMenuContentGroupsModels"]
      }
    },
    read: {
      query: GET_CONTENT_MODEL_GROUP,
      options: {
        client: apolloClient
      }
    },
    create: {
      mutation: CREATE_CONTENT_MODEL_GROUP,
      options: {
        client: apolloClient,
        refetchQueries: ["HeadlessCmsListMenuContentGroupsModels"]
      }
    },
    update: {
      mutation: UPDATE_CONTENT_MODEL_GROUP,
      options: {
        client: apolloClient
      }
    },
    list: {
      query: LIST_CONTENT_MODEL_GROUPS,
      variables: {
        sort: {
          savedOn: -1
        }
      },
      options: {
        client: apolloClient
      }
    }
  }, function (_ref) {
    var actions = _ref.actions;
    return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(SplitView, null, /*#__PURE__*/React.createElement(LeftPanel, {
      span: 4
    }, /*#__PURE__*/React.createElement(ContentModelGroupsDataList, null)), /*#__PURE__*/React.createElement(RightPanel, {
      span: 8
    }, /*#__PURE__*/React.createElement(ContentModelGroupsForm, null))), /*#__PURE__*/React.createElement(FloatingActionButton, {
      "data-testid": "new-record-button",
      onClick: actions.resetForm
    }));
  });
}

export default ContentModelGroups;
