// @webiny/app-i18n/admin/plugins/richTextEditor/code/index.js
import * as React from "react";
import { css } from "emotion";
import { isKeyHotkey } from "is-hotkey";
import { ReactComponent as FormatCodeIcon } from "@svgr/webpack!@webiny/app-i18n/admin/assets/icons/format_code.svg";
var isCodeHotkey = isKeyHotkey("mod+`");
var mark = "code";
/* prettier-ignore */

var codeStyle = /*#__PURE__*/css("background-color:var(--mdc-theme-background);padding:4px 8px;line-height:1.5rem;label:codeStyle;");
var plugin = {
  name: "i18n-input-rich-text-editor-code",
  type: "i18n-input-rich-text-editor",
  plugin: {
    name: "code",
    menu: {
      render: function render(_ref) {
        var MenuButton = _ref.MenuButton,
            editor = _ref.editor;
        return /*#__PURE__*/React.createElement(MenuButton, {
          onClick: function onClick() {
            return editor.toggleMark(mark);
          },
          active: editor.hasMark(mark)
        }, /*#__PURE__*/React.createElement(FormatCodeIcon, null));
      }
    },
    editor: {
      onKeyDown: function onKeyDown(_ref2, next) {
        var event = _ref2.event,
            editor = _ref2.editor;

        // Decide what to do based on the key code...
        if (isCodeHotkey(event)) {
          event.preventDefault();
          editor.toggleMark(mark);
          return true;
        }

        return next();
      },
      renderLeaf: function renderLeaf(_ref3) {
        var leaf = _ref3.leaf,
            attributes = _ref3.attributes,
            children = _ref3.children;

        if (leaf[mark] === true) {
          return /*#__PURE__*/React.createElement("code", Object.assign({
            className: "webiny-rich-typography-code"
          }, attributes), children);
        }

        return children;
      }
    }
  }
};
export default plugin;
