// @webiny/app-headless-cms/admin/plugins/contentDetails/contentForm/index.js
import * as React from "react";
import { renderPlugins } from "@webiny/app/plugins";
import { Tab } from "@webiny/ui/Tabs";
import styled from "@emotion/styled";
import { Elevation } from "@webiny/ui/Elevation";
import ContentForm from "./ContentForm";
import { CircularProgress } from "@webiny/ui/Progress";
import { css } from "emotion";
var RenderBlock = /*#__PURE__*/styled("div", {
  target: "e16b75ce0",
  label: "RenderBlock"
})({
  position: "relative",
  zIndex: 0,
  backgroundColor: "var(--mdc-theme-background)",
  height: "100%",
  overflow: "scroll",
  padding: 25
});
var elevationStyles = /*#__PURE__*/css({
  position: "relative"
}, "label:elevationStyles;");
var plugins = [{
  name: "cms-content-details-revision-content-preview",
  type: "cms-content-details-revision-content",
  render: function render(props) {
    return /*#__PURE__*/React.createElement(Tab, {
      label: "Content",
      disabled: props.getLoading()
    }, /*#__PURE__*/React.createElement(RenderBlock, null, /*#__PURE__*/React.createElement(Elevation, {
      z: 2,
      className: elevationStyles
    }, props.getLoading() && /*#__PURE__*/React.createElement(CircularProgress, null), renderPlugins("cms-content-details-revision-content-preview", props))));
  }
}, {
  name: "cms-content-details-content-form",
  type: "cms-content-details-revision-content-preview",
  render: function render(props) {
    return /*#__PURE__*/React.createElement(ContentForm, props);
  }
}];
export default plugins;
