// @webiny/app-admin/plugins/Menu/Navigation/index.js
import _taggedTemplateLiteral from "@babel/runtime/helpers/taggedTemplateLiteral";
import _sortBy from "lodash/sortBy";

function _templateObject5() {
  var data = _taggedTemplateLiteral(["\xA9 {year} Webiny Ltd, London, UK"]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _taggedTemplateLiteral(["Webiny.com"]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _taggedTemplateLiteral(["Source"]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteral(["Community"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteral(["Documentation"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

import React, { useMemo, useEffect } from "react";
import { Drawer, DrawerContent, DrawerHeader } from "@webiny/ui/Drawer";
import { List, ListItem, ListItemGraphic } from "@webiny/ui/List";
import { IconButton } from "@webiny/ui/Button";
import { Icon } from "@webiny/ui/Icon";
import { getPlugin, getPlugins } from "@webiny/plugins";
import { useNavigation, Menu, Item, Section } from "./components";
import { logoStyle, MenuFooter, MenuHeader, navContent, navHeader, subFooter } from "./Styled";
import { ReactComponent as MenuIcon } from "@svgr/webpack!@webiny/app-admin/assets/icons/baseline-menu-24px.svg";
import { ReactComponent as DocsIcon } from "@svgr/webpack!@webiny/app-admin/assets/icons/icon-documentation.svg";
import { ReactComponent as CommunityIcon } from "@svgr/webpack!@webiny/app-admin/assets/icons/icon-community.svg";
import { ReactComponent as GithubIcon } from "@svgr/webpack!@webiny/app-admin/assets/icons/github-brands.svg";
import { i18n } from "@webiny/app/i18n";
var t = i18n.ns("app-admin/navigation");

var Navigation = function Navigation() {
  var _useNavigation = useNavigation(),
      hideMenu = _useNavigation.hideMenu,
      menuIsShown = _useNavigation.menuIsShown,
      initSections = _useNavigation.initSections;

  useEffect(initSections, []);
  var logo = useMemo(function () {
    var logoPlugin = getPlugin("admin-header-logo");

    if (logoPlugin) {
      return /*#__PURE__*/React.cloneElement(logoPlugin.render(), {
        className: logoStyle
      });
    }

    return null;
  }, []);
  var menus = [];
  var menuPlugins = getPlugins("admin-menu"); // First we sort by order (default: 50), and then by plugin name. In other words, if order isn't defined,
  // then we just sort by plugin name.

  menuPlugins && _sortBy(menuPlugins, [function (p) {
    return p.order || 50;
  }, function (p) {
    return p.name;
  }]).forEach(function (plugin) {
    menus.push( /*#__PURE__*/React.createElement(React.Fragment, {
      key: plugin.name
    }, plugin.render({
      Menu: Menu,
      Section: Section,
      Item: Item
    })));
  });
  return /*#__PURE__*/React.createElement(Drawer, {
    modal: true,
    open: menuIsShown(),
    onClose: hideMenu
  }, /*#__PURE__*/React.createElement(DrawerHeader, {
    className: navHeader
  }, /*#__PURE__*/React.createElement(MenuHeader, null, /*#__PURE__*/React.createElement(IconButton, {
    icon: /*#__PURE__*/React.createElement(MenuIcon, null),
    onClick: hideMenu
  }), logo)), /*#__PURE__*/React.createElement(DrawerContent, {
    className: navContent
  }, menus), /*#__PURE__*/React.createElement(MenuFooter, null, /*#__PURE__*/React.createElement(List, {
    nonInteractive: true
  }, /*#__PURE__*/React.createElement("a", {
    href: "https://docs.webiny.com/",
    rel: "noopener noreferrer",
    target: "_blank"
  }, /*#__PURE__*/React.createElement(ListItem, {
    ripple: false
  }, /*#__PURE__*/React.createElement(ListItemGraphic, null, /*#__PURE__*/React.createElement(Icon, {
    icon: /*#__PURE__*/React.createElement(DocsIcon, null)
  })), t(_templateObject()))), /*#__PURE__*/React.createElement("a", {
    href: "https://community.webiny.com/",
    rel: "noopener noreferrer",
    target: "_blank"
  }, /*#__PURE__*/React.createElement(ListItem, {
    ripple: false
  }, /*#__PURE__*/React.createElement(ListItemGraphic, null, /*#__PURE__*/React.createElement(Icon, {
    icon: /*#__PURE__*/React.createElement(CommunityIcon, null)
  })), t(_templateObject2()))), /*#__PURE__*/React.createElement("a", {
    href: "https://github.com/webiny/webiny-js",
    rel: "noopener noreferrer",
    target: "_blank"
  }, /*#__PURE__*/React.createElement(ListItem, {
    ripple: false
  }, /*#__PURE__*/React.createElement(ListItemGraphic, null, /*#__PURE__*/React.createElement(Icon, {
    icon: /*#__PURE__*/React.createElement(GithubIcon, null)
  })), t(_templateObject3()))), /*#__PURE__*/React.createElement(ListItem, {
    ripple: false,
    className: subFooter
  }, /*#__PURE__*/React.createElement("div", null, /*#__PURE__*/React.createElement("a", {
    href: "https://www.webiny.com/",
    rel: "noopener noreferrer",
    target: "_blank"
  }, t(_templateObject4())), /*#__PURE__*/React.createElement("p", null, t(_templateObject5())({
    year: new Date().getFullYear()
  })))))));
};

export default Navigation;
