// @webiny/app-page-builder/admin/plugins/pageDetails/header/revisionSelector/RevisionSelector.js
import _get from "lodash/get";
import React from "react";
import { css } from "emotion";
import { useRouter } from "@webiny/react-router";
import { ButtonDefault } from "@webiny/ui/Button";
import { Icon } from "@webiny/ui/Icon";
import { ReactComponent as DownButton } from "@svgr/webpack!@webiny/app-page-builder/admin/assets/round-arrow_drop_down-24px.svg";
import { MenuItem } from "@rmwc/menu";
import { Typography } from "@webiny/ui/Typography";
import { Menu } from "@webiny/ui/Menu";
import { usePageDetails } from "@webiny/app-page-builder/admin/hooks/usePageDetails";
var buttonStyle = /*#__PURE__*/css({
  "&.mdc-button": {
    color: "var(--mdc-theme-text-primary-on-background) !important"
  }
}, "label:buttonStyle;");
var menuList = /*#__PURE__*/css({
  ".mdc-list-item": {
    flexDirection: "column",
    justifyContent: "center",
    alignItems: "baseline",
    textAlign: "left"
  }
}, "label:menuList;");

var RevisionSelector = function RevisionSelector() {
  var _usePageDetails = usePageDetails(),
      page = _usePageDetails.page;

  var _useRouter = useRouter(),
      location = _useRouter.location,
      history = _useRouter.history;

  var query = new URLSearchParams(location.search);
  return /*#__PURE__*/React.createElement(Menu, {
    className: menuList,
    onSelect: function onSelect(evt) {
      query.set("id", page.revisions[evt.detail.index].id);
      history.push({
        search: query.toString()
      });
    },
    handle: /*#__PURE__*/React.createElement(ButtonDefault, {
      className: buttonStyle
    }, "v", page.version, " ", /*#__PURE__*/React.createElement(Icon, {
      icon: /*#__PURE__*/React.createElement(DownButton, null)
    }))
  }, (_get(page, "revisions") || []).map(function (rev) {
    var status = "draft";

    if (rev.published) {
      status = "published";
    }

    if (rev.locked && !rev.published) {
      status = "locked";
    }

    return /*#__PURE__*/React.createElement(MenuItem, {
      key: rev.id
    }, /*#__PURE__*/React.createElement(Typography, {
      use: "body2"
    }, "v", rev.version), /*#__PURE__*/React.createElement(Typography, {
      use: "caption"
    }, "(", status, ")"));
  }));
};

export default RevisionSelector;
