// @webiny/app-form-builder/admin/plugins/fields/components/OptionsListComponents/EditFieldOptionDialog.js
import _taggedTemplateLiteral from "@babel/runtime/helpers/taggedTemplateLiteral";

function _templateObject4() {
  var data = _taggedTemplateLiteral(["Save"]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _taggedTemplateLiteral(["Value"]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteral(["Label"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteral(["Edit option"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

import React from "react";
import { css } from "emotion";
import { Form } from "@webiny/form";
import { Grid, Cell } from "@webiny/ui/Grid";
import { Input } from "@webiny/ui/Input";
import { I18NInput } from "@webiny/app-i18n/admin/components";
import { i18n } from "@webiny/app/i18n";
import { Hotkeys } from "react-hotkeyz";
import { validation } from "@webiny/validation";
import { Dialog, DialogTitle, DialogContent, DialogActions, DialogButton } from "@webiny/ui/Dialog";
var t = i18n.namespace("Forms.FormEditor.EditFieldOptionDialog");
var narrowDialog = /*#__PURE__*/css({
  ".mdc-dialog__surface": {
    width: 600,
    minWidth: 600
  }
}, "label:narrowDialog;");

var EditFieldOptionDialog = function EditFieldOptionDialog(props) {
  var onClose = props.onClose,
      options = props.options,
      open = props.open,
      onSubmit = props.onSubmit,
      option = props.option,
      optionIndex = props.optionIndex;
  return /*#__PURE__*/React.createElement(Dialog, {
    open: open,
    onClose: onClose,
    className: narrowDialog
  }, option !== null && /*#__PURE__*/React.createElement(Hotkeys, {
    zIndex: 115,
    keys: {
      esc: function esc(e) {
        e.preventDefault();
        e.stopPropagation();
        onClose();
      }
    }
  }, /*#__PURE__*/React.createElement(Form, {
    data: option,
    onSubmit: onSubmit
  }, function (_ref) {
    var Bind = _ref.Bind,
        submit = _ref.submit;
    return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(DialogTitle, null, t(_templateObject())), /*#__PURE__*/React.createElement(DialogContent, null, /*#__PURE__*/React.createElement(Grid, null, /*#__PURE__*/React.createElement(Cell, {
      span: 12
    }, /*#__PURE__*/React.createElement(Bind, {
      name: "label"
    }, /*#__PURE__*/React.createElement(I18NInput, {
      label: t(_templateObject2()),
      showTranslateIcon: false
    }))), /*#__PURE__*/React.createElement(Cell, {
      span: 12
    }, /*#__PURE__*/React.createElement(Bind, {
      name: "value",
      validators: function validators(value) {
        validation.validateSync(value, "required");

        if (!Array.isArray(options)) {
          return true;
        }

        for (var i = 0; i < options.length; i++) {
          var current = options[i];

          if (current.value === value && i !== optionIndex) {
            throw new Error("Option with value \"".concat(value, "\" already exists."));
          }
        }
      }
    }, /*#__PURE__*/React.createElement(Input, {
      label: t(_templateObject3())
    }))))), /*#__PURE__*/React.createElement(DialogActions, null, /*#__PURE__*/React.createElement(DialogButton, {
      onClick: submit
    }, t(_templateObject4()))));
  })));
};

export default EditFieldOptionDialog;
