// @webiny/app-security/admin/views/Groups/GroupsDataList.js
import _taggedTemplateLiteral from "@babel/runtime/helpers/taggedTemplateLiteral";

function _templateObject5() {
  var data = _taggedTemplateLiteral(["Name Z-A"]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _taggedTemplateLiteral(["Name A-Z"]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _taggedTemplateLiteral(["Oldest to newest"]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteral(["Newest to oldest"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteral(["Security Groups"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

import React from "react";
import { i18n } from "@webiny/app/i18n";
import { ConfirmationDialog } from "@webiny/ui/ConfirmationDialog";
import { DataList, ScrollList, ListItem, ListItemText, ListItemTextSecondary, ListItemMeta, ListActions } from "@webiny/ui/List";
import { DeleteIcon } from "@webiny/ui/List/DataList/icons";
import { useCrud } from "@webiny/app-admin/hooks/useCrud";
var t = i18n.ns("app-security/admin/groups/data-list");

var GroupsDataList = function GroupsDataList() {
  var _useCrud = useCrud(),
      actions = _useCrud.actions,
      list = _useCrud.list;

  return /*#__PURE__*/React.createElement(DataList, Object.assign({}, list, {
    title: t(_templateObject()),
    sorters: [{
      label: t(_templateObject2()),
      sorters: {
        savedOn: -1
      }
    }, {
      label: t(_templateObject3()),
      sorters: {
        savedOn: 1
      }
    }, {
      label: t(_templateObject4()),
      sorters: {
        name: 1
      }
    }, {
      label: t(_templateObject5()),
      sorters: {
        name: -1
      }
    }]
  }), function (_ref) {
    var data = _ref.data,
        select = _ref.select,
        isSelected = _ref.isSelected;
    return /*#__PURE__*/React.createElement(ScrollList, {
      "data-testid": "default-data-list"
    }, data.map(function (item) {
      return /*#__PURE__*/React.createElement(ListItem, {
        key: item.id,
        selected: isSelected(item)
      }, /*#__PURE__*/React.createElement(ListItemText, {
        onClick: function onClick() {
          return select(item);
        }
      }, item.name, /*#__PURE__*/React.createElement(ListItemTextSecondary, null, item.description)), /*#__PURE__*/React.createElement(ListItemMeta, null, /*#__PURE__*/React.createElement(ListActions, null, /*#__PURE__*/React.createElement(ConfirmationDialog, null, function (_ref2) {
        var showConfirmation = _ref2.showConfirmation;
        return /*#__PURE__*/React.createElement(DeleteIcon, {
          onClick: function onClick() {
            return showConfirmation(function () {
              return actions.delete(item);
            });
          }
        });
      }))));
    }));
  });
};

export default GroupsDataList;
