// @webiny/app-page-builder/render/plugins/slate/link/index.js
import _defineProperty from "@babel/runtime/helpers/defineProperty";
import _objectSpread from "@babel/runtime/helpers/objectSpread2";
import React from "react";
import { Link } from "@webiny/react-router";
export default (function () {
  return {
    name: "pb-render-slate-editor-link",
    type: "pb-render-slate-editor",
    slate: {
      renderNode: function renderNode(props, next) {
        var attributes = props.attributes,
            children = props.children,
            node = props.node; // @ts-ignore

        if (node.type === "link") {
          // @ts-ignore
          var data = node.data;
          var href = data.get("href");
          var noFollow = data.get("noFollow");
          var newTab = data.get("newTab");
          var isInternal = href.startsWith("/");
          var LinkComponent = isInternal ? Link : "a";

          var linkProps = _objectSpread(_objectSpread({}, attributes), {}, _defineProperty({
            rel: noFollow ? "nofollow" : null,
            target: newTab ? "_blank" : "_self"
          }, isInternal ? "to" : "href", href));

          return /*#__PURE__*/React.createElement(LinkComponent, linkProps, children);
        }

        return next();
      }
    }
  };
});
