// @webiny/app-page-builder/editor/components/Slate/index.js
import Slate from "./Slate";
/**
 * Create Slate editor value object using given text and typography.
 *
 * @param text
 * @param typography
 * @returns {{object, document}}
 */

export var createValue = function createValue(text) {
  var typography = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : "body1";
  return createDocument(createTextLeaf(text), typography);
};
/**
 * Create Slate editor text leaf
 * @param text
 * @returns {{object: string, text: *}}
 */

var createTextLeaf = function createTextLeaf(text) {
  return {
    object: "leaf",
    text: text
  };
};
/**
 * Create Slate editor document
 * @param leaf
 * @param typography
 * @returns {{object: string, document: {object: string, nodes: *[]}}}
 */


var createDocument = function createDocument(leaf) {
  var typography = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : "body1";
  return {
    object: "value",
    document: {
      object: "document",
      nodes: [{
        object: "block",
        type: typography,
        nodes: [{
          object: "text",
          leaves: [leaf]
        }]
      }]
    }
  };
};

export default Slate;
