// @webiny/app-page-builder/admin/plugins/pageDetails/previewContent/index.js
import * as React from "react";
import { renderPlugins } from "@webiny/app/plugins";
import { Tab } from "@webiny/ui/Tabs";
import styled from "@emotion/styled";
import { Elevation } from "@webiny/ui/Elevation";
import PagePreview from "./PagePreview";
import { CircularProgress } from "@webiny/ui/Progress";
var RenderBlock = /*#__PURE__*/styled("div", {
  target: "e11br6ii0",
  label: "RenderBlock"
})({
  position: "relative",
  zIndex: 0,
  backgroundColor: "var(--mdc-theme-background)",
  height: "100%",
  overflow: "scroll",
  padding: 25
});
var plugins = [{
  name: "pb-page-details-revision-content-preview",
  type: "pb-page-details-revision-content",
  render: function render(_ref) {
    var pageDetails = _ref.pageDetails,
        loading = _ref.loading,
        refreshPages = _ref.refreshPages;
    return /*#__PURE__*/React.createElement(Tab, {
      label: "Page preview",
      disabled: loading
    }, /*#__PURE__*/React.createElement(RenderBlock, null, /*#__PURE__*/React.createElement(Elevation, {
      z: 2
    }, /*#__PURE__*/React.createElement("div", {
      style: {
        position: "relative"
      }
    }, loading && /*#__PURE__*/React.createElement(CircularProgress, null), renderPlugins("pb-page-details-revision-content-preview", {
      pageDetails: pageDetails,
      refreshPages: refreshPages
    })))));
  }
}, {
  name: "pb-page-details-revision-preview",
  type: "pb-page-details-revision-content-preview",
  render: function render(_ref2) {
    var pageDetails = _ref2.pageDetails;
    return /*#__PURE__*/React.createElement(PagePreview, {
      pageDetails: pageDetails
    });
  }
}];
export default plugins;
