// @webiny/app-headless-cms/admin/views/Content/ContentDataList.js
import _taggedTemplateLiteral from "@babel/runtime/helpers/taggedTemplateLiteral";
import _objectSpread from "@babel/runtime/helpers/objectSpread2";
import _upperFirst from "lodash/upperFirst";

function _templateObject2() {
  var data = _taggedTemplateLiteral(["Last modified: {time}."]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteral(["Created by: {user}."]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

import React from "react";
import { i18n } from "@webiny/app/i18n";
import CurrentEnvironmentLabel from "./../../components/CurrentEnvironmentLabel";
import TimeAgo from "timeago-react";
import { Typography } from "@webiny/ui/Typography";
import { css } from "emotion";
import { DataList, List, ListItem, ListItemText, ListItemTextSecondary, ListItemMeta } from "@webiny/ui/List";
import { I18NValue } from "@webiny/app-i18n/components";
var t = i18n.ns("app-headless-cms/admin/contents/data-list");
var rightAlign = /*#__PURE__*/css({
  alignItems: "flex-end !important",
  justifyContent: "center !important"
}, "label:rightAlign;");
var listItemMinHeight = /*#__PURE__*/css({
  minHeight: "66px !important"
}, "label:listItemMinHeight;");

var getValue = function getValue(item) {
  if (!item.value) {
    return _objectSpread(_objectSpread({}, item), {}, {
      value: "Untitled"
    });
  }

  return item;
};

var ContentDataList = function ContentDataList(_ref) {
  var contentModel = _ref.contentModel,
      dataList = _ref.dataList;
  return /*#__PURE__*/React.createElement(DataList, Object.assign({}, dataList, {
    title: contentModel.pluralizedName,
    actions: /*#__PURE__*/React.createElement(CurrentEnvironmentLabel, {
      style: {
        justifyContent: "flex-end"
      }
    })
  }), function (_ref2) {
    var data = _ref2.data,
        isSelected = _ref2.isSelected,
        select = _ref2.select;
    return /*#__PURE__*/React.createElement(List, {
      "data-testid": "default-data-list"
    }, data.map(function (item) {
      return /*#__PURE__*/React.createElement(ListItem, {
        key: item.id,
        className: listItemMinHeight,
        selected: isSelected(item)
      }, /*#__PURE__*/React.createElement(ListItemText, {
        onClick: function onClick() {
          return select(item);
        }
      }, /*#__PURE__*/React.createElement(I18NValue, {
        value: getValue(item.meta.title)
      }), /*#__PURE__*/React.createElement(ListItemTextSecondary, null, item.createdBy && item.createdBy.firstName && /*#__PURE__*/React.createElement(React.Fragment, null, t(_templateObject())({
        user: item.createdBy.firstName
      }), " "), t(_templateObject2())({
        time: /*#__PURE__*/React.createElement(TimeAgo, {
          datetime: item.savedOn
        })
      }))), /*#__PURE__*/React.createElement(ListItemMeta, {
        className: rightAlign
      }, /*#__PURE__*/React.createElement(Typography, {
        use: "subtitle2"
      }, _upperFirst(item.meta.status), " (v", item.meta.version, ")")));
    }));
  });
};

export default ContentDataList;
