// @webiny/app-i18n/admin/plugins/menus.js
import _taggedTemplateLiteral from "@babel/runtime/helpers/taggedTemplateLiteral";

function _templateObject3() {
  var data = _taggedTemplateLiteral(["Locales"]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteral(["Locales"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteral(["Languages"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

import React from "react";
import { i18n } from "@webiny/app/i18n";
import { SecureView } from "@webiny/app-security/components";
import { ReactComponent as I18NIcon } from "@svgr/webpack!./../assets/icons/round-translate-24px.svg";
var t = i18n.ns("app-form-builder/admin/menus");
var ROLE_I18N_LOCALES = ["i18n:locale:crud"];
var plugin = {
  type: "admin-menu",
  name: "menu-languages",
  render: function render(_ref) {
    var Menu = _ref.Menu,
        Section = _ref.Section,
        Item = _ref.Item;
    return /*#__PURE__*/React.createElement(SecureView, {
      scopes: ROLE_I18N_LOCALES
    }, /*#__PURE__*/React.createElement(Menu, {
      name: "languages",
      label: t(_templateObject()),
      icon: /*#__PURE__*/React.createElement(I18NIcon, null)
    }, /*#__PURE__*/React.createElement(Section, {
      label: t(_templateObject2())
    }, /*#__PURE__*/React.createElement(Item, {
      label: t(_templateObject3()),
      path: "/i18n/locales"
    }))));
  }
};
export default plugin;
