// @webiny/app-page-builder/admin/plugins/menus.js
import _taggedTemplateLiteral from "@babel/runtime/helpers/taggedTemplateLiteral";

function _templateObject5() {
  var data = _taggedTemplateLiteral(["Menus"]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _taggedTemplateLiteral(["Pages"]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _taggedTemplateLiteral(["Categories"]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteral(["Pages"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteral(["Content"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

import React from "react";
import { ReactComponent as PagesIcon } from "@svgr/webpack!@webiny/app-page-builder/admin/assets/round-ballot-24px.svg";
import { i18n } from "@webiny/app/i18n";
import { getPlugins } from "@webiny/plugins";
import { SecureView } from "@webiny/app-security/components";
var t = i18n.ns("app-form-builder/admin/menus");
var ROLE_PB_MENUS = ["pb:menus:crud"];
var ROLE_PB_CATEGORIES = ["pb:category:crud"];
var ROLE_PB_EDITOR = ["pb:page:crud"];
var plugin = {
  type: "admin-menu",
  name: "menu-content",
  render: function render(_ref) {
    var Menu = _ref.Menu,
        Section = _ref.Section,
        Item = _ref.Item;
    return /*#__PURE__*/React.createElement(SecureView, {
      scopes: {
        menus: ROLE_PB_MENUS,
        categories: ROLE_PB_CATEGORIES,
        editor: ROLE_PB_EDITOR
      }
    }, function (_ref2) {
      var scopes = _ref2.scopes;
      var menus = scopes.menus,
          categories = scopes.categories,
          editor = scopes.editor;

      if (!menus && !categories && !editor) {
        return null;
      }

      return /*#__PURE__*/React.createElement(Menu, {
        name: "content-2",
        label: t(_templateObject()),
        icon: /*#__PURE__*/React.createElement(PagesIcon, null)
      }, /*#__PURE__*/React.createElement(Section, {
        label: t(_templateObject2())
      }, categories && /*#__PURE__*/React.createElement(Item, {
        label: t(_templateObject3()),
        path: "/page-builder/categories"
      }), editor && /*#__PURE__*/React.createElement(Item, {
        label: t(_templateObject4()),
        path: "/page-builder/pages"
      }), menus && /*#__PURE__*/React.createElement(Item, {
        label: t(_templateObject5()),
        path: "/page-builder/menus"
      })), getPlugins("admin-menu-content-section").map(function (plugin) {
        return /*#__PURE__*/React.createElement(React.Fragment, {
          key: plugin.name
        }, plugin.render({
          Section: Section,
          Item: Item
        }));
      }));
    });
  }
};
export default plugin;
