// @webiny/app-page-builder/admin/plugins/routes.js
import _objectWithoutProperties from "@babel/runtime/helpers/objectWithoutProperties";
import React, { Suspense, lazy } from "react";
import { Route } from "@webiny/react-router";
import Helmet from "react-helmet";
import { AdminLayout } from "@webiny/app-admin/components/AdminLayout";
import { SecureRoute } from "@webiny/app-security/components";
import { CircularProgress } from "@webiny/ui/Progress";
import { EditorPluginsLoader } from "../components/EditorPluginsLoader";

var Loader = function Loader(_ref) {
  var children = _ref.children,
      props = _objectWithoutProperties(_ref, ["children"]);

  return /*#__PURE__*/React.createElement(Suspense, {
    fallback: /*#__PURE__*/React.createElement(CircularProgress, null)
  }, /*#__PURE__*/React.cloneElement(children, props));
};

var Categories = /*#__PURE__*/lazy(function () {
  return import("@webiny/app-page-builder/admin/views/Categories/Categories");
});
var Menus = /*#__PURE__*/lazy(function () {
  return import("@webiny/app-page-builder/admin/views/Menus/Menus");
});
var Pages = /*#__PURE__*/lazy(function () {
  return import("@webiny/app-page-builder/admin/views/Pages/Pages");
});
var Editor = /*#__PURE__*/lazy(function () {
  return import("@webiny/app-page-builder/admin/views/Pages/Editor");
});
var ROLE_PB_CATEGORY = ["pb:category:crud"];
var ROLE_PB_MENUS = ["pb:menu:crud"];
var ROLE_PB_PAGES = ["pb:page:crud"];
var plugins = [{
  name: "route-pb-categories",
  type: "route",
  route: /*#__PURE__*/React.createElement(Route, {
    exact: true,
    path: "/page-builder/categories",
    render: function render() {
      return /*#__PURE__*/React.createElement(SecureRoute, {
        scopes: ROLE_PB_CATEGORY
      }, /*#__PURE__*/React.createElement(AdminLayout, null, /*#__PURE__*/React.createElement(Helmet, {
        title: "Page Builder - Categories"
      }), /*#__PURE__*/React.createElement(Loader, null, /*#__PURE__*/React.createElement(Categories, null))));
    }
  })
}, {
  name: "route-pb-menus",
  type: "route",
  route: /*#__PURE__*/React.createElement(Route, {
    exact: true,
    path: "/page-builder/menus",
    render: function render() {
      return /*#__PURE__*/React.createElement(SecureRoute, {
        scopes: ROLE_PB_MENUS
      }, /*#__PURE__*/React.createElement(AdminLayout, null, /*#__PURE__*/React.createElement(Helmet, {
        title: "Page Builder - Menus"
      }), /*#__PURE__*/React.createElement(Loader, null, /*#__PURE__*/React.createElement(Menus, null))));
    }
  })
}, {
  name: "route-pb-pages",
  type: "route",
  route: /*#__PURE__*/React.createElement(Route, {
    exact: true,
    path: "/page-builder/pages",
    render: function render(_ref2) {
      var location = _ref2.location;
      return /*#__PURE__*/React.createElement(SecureRoute, {
        scopes: ROLE_PB_PAGES
      }, /*#__PURE__*/React.createElement(EditorPluginsLoader, {
        location: location
      }, /*#__PURE__*/React.createElement(AdminLayout, null, /*#__PURE__*/React.createElement(Helmet, {
        title: "Page Builder - Pages"
      }), /*#__PURE__*/React.createElement(Loader, null, /*#__PURE__*/React.createElement(Pages, null)))));
    }
  })
}, {
  name: "route-pb-editor",
  type: "route",
  route: /*#__PURE__*/React.createElement(Route, {
    exact: true,
    path: "/page-builder/editor/:id",
    render: function render(_ref3) {
      var location = _ref3.location;
      return /*#__PURE__*/React.createElement(SecureRoute, {
        scopes: ROLE_PB_PAGES
      }, /*#__PURE__*/React.createElement(EditorPluginsLoader, {
        location: location
      }, /*#__PURE__*/React.createElement(Helmet, {
        title: "Page Builder - Edit page"
      }), /*#__PURE__*/React.createElement(Loader, null, /*#__PURE__*/React.createElement(Editor, null))));
    }
  })
}];
export default plugins;
