// @webiny/app-page-builder/admin/plugins/scopesList.js
import _taggedTemplateLiteral from "@babel/runtime/helpers/taggedTemplateLiteral";

function _templateObject10() {
  var data = _taggedTemplateLiteral(["Allows reading oEmbed data in the Page Builder editor."]);

  _templateObject10 = function _templateObject10() {
    return data;
  };

  return data;
}

function _templateObject9() {
  var data = _taggedTemplateLiteral(["Read oEmbed data"]);

  _templateObject9 = function _templateObject9() {
    return data;
  };

  return data;
}

function _templateObject8() {
  var data = _taggedTemplateLiteral(["Allows CRUD operations on all elements."]);

  _templateObject8 = function _templateObject8() {
    return data;
  };

  return data;
}

function _templateObject7() {
  var data = _taggedTemplateLiteral(["Elements CRUD"]);

  _templateObject7 = function _templateObject7() {
    return data;
  };

  return data;
}

function _templateObject6() {
  var data = _taggedTemplateLiteral(["Allows CRUD operations on all categories."]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = _taggedTemplateLiteral(["Categories CRUD"]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _taggedTemplateLiteral(["Allows CRUD operations on all menus."]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _taggedTemplateLiteral(["Menus CRUD"]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteral(["Allows CRUD operations on all pages."]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteral(["Pages CRUD"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

import { i18n } from "@webiny/app/i18n";
var t = i18n.ns("app-page-builder/admin/scopesList");
export default [{
  name: "security-scopes-list-page-builder",
  type: "security-scopes-list",
  scopes: [{
    scope: "pb:page:crud",
    title: t(_templateObject()),
    description: t(_templateObject2())
  }, {
    scope: "pb:menu:crud",
    title: t(_templateObject3()),
    description: t(_templateObject4())
  }, {
    scope: "pb:category:crud",
    title: t(_templateObject5()),
    description: t(_templateObject6())
  }, {
    scope: "pb:element:crud",
    title: t(_templateObject7()),
    description: t(_templateObject8())
  }, {
    scope: "pb:oembed:read",
    title: t(_templateObject9()),
    description: t(_templateObject10())
  }]
}];
