// @webiny/app-i18n/admin/views/I18NLocales/graphql.js
import _taggedTemplateLiteral from "@babel/runtime/helpers/taggedTemplateLiteral";

function _templateObject6() {
  var data = _taggedTemplateLiteral(["\n    query searchLocaleCodes($search: String!) {\n        i18n {\n            codes: searchLocaleCodes(search: $search) {\n                data\n            }\n        }\n    }\n"]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = _taggedTemplateLiteral(["\n    mutation deleteI18NLocale($id: ID!) {\n        i18n {\n            deleteI18NLocale(id: $id) {\n                data\n                error {\n                    code\n                    message\n                }\n            }\n        }\n    }\n"]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _taggedTemplateLiteral(["\n    mutation updateI18NLocale($id: ID!, $data: I18NLocaleInput!){\n        i18n {\n            locale: updateI18NLocale(id: $id, data: $data) {\n                data {\n                    ", "\n                }\n                error {\n                    code\n                    message\n                    data\n                }\n            }\n        }\n    }\n"]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _taggedTemplateLiteral(["\n    mutation createI18NLocale($data: I18NLocaleInput!){\n        i18n {\n            locale: createI18NLocale(data: $data) {\n                data {\n                    ", "\n                }\n                error {\n                    code\n                    message\n                    data\n                }\n            }\n        }\n    }\n"]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteral(["\n    query getLocale($id: ID!) {\n        i18n {\n            locale: getI18NLocale(id: $id){\n                data {\n                    ", "\n                }\n                error {\n                    code\n                    message\n                }\n            }\n        }\n    }\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteral(["\n    query listI18NLocales(\n        $where: JSON\n        $sort: JSON\n        $search: I18NLocaleSearchInput\n        $limit: Int\n        $after: String\n        $before: String\n    ) {\n        i18n {\n            i18NLocales: listI18NLocales(\n                where: $where\n                sort: $sort\n                search: $search\n                limit: $limit\n                after: $after\n                before: $before\n            ) {\n                data {\n                    ", "\n                }\n                meta {\n                    cursors {\n                        next\n                        previous\n                    }\n                    hasNextPage\n                    hasPreviousPage\n                    totalCount\n                }\n            }\n        }\n    }\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

import gql from "graphql-tag";
var BASE_FIELDS = "\n    id\n    code\n    default\n    createdOn\n";
export var LIST_LOCALES = gql(_templateObject(), BASE_FIELDS);
export var READ_LOCALE = gql(_templateObject2(), BASE_FIELDS);
export var CREATE_LOCALE = gql(_templateObject3(), BASE_FIELDS);
export var UPDATE_LOCALE = gql(_templateObject4(), BASE_FIELDS);
export var DELETE_LOCALE = gql(_templateObject5());
export var SEARCH_LOCALE_CODES = gql(_templateObject6());
