// @webiny/app-i18n/admin/views/I18NLocales/I18NLocales.js
import React from "react";
import { SplitView, LeftPanel, RightPanel } from "@webiny/app-admin/components/SplitView";
import { FloatingActionButton } from "@webiny/app-admin/components/FloatingActionButton";
import { CrudProvider } from "@webiny/app-admin/contexts/Crud";
import I18NLocalesDataList from "./I18NLocalesDataList";
import I18NLocalesForm from "./I18NLocalesForm";
import { READ_LOCALE, LIST_LOCALES, CREATE_LOCALE, UPDATE_LOCALE, DELETE_LOCALE } from "./graphql";
export default function I18NLocales() {
  return /*#__PURE__*/React.createElement(CrudProvider, {
    delete: DELETE_LOCALE,
    read: READ_LOCALE,
    create: CREATE_LOCALE,
    update: UPDATE_LOCALE,
    list: {
      query: LIST_LOCALES,
      variables: {
        sort: {
          savedOn: -1
        }
      }
    }
  }, function (_ref) {
    var actions = _ref.actions;
    return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(SplitView, null, /*#__PURE__*/React.createElement(LeftPanel, null, /*#__PURE__*/React.createElement(I18NLocalesDataList, null)), /*#__PURE__*/React.createElement(RightPanel, null, /*#__PURE__*/React.createElement(I18NLocalesForm, null))), /*#__PURE__*/React.createElement(FloatingActionButton, {
      onClick: actions.resetForm
    }));
  });
}
