// @webiny/app-security/admin/views/PersonalAccessTokens/TokenList.js
import _taggedTemplateLiteral from "@babel/runtime/helpers/taggedTemplateLiteral";

function _templateObject() {
  var data = _taggedTemplateLiteral(["No tokens have been generated yet."]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

import React from "react";
import { SimpleListItem } from "@webiny/ui/List";
import TokenListItem from "./TokenListItem";
import { i18n } from "@webiny/app/i18n";
var t = i18n.ns("app-security/admin/roles/data-list");

var TokenList = function TokenList(_ref) {
  var setFormIsLoading = _ref.setFormIsLoading,
      data = _ref.data,
      setValue = _ref.setValue;

  if (data.personalAccessTokens && data.personalAccessTokens.length > 0) {
    return data.personalAccessTokens.map(function (PAT) {
      return /*#__PURE__*/React.createElement(TokenListItem, {
        setFormIsLoading: setFormIsLoading,
        key: PAT.id,
        PAT: PAT,
        data: data,
        setValue: setValue
      });
    });
  }

  return /*#__PURE__*/React.createElement(SimpleListItem, {
    text: t(_templateObject())
  });
};

export default TokenList;
