// @webiny/app-page-builder/editor/plugins/elementSettings/components/ColorPicker.js
import _get from "lodash/get";
import * as React from "react";
import { connect } from "react-redux";
import { Typography } from "@webiny/ui/Typography";
import { Grid, Cell } from "@webiny/ui/Grid";
import ColorPickerCmp from "@webiny/app-page-builder/editor/components/ColorPicker";
import { getActiveElement } from "@webiny/app-page-builder/editor/selectors";

var ColorPicker = function ColorPicker(_ref) {
  var label = _ref.label,
      value = _ref.value,
      updatePreview = _ref.updatePreview,
      updateValue = _ref.updateValue;
  return /*#__PURE__*/React.createElement(Grid, null, /*#__PURE__*/React.createElement(Cell, {
    span: 4
  }, /*#__PURE__*/React.createElement(Typography, {
    use: "overline"
  }, label)), /*#__PURE__*/React.createElement(Cell, {
    span: 8
  }, /*#__PURE__*/React.createElement(ColorPickerCmp, {
    compact: true,
    value: value,
    onChange: updatePreview,
    onChangeComplete: updateValue
  })));
};

export default connect(function (state, _ref2) {
  var value = _ref2.value,
      valueKey = _ref2.valueKey,
      defaultValue = _ref2.defaultValue;
  return {
    value: valueKey ? _get(getActiveElement(state), valueKey, defaultValue) : value
  };
})( /*#__PURE__*/React.memo(ColorPicker));
