// @webiny/app-page-builder/editor/plugins/elementSettings/components/SliderWithInput.js
import _get from "lodash/get";
import * as React from "react";
import { connect } from "react-redux";
import { Input } from "@webiny/ui/Input";
import { Grid, Cell } from "@webiny/ui/Grid";
import { Icon } from "@webiny/ui/Icon";
import { Slider } from "@webiny/ui/Slider";
import { InputContainer } from "@webiny/app-page-builder/editor/plugins/elementSettings/components/StyledComponents";
import { getActiveElement } from "@webiny/app-page-builder/editor/selectors";

var SliderWithInput = function SliderWithInput(_ref) {
  var value = _ref.value,
      icon = _ref.icon,
      placeholder = _ref.placeholder,
      updateValue = _ref.updateValue,
      updatePreview = _ref.updatePreview,
      className = _ref.className;
  return /*#__PURE__*/React.createElement(Grid, {
    className: className
  }, /*#__PURE__*/React.createElement(Cell, {
    span: 2
  }, /*#__PURE__*/React.createElement(Icon, {
    icon: icon
  })), /*#__PURE__*/React.createElement(Cell, {
    span: 6
  }, /*#__PURE__*/React.createElement(Slider, {
    value: value,
    onChange: updateValue,
    onInput: updatePreview,
    min: 0,
    max: 100,
    step: 1
  })), /*#__PURE__*/React.createElement(Cell, {
    span: 4
  }, /*#__PURE__*/React.createElement(InputContainer, null, /*#__PURE__*/React.createElement(Input, {
    placeholder: placeholder || "px",
    value: value,
    onChange: updateValue
  }))));
};

export default connect(function (state, _ref2) {
  var valueKey = _ref2.valueKey;
  return {
    value: _get(getActiveElement(state), valueKey, 0)
  };
})(SliderWithInput);
