// @webiny/app-google-tag-manager/render/index.js
import React, { useEffect } from "react";
import { Query } from "react-apollo";
import query from "./graphql";

function GoogleTagManager(_ref) {
  var settings = _ref.settings;
  useEffect(function () {
    // @ts-ignore
    if (!settings || settings.enabled !== true || window.dataLayer) {
      return;
    } // eslint-disable


    document.querySelector("body").prepend(new window.DOMParser().parseFromString("<noscript><iframe src=\"https://www.googletagmanager.com/ns.html?id=".concat(settings.code, "\"\n                  height=\"0\" width=\"0\" style=\"display:none;visibility:hidden\"></iframe></noscript>"), "text/xml").firstChild);

    (function (w, d, s, l, i) {
      w[l] = w[l] || [];
      w[l].push({
        "gtm.start": new Date().getTime(),
        event: "gtm.js"
      }); // eslint-disable-next-line

      var f = d.getElementsByTagName(s)[0],
          j = d.createElement(s),
          dl = l != "dataLayer" ? "&l=" + l : ""; // @ts-ignore

      j.async = true; // @ts-ignore

      j.src = "https://www.googletagmanager.com/gtm.js?id=" + i + dl;
      f.parentNode.insertBefore(j, f);
    })(window, document, "script", "dataLayer", settings.code);
  }, []);
  return null;
}

export default (function () {
  return {
    type: "addon-render",
    name: "addon-render-google-tag-manager",
    component: /*#__PURE__*/React.createElement(Query, {
      query: query
    }, function (_ref2) {
      var data = _ref2.data,
          loading = _ref2.loading;
      return loading ? null : /*#__PURE__*/React.createElement(GoogleTagManager, {
        settings: data.googleTagManager.getSettings.data
      });
    })
  };
});
