// @webiny/app-i18n/site/plugins/richTextEditor/typography/index.js
import _objectSpread from "@babel/runtime/helpers/objectSpread2";
import * as React from "react";
import theme from "./theme";
var plugin = {
  name: "i18n-value-rich-text-editor-typography",
  type: "i18n-value-rich-text-editor",
  plugin: {
    name: "typography",
    editor: {
      renderElement: function renderElement(props, next) {
        var attributes = props.attributes,
            children = props.children,
            element = props.element; // @ts-ignore

        var type = element.type; // @ts-ignore

        var typography = theme.typography;

        if (typography.hasOwnProperty(type) && typography[type].component) {
          var _typography$type = typography[type],
              Component = _typography$type.component,
              _typography$type$clas = _typography$type.className,
              className = _typography$type$clas === void 0 ? null : _typography$type$clas;

          var nodeProps = _objectSpread(_objectSpread({}, attributes), {}, {
            className: className
          });

          if (typeof Component !== "string") {
            nodeProps = props;
          }

          return /*#__PURE__*/React.createElement(Component, nodeProps, children);
        }

        return next();
      }
    }
  }
};
export default plugin;
