// @webiny/app-admin/plugins/FileManager/fileImage/DeleteAction.js
import _regeneratorRuntime from "@babel/runtime/regenerator";
import _asyncToGenerator from "@babel/runtime/helpers/asyncToGenerator";
import _slicedToArray from "@babel/runtime/helpers/slicedToArray";
import _taggedTemplateLiteral from "@babel/runtime/helpers/taggedTemplateLiteral";

function _templateObject2() {
  var data = _taggedTemplateLiteral(["File deleted."]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteral(["\n    mutation deleteFile($id: ID!) {\n        files {\n            deleteFile(id: $id) {\n                data\n            }\n        }\n    }\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

import React from "react";
import { Tooltip } from "@webiny/ui/Tooltip";
import { IconButton } from "@webiny/ui/Button";
import { useSnackbar } from "@webiny/app-admin/hooks/useSnackbar";
import { ReactComponent as DeleteIcon } from "@svgr/webpack!../../../components/FileManager/icons/delete.svg";
import { useMutation } from "react-apollo";
import gql from "graphql-tag";
import { i18n } from "@webiny/app/i18n";
var t = i18n.ns("app-admin/file-manager/files/delete-action");
var DELETE_FILE = gql(_templateObject());

var DeleteAction = function DeleteAction(props) {
  var file = props.file;

  var _useSnackbar = useSnackbar(),
      showSnackbar = _useSnackbar.showSnackbar;

  var _useMutation = useMutation(DELETE_FILE, {
    variables: {
      id: file.id
    }
  }),
      _useMutation2 = _slicedToArray(_useMutation, 1),
      deleteFile = _useMutation2[0];

  return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(Tooltip, {
    content: /*#__PURE__*/React.createElement("span", null, "Edit image"),
    placement: "bottom"
  }, /*#__PURE__*/React.createElement(IconButton, {
    icon: /*#__PURE__*/React.createElement(DeleteIcon, {
      style: {
        margin: "0 8px 0 0"
      }
    }),
    onClick: /*#__PURE__*/_asyncToGenerator( /*#__PURE__*/_regeneratorRuntime.mark(function _callee() {
      return _regeneratorRuntime.wrap(function _callee$(_context) {
        while (1) {
          switch (_context.prev = _context.next) {
            case 0:
              _context.next = 2;
              return deleteFile();

            case 2:
              showSnackbar(t(_templateObject2()));

            case 3:
            case "end":
              return _context.stop();
          }
        }
      }, _callee);
    }))
  })));
};

export default DeleteAction;
