// @webiny/app-page-builder/admin/plugins/settings/index.js
import _taggedTemplateLiteral from "@babel/runtime/helpers/taggedTemplateLiteral";

function _templateObject5() {
  var data = _taggedTemplateLiteral(["General"]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _taggedTemplateLiteral(["Website"]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _taggedTemplateLiteral(["Page Builder"]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteral(["Page Builder - General Settings"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteral(["Page Builder - Website Settings"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

import * as React from "react";
import { Route } from "@webiny/react-router";
import { AdminLayout } from "@webiny/app-admin/components/AdminLayout";
import PageBuilderSettings from "./components/PageBuilderSettings";
import GeneralSettings from "./components/generalSettings/GeneralSettings";
import { SecureRoute } from "@webiny/app-security/components";
import { i18n } from "@webiny/app/i18n";
import { getPlugins } from "@webiny/plugins";
import Helmet from "react-helmet";
var t = i18n.ns("app-page-builder/admin/menus");
var ROLE_PB_SETTINGS = ["pb:settings"];
var plugins = [{
  type: "route",
  name: "route-settings-website",
  route: /*#__PURE__*/React.createElement(Route, {
    path: "/settings/page-builder/website",
    render: function render() {
      return /*#__PURE__*/React.createElement(AdminLayout, null, /*#__PURE__*/React.createElement(Helmet, {
        title: t(_templateObject())
      }), /*#__PURE__*/React.createElement(SecureRoute, {
        scopes: ROLE_PB_SETTINGS
      }, /*#__PURE__*/React.createElement(PageBuilderSettings, null)));
    }
  })
}, {
  type: "route",
  name: "route-settings-general",
  route: /*#__PURE__*/React.createElement(Route, {
    path: "/settings/page-builder/general",
    render: function render() {
      return /*#__PURE__*/React.createElement(AdminLayout, null, /*#__PURE__*/React.createElement(Helmet, {
        title: t(_templateObject2())
      }), /*#__PURE__*/React.createElement(SecureRoute, {
        scopes: ROLE_PB_SETTINGS
      }, /*#__PURE__*/React.createElement(GeneralSettings, null)));
    }
  })
}, {
  type: "admin-menu-settings",
  name: "menu-settings-page-builder",
  render: function render(_ref) {
    var Section = _ref.Section,
        Item = _ref.Item;
    return /*#__PURE__*/React.createElement(Section, {
      label: t(_templateObject3())
    }, getPlugins("menu-settings-page-builder").map(function (plugin) {
      return /*#__PURE__*/React.createElement(React.Fragment, {
        key: plugin.name + new Date()
      }, plugin.render({
        Item: Item
      }));
    }));
  }
}, {
  type: "menu-settings-page-builder",
  name: "menu-settings-website",
  render: function render(_ref2) {
    var Item = _ref2.Item;
    return /*#__PURE__*/React.createElement(Item, {
      label: t(_templateObject4()),
      path: "/settings/page-builder/website"
    });
  }
}, {
  type: "menu-settings-page-builder",
  name: "menu-settings-general",
  render: function render(_ref3) {
    var Item = _ref3.Item;
    return /*#__PURE__*/React.createElement(Item, {
      label: t(_templateObject5()),
      path: "/settings/page-builder/general"
    });
  }
}];
export default plugins;
