// @webiny/app-admin/plugins/Menu/Navigation/components/Item.js
import * as React from "react";
import { List, ListItem } from "@webiny/ui/List";
import { Link } from "@webiny/react-router";
import useNavigation from "./useNavigation";
import { css } from "emotion";
var linkStyle = /*#__PURE__*/css({
  color: "var(--mdc-theme-text-primary-on-background)",
  textDecoration: "none",
  display: "block",
  width: "100%",
  outline: "none",
  paddingLeft: 65,
  "&:hover": {
    textDecoration: "none"
  }
}, "label:linkStyle;");
var submenuItems = /*#__PURE__*/css({
  ".mdc-drawer &.mdc-list-item": {
    paddingLeft: 0
  }
}, "label:submenuItems;");
var submenuList = /*#__PURE__*/css({
  "&.mdc-list": {
    padding: 0
  }
}, "label:submenuList;");
export default function Item(props) {
  var path = props.path,
      label = props.label,
      style = props.style,
      onClick = props.onClick;

  var _useNavigation = useNavigation(),
      hideMenu = _useNavigation.hideMenu;

  return /*#__PURE__*/React.createElement(List, {
    className: submenuList,
    style: style
  }, /*#__PURE__*/React.createElement(ListItem, {
    className: submenuItems
  }, path ? /*#__PURE__*/React.createElement(Link, {
    className: linkStyle,
    to: path,
    onClick: hideMenu
  }, label) : /*#__PURE__*/React.createElement("span", {
    onClick: onClick,
    className: linkStyle
  }, label)));
}
