// @webiny/app-headless-cms/admin/plugins/contentDetails/header/localeSelector/LocaleSelector.js
import React from "react";
import { css } from "emotion";
import { ButtonDefault } from "@webiny/ui/Button";
import { Icon } from "@webiny/ui/Icon";
import { ReactComponent as DownButton } from "@svgr/webpack!@webiny/app-headless-cms/admin/icons/round-arrow_drop_down-24px.svg";
import { ReactComponent as DoneIcon } from "@svgr/webpack!@webiny/app-headless-cms/admin/icons/done-24px.svg";
import { Menu, MenuItem } from "@webiny/ui/Menu";
import { useI18N } from "@webiny/app-i18n/hooks/useI18N";
import { ListItemGraphic } from "@webiny/ui/List";
var buttonStyle = /*#__PURE__*/css({
  "&.mdc-button": {
    color: "var(--mdc-theme-text-primary-on-background) !important"
  }
}, "label:buttonStyle;");
var menuStyles = /*#__PURE__*/css({
  width: 175,
  right: 0,
  ".mdc-list-item__graphic": {
    marginRight: 10
  }
}, "label:menuStyles;");

var LocaleSelector = function LocaleSelector(_ref) {
  var getLocale = _ref.getLocale,
      setLocale = _ref.setLocale,
      getLoading = _ref.getLoading;
  var i18n = useI18N();
  return /*#__PURE__*/React.createElement(Menu, {
    className: menuStyles,
    handle: /*#__PURE__*/React.createElement(ButtonDefault, {
      className: buttonStyle,
      disabled: getLoading()
    }, i18n.getLocales().find(function (item) {
      return item.id === getLocale();
    }).code, /*#__PURE__*/React.createElement(Icon, {
      icon: /*#__PURE__*/React.createElement(DownButton, null)
    }))
  }, i18n.getLocales().map(function (item) {
    return /*#__PURE__*/React.createElement(MenuItem, {
      key: item.id,
      onClick: function onClick() {
        return setLocale(item.id);
      }
    }, /*#__PURE__*/React.createElement(ListItemGraphic, null, item.id === getLocale() && /*#__PURE__*/React.createElement(Icon, {
      icon: /*#__PURE__*/React.createElement(DoneIcon, null)
    })), item.code);
  }));
};

export default LocaleSelector;
