// @webiny/app-admin/plugins/Menu/Hamburger.js
import React, { useCallback } from "react";
import ReactDOM from "react-dom";
import { get, set } from "dot-prop-immutable";
import { useUi } from "@webiny/app/hooks/useUi";
import { IconButton } from "@webiny/ui/Button";
import { ReactComponent as MenuIcon } from "@svgr/webpack!@webiny/app-admin/assets/icons/baseline-menu-24px.svg";
import Navigation from "./Navigation";
var el = null;

var getElement = function getElement() {
  if (!el) {
    el = document.createElement("div");
    el.id = "navigation-drawer";
    document.body.appendChild(el);
  }

  return el;
};

var Hamburger = function Hamburger() {
  var ui = useUi();
  var toggleMenu = useCallback(function () {
    ui.setState(function (ui) {
      return set(ui, "appsMenu.show", !(get(ui, "appsMenu.show") || false));
    });
  }, [ui]);
  return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(IconButton, {
    icon: /*#__PURE__*/React.createElement(MenuIcon, {
      style: {
        color: "white"
      }
    }),
    onClick: toggleMenu
  }), /*#__PURE__*/ReactDOM.createPortal( /*#__PURE__*/React.createElement(Navigation, null), getElement()));
};

export default Hamburger;
