// @webiny/app-plugin-admin-welcome-screen/components/Welcome.js
import React from "react";
import { css } from "emotion";
import styled from "@emotion/styled";
import { Link } from "@webiny/react-router";
import { SimpleForm, SimpleFormHeader, SimpleFormContent, SimpleFormFooter } from "@webiny/app-admin/components/SimpleForm";
import { useSecurity } from "@webiny/app-security/hooks/useSecurity";
import { getPlugins } from "@webiny/plugins";
import { Typography } from "@webiny/ui/Typography";
import { Cell, Grid } from "@webiny/ui/Grid";
import { Elevation } from "@webiny/ui/Elevation";
import GithubIcon from "../icons/github-logo.svg";
import SlackIcon from "../icons/slack-logo.svg";
import MediumIcon from "../icons/medium-logo.svg";
import TwitterIcon from "../icons/twitter-logo.svg";
import TextbookIcon from "../icons/textbook.svg";
import LaptopIcon from "../icons/laptop.svg";
var linkStyle = /*#__PURE__*/css({
  textDecoration: "none",
  "&:hover": {
    textDecoration: "none"
  },
  color: "var(--mdc-theme-text-primary-on-background)"
}, "label:linkStyle;");
var imageStyle = /*#__PURE__*/css({
  width: "30px",
  height: "30px",
  marginBottom: "5px"
}, "label:imageStyle;");
var communityStyle = /*#__PURE__*/css({
  textAlign: "left"
}, "label:communityStyle;");
var widgetTitleStyle = /*#__PURE__*/css({
  fontWeight: 600,
  paddingTop: "1rem",
  paddingBottom: "1rem",
  textAlign: "center"
}, "label:widgetTitleStyle;");
var widgetDescriptionStyle = /*#__PURE__*/css({
  textAlign: "center",
  paddingLeft: "20px",
  paddingRight: "20px"
}, "label:widgetDescriptionStyle;");
var iconTextStyle = /*#__PURE__*/css({
  textAlign: "center"
}, "label:iconTextStyle;");
var pFormContentStyle = /*#__PURE__*/css({
  fontWeight: 600
}, "label:pFormContentStyle;");
var pGetStartedStyle = /*#__PURE__*/css({
  paddingLeft: "1.5rem",
  paddingTop: "1.5rem"
}, "label:pGetStartedStyle;");
var footerContainerStyle = /*#__PURE__*/css({
  display: "flex",
  padding: "1rem"
}, "label:footerContainerStyle;");
var widgetButtonStyle = /*#__PURE__*/css({
  textAlign: "center"
}, "label:widgetButtonStyle;");
var footerTextStyle = /*#__PURE__*/css({
  backgroundColor: "var(--mdc-theme-on-background)",
  margin: "1rem"
}, "label:footerTextStyle;");
var footerLinkTextStyle = /*#__PURE__*/css({
  fontWeight: 600,
  paddingLeft: "1rem"
}, "label:footerLinkTextStyle;");
var ContentTheme = /*#__PURE__*/styled("div", {
  target: "e12owhql0",
  label: "ContentTheme"
})({
  color: "var(--mdc-theme-text-primary-on-background)"
});
var Widget = /*#__PURE__*/styled("div", {
  target: "e12owhql1",
  label: "Widget"
})({
  marginLeft: "1.5rem",
  marginRight: "1.5rem",
  marginBottom: "2rem"
});

var Welcome = function Welcome() {
  var security = useSecurity();

  if (!security || !security.user) {
    return null;
  }

  var fullName = security.user.fullName;
  return /*#__PURE__*/React.createElement(Grid, null, /*#__PURE__*/React.createElement(Cell, {
    span: 12
  }, /*#__PURE__*/React.createElement(SimpleForm, null, /*#__PURE__*/React.createElement(SimpleFormHeader, {
    title: "Hi ".concat(fullName.split(" ")[0], "!")
  }), /*#__PURE__*/React.createElement(SimpleFormContent, null, /*#__PURE__*/React.createElement(ContentTheme, null, /*#__PURE__*/React.createElement(Cell, {
    span: 12
  }, /*#__PURE__*/React.createElement(Typography, {
    use: "headline6"
  }, /*#__PURE__*/React.createElement("p", {
    className: pGetStartedStyle
  }, "To get started - pick one of the actions below:"), /*#__PURE__*/React.createElement("br", null))), /*#__PURE__*/React.createElement("div", {
    style: {
      display: "flex"
    }
  }, getPlugins("admin-welcome-screen-widget").map(function (pl) {
    return /*#__PURE__*/React.createElement(Widget, {
      key: pl.name
    }, /*#__PURE__*/React.createElement(Elevation, {
      z: 2,
      style: {
        padding: 10
      }
    }, /*#__PURE__*/React.createElement(Typography, {
      use: "headline6"
    }, /*#__PURE__*/React.createElement("p", {
      className: widgetTitleStyle
    }, pl.widget.title)), /*#__PURE__*/React.createElement(Typography, {
      use: "body1"
    }, /*#__PURE__*/React.createElement("p", {
      className: widgetDescriptionStyle
    }, pl.widget.description)), /*#__PURE__*/React.createElement("div", {
      className: widgetButtonStyle
    }, pl.widget.cta)));
  })))), /*#__PURE__*/React.createElement(SimpleFormFooter, null, /*#__PURE__*/React.createElement(Grid, null, /*#__PURE__*/React.createElement(Cell, {
    span: 8
  }, /*#__PURE__*/React.createElement(Typography, {
    use: "headline6",
    className: communityStyle
  }, /*#__PURE__*/React.createElement("p", {
    className: pFormContentStyle
  }, "Learn more about Webiny:"))), /*#__PURE__*/React.createElement(Cell, {
    span: 4,
    style: {
      marginLeft: "1rem"
    }
  }, /*#__PURE__*/React.createElement(Typography, {
    use: "headline6",
    className: communityStyle
  }, /*#__PURE__*/React.createElement("p", {
    className: pFormContentStyle
  }, "Join our community:"))), /*#__PURE__*/React.createElement(Cell, {
    span: 8
  }, /*#__PURE__*/React.createElement(Typography, {
    use: "body1",
    className: communityStyle
  }, /*#__PURE__*/React.createElement("p", null, "Explore the Webiny documentation, learn about the architecture and check out code examples and guides:"))), /*#__PURE__*/React.createElement(Cell, {
    span: 4,
    style: {
      marginLeft: "1rem"
    }
  }, /*#__PURE__*/React.createElement(Typography, {
    use: "body1",
    className: communityStyle
  }, /*#__PURE__*/React.createElement("p", null, "Get to know Webiny team members, discuss new ideas and get help:"))), /*#__PURE__*/React.createElement(Cell, {
    span: 4,
    className: footerTextStyle,
    style: {
      margin: "1rem 1rem 1rem 0rem"
    }
  }, /*#__PURE__*/React.createElement(Link, {
    to: "https://docs.webiny.com/",
    className: linkStyle
  }, /*#__PURE__*/React.createElement(Typography, {
    use: "headline5"
  }, /*#__PURE__*/React.createElement("div", {
    className: footerContainerStyle
  }, /*#__PURE__*/React.createElement("img", {
    className: imageStyle,
    src: TextbookIcon
  }), /*#__PURE__*/React.createElement("p", {
    className: footerLinkTextStyle
  }, "Documentation"))))), /*#__PURE__*/React.createElement(Cell, {
    span: 4,
    className: footerTextStyle,
    style: {
      margin: "1rem 1rem 1rem 9px"
    }
  }, /*#__PURE__*/React.createElement(Link, {
    to: "https://github.com/webiny/webiny-examples",
    className: linkStyle
  }, /*#__PURE__*/React.createElement(Typography, {
    use: "headline5"
  }, /*#__PURE__*/React.createElement("span", {
    className: footerContainerStyle
  }, /*#__PURE__*/React.createElement("img", {
    className: imageStyle,
    src: LaptopIcon
  }), /*#__PURE__*/React.createElement("p", {
    className: footerLinkTextStyle
  }, "Code examples"))))), /*#__PURE__*/React.createElement(Cell, {
    span: 1,
    className: iconTextStyle,
    align: "middle"
  }, /*#__PURE__*/React.createElement(Link, {
    to: "https://github.com/webiny/webiny-js",
    className: linkStyle
  }, /*#__PURE__*/React.createElement("img", {
    className: imageStyle,
    src: GithubIcon
  }), /*#__PURE__*/React.createElement("p", null, "Github"))), /*#__PURE__*/React.createElement(Cell, {
    span: 1,
    className: iconTextStyle,
    align: "middle"
  }, /*#__PURE__*/React.createElement(Link, {
    to: "https://www.webiny.com/slack/",
    className: linkStyle
  }, /*#__PURE__*/React.createElement("img", {
    className: imageStyle,
    src: SlackIcon
  }), /*#__PURE__*/React.createElement("p", null, "Slack"))), /*#__PURE__*/React.createElement(Cell, {
    span: 1,
    className: iconTextStyle,
    align: "middle"
  }, /*#__PURE__*/React.createElement(Link, {
    to: "https://blog.webiny.com",
    className: linkStyle
  }, /*#__PURE__*/React.createElement("img", {
    className: imageStyle,
    src: MediumIcon
  }), /*#__PURE__*/React.createElement("p", null, "Medium"))), /*#__PURE__*/React.createElement(Cell, {
    span: 1,
    className: iconTextStyle,
    align: "middle"
  }, /*#__PURE__*/React.createElement(Link, {
    to: "https://twitter.com/WebinyPlatform",
    className: linkStyle
  }, /*#__PURE__*/React.createElement("img", {
    className: imageStyle,
    src: TwitterIcon
  }), /*#__PURE__*/React.createElement("p", null, "Twitter"))))))));
};

export default Welcome;
