// @webiny/app-headless-cms/admin/plugins/fieldRenderers/file/FileUpload.js
import _classCallCheck from "@babel/runtime/helpers/classCallCheck";
import _createClass from "@babel/runtime/helpers/createClass";
import _inherits from "@babel/runtime/helpers/inherits";
import _createSuper from "@babel/runtime/helpers/createSuper";
import _defineProperty from "@babel/runtime/helpers/defineProperty";
import * as React from "react";
import { Image } from "@webiny/app/components/Image";
import * as Ui from "@webiny/ui/ImageUpload";
import { FileManager } from "@webiny/app-admin/components/FileManager";
import { FormElementMessage } from "@webiny/ui/FormElementMessage";
import styled from "@emotion/styled";
var FileUploadWrapper = /*#__PURE__*/styled("div", {
  target: "ej2a27i0",
  label: "FileUploadWrapper"
})({
  position: "relative",
  ".disabled": {
    opacity: 0.75,
    pointerEvents: "none"
  },
  ".mdc-text-field-helper-text": {
    color: "var(--mdc-theme-text-secondary-on-background)"
  }
});

var FileUpload = /*#__PURE__*/function (_React$Component) {
  _inherits(FileUpload, _React$Component);

  var _super = _createSuper(FileUpload);

  function FileUpload() {
    _classCallCheck(this, FileUpload);

    return _super.apply(this, arguments);
  }

  _createClass(FileUpload, [{
    key: "render",
    value: function render() {
      var _this$props = this.props,
          className = _this$props.className,
          onChange = _this$props.onChange,
          value = _this$props.value,
          validation = _this$props.validation,
          description = _this$props.description,
          accept = _this$props.accept,
          maxSize = _this$props.maxSize,
          multipleMaxCount = _this$props.multipleMaxCount,
          multipleMaxSize = _this$props.multipleMaxSize,
          imagePreviewProps = _this$props.imagePreviewProps,
          round = _this$props.round,
          multiple = _this$props.multiple,
          placeholder = _this$props.placeholder,
          styles = _this$props.styles,
          renderImagePreview = _this$props.renderImagePreview;

      var defaultRenderImagePreview = function defaultRenderImagePreview(renderImageProps) {
        return /*#__PURE__*/React.createElement(Image, Object.assign({}, renderImageProps, imagePreviewProps));
      };

      return /*#__PURE__*/React.createElement(FileUploadWrapper, {
        className: className
      }, /*#__PURE__*/React.createElement(FileManager, {
        onChange: onChange,
        accept: accept,
        maxSize: maxSize,
        multiple: multiple,
        multipleMaxCount: multipleMaxCount,
        multipleMaxSize: multipleMaxSize
      }, function (_ref) {
        var showFileManager = _ref.showFileManager;
        return /*#__PURE__*/React.createElement(Ui.Image, {
          renderImagePreview: renderImagePreview || defaultRenderImagePreview,
          style: styles,
          value: value,
          uploadImage: showFileManager,
          removeImage: onChange,
          round: round,
          placeholder: placeholder
        });
      }), validation.isValid === false && /*#__PURE__*/React.createElement(FormElementMessage, {
        error: true
      }, validation.message), validation.isValid !== false && description && /*#__PURE__*/React.createElement(FormElementMessage, null, description));
    }
  }]);

  return FileUpload;
}(React.Component);

_defineProperty(FileUpload, "defaultProps", {
  validation: {
    isValid: null
  },
  styles: {
    width: "100%",
    height: "auto"
  }
});

export { FileUpload as default };
