// @webiny/app-headless-cms/admin/plugins/fieldRenderers/file/fileField.js
import _taggedTemplateLiteral from "@babel/runtime/helpers/taggedTemplateLiteral";

function _templateObject2() {
  var data = _taggedTemplateLiteral(["Enables selecting a single file via File Manager."]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteral(["File Input"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

import React from "react";
import SingleFile from "./SingleFile";
import { I18NValue } from "@webiny/app-i18n/components";
import { i18n } from "@webiny/app/i18n";
import { Cell, Grid } from "@webiny/ui/Grid";
var t = i18n.ns("app-headless-cms/admin/fields/file");
var plugin = {
  type: "cms-editor-field-renderer",
  name: "cms-editor-field-renderer-file",
  renderer: {
    rendererName: "file-input",
    name: t(_templateObject()),
    description: t(_templateObject2()),
    canUse: function canUse(_ref) {
      var field = _ref.field;
      return field.type === "file" && !field.multipleValues;
    },
    render: function render(_ref2) {
      var field = _ref2.field,
          getBind = _ref2.getBind,
          Label = _ref2.Label;
      var Bind = getBind();
      return /*#__PURE__*/React.createElement(Grid, null, /*#__PURE__*/React.createElement(Cell, {
        span: 12
      }, /*#__PURE__*/React.createElement(Label, null, /*#__PURE__*/React.createElement(I18NValue, {
        value: field.label
      })), /*#__PURE__*/React.createElement(Bind, null, function (bind) {
        return /*#__PURE__*/React.createElement(SingleFile, {
          field: field,
          bind: bind
        });
      })));
    }
  }
};
export default plugin;
