// @webiny/app-admin/components/withCrud/getFormData.js
import _get from "lodash/get";

var process = function process(_ref) {
  var data = _ref.data,
      form = _ref.form;

  var responseHandler = _get(form, "get.response");

  if (!responseHandler) {
    return data;
  }

  if (typeof responseHandler === "function") {
    return responseHandler(data);
  }

  var responseHandlers = {
    data: _get(responseHandler, "data"),
    error: _get(responseHandler, "error")
  };
  var formData = {
    data: undefined,
    error: undefined
  };

  if (typeof responseHandlers.data === "function") {
    formData.data = responseHandlers.data(data);
  }

  if (typeof responseHandlers.error === "function") {
    formData.error = responseHandlers.error(data);
  }

  return formData;
};

var getFormData = function getFormData(_ref2) {
  var data = _ref2.data,
      form = _ref2.form;
  var formData = process({
    data: data,
    form: form
  });
  return {
    loading: _get(data, "loading") || false,
    data: _get(formData, "data") || {},
    error: _get(formData, "error") || null
  };
};

export default getFormData;
