// @webiny/app-file-manager/admin/plugins/menus.js
import _taggedTemplateLiteral from "@babel/runtime/helpers/taggedTemplateLiteral";

function _templateObject2() {
  var data = _taggedTemplateLiteral(["General"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteral(["File Manager"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

import * as React from "react";
import { Route } from "@webiny/react-router";
import { Helmet } from "react-helmet";
import { AdminLayout } from "@webiny/app-admin/components/AdminLayout";
import FileManagerSettings from "../views/FileManagerSettings";
import { SecureRoute, SecureView } from "@webiny/app-security/components";
import { i18n } from "@webiny/app/i18n";
var t = i18n.ns("app-file-manager/admin");
var ROLE_FM_SETTINGS = ["files:settings"];
export default [{
  type: "route",
  name: "route-file-manager-settings-general",
  route: /*#__PURE__*/React.createElement(Route, {
    path: "/settings/file-manager/general",
    render: function render() {
      return /*#__PURE__*/React.createElement(AdminLayout, null, /*#__PURE__*/React.createElement(Helmet, {
        title: "File Manager Settings - General"
      }), /*#__PURE__*/React.createElement(SecureRoute, {
        scopes: ROLE_FM_SETTINGS
      }, /*#__PURE__*/React.createElement(FileManagerSettings, null)));
    }
  })
}, {
  type: "admin-menu-settings",
  name: "menu-file-manager-settings",
  render: function render(_ref) {
    var Section = _ref.Section,
        Item = _ref.Item;
    return /*#__PURE__*/React.createElement(SecureView, {
      scopes: ROLE_FM_SETTINGS
    }, /*#__PURE__*/React.createElement(Section, {
      label: t(_templateObject())
    }, /*#__PURE__*/React.createElement(Item, {
      label: t(_templateObject2()),
      path: "/settings/file-manager/general"
    })));
  }
}];
