// @webiny/app-page-builder/render/plugins/elements/button/Button.js
import _objectSpread from "@babel/runtime/helpers/objectSpread2";
import React from "react";
import { get } from "dot-prop-immutable";
import Slate from "@webiny/app-page-builder/render/components/Slate";
import { ElementRoot } from "@webiny/app-page-builder/render/components/ElementRoot";
import { Link } from "@webiny/react-router";

var Button = function Button(_ref) {
  var element = _ref.element;

  var _ref2 = element.data || {},
      _ref2$type = _ref2.type,
      type = _ref2$type === void 0 ? "default" : _ref2$type,
      _ref2$icon = _ref2.icon,
      icon = _ref2$icon === void 0 ? {} : _ref2$icon,
      _ref2$link = _ref2.link,
      link = _ref2$link === void 0 ? {} : _ref2$link;

  var _icon$svg = icon.svg,
      svg = _icon$svg === void 0 ? null : _icon$svg;
  var alignItems = get(element, "data.settings.horizontalAlignFlex") || "flex-start";
  var _icon$position = icon.position,
      position = _icon$position === void 0 ? "left" : _icon$position;
  var classes = ["webiny-pb-base-page-element-style", "webiny-pb-page-element-button", "webiny-pb-page-element-button--" + type, "webiny-pb-page-element-button__icon--" + position];
  var content = /*#__PURE__*/React.createElement(React.Fragment, null, svg && /*#__PURE__*/React.createElement("span", {
    dangerouslySetInnerHTML: {
      __html: svg
    }
  }), /*#__PURE__*/React.createElement(Slate, {
    value: element.data.text
  }));
  return /*#__PURE__*/React.createElement(ElementRoot, {
    element: element
  }, function (_ref3) {
    var getAllClasses = _ref3.getAllClasses,
        elementStyle = _ref3.elementStyle,
        elementAttributes = _ref3.elementAttributes;
    return /*#__PURE__*/React.createElement("div", Object.assign({
      style: _objectSpread(_objectSpread({}, elementStyle), {}, {
        display: "flex",
        justifyContent: alignItems
      })
    }, elementAttributes), /*#__PURE__*/React.createElement(Link, {
      to: link.href || null,
      target: link.newTab ? "_blank" : "_self",
      className: getAllClasses.apply(void 0, classes)
    }, content));
  });
};

export default Button;
