// @webiny/app-plugin-security-cognito-theme/site/components/Input.js
import React from "react";
import { I18NValue } from "@webiny/app-i18n/components";

var HelperMessage = function HelperMessage(props) {
  return /*#__PURE__*/React.createElement("div", {
    className: "webiny-cognito-field__helper-text" + (props.isValid === false ? " webiny-cognito-field__helper-text--error" : "")
  }, props.isValid === false ? props.errorMessage : props.helperMessage);
};

function Input(props) {
  var _onChange = props.onChange,
      value = props.value,
      validation = props.validation,
      validate = props.validate;

  var onBlur = function onBlur(e) {
    if (validate) {
      e.persist();
      validate();
    }
  };

  return /*#__PURE__*/React.createElement("div", {
    className: "webiny-cognito-field"
  }, /*#__PURE__*/React.createElement("label", {
    className: "webiny-cognito-field__label"
  }, /*#__PURE__*/React.createElement(I18NValue, {
    value: props.label
  })), /*#__PURE__*/React.createElement("input", {
    onBlur: onBlur,
    onChange: function onChange(e) {
      return _onChange(e.target.value);
    },
    value: value || "",
    placeholder: I18NValue(props.placeholderText),
    type: props.type,
    className: "webiny-cognito-field__input"
  }), /*#__PURE__*/React.createElement(HelperMessage, {
    isValid: validation.isValid,
    errorMessage: validation.message,
    helperMessage: /*#__PURE__*/React.createElement(I18NValue, {
      value: props.helpText
    })
  }));
}

export default Input;
