// @webiny/app-plugin-security-cognito-theme/site/components/Layout.js
import React from "react";
import classNames from "classnames";
export function ViewContainer(_ref) {
  var logo = _ref.logo,
      children = _ref.children;
  return /*#__PURE__*/React.createElement("div", {
    className: "webiny-cognito-view"
  }, /*#__PURE__*/React.createElement("div", {
    className: "webiny-cognito-view__logo-bar"
  }, /*#__PURE__*/React.createElement("img", {
    src: logo,
    alt: "Logo"
  })), /*#__PURE__*/React.createElement("div", {
    className: "webiny-cognito-view__content-box"
  }, children));
}
export function Content(_ref2) {
  var children = _ref2.children;
  return /*#__PURE__*/React.createElement("div", {
    className: "webiny-cognito-view__content"
  }, children);
}
export function Header(_ref3) {
  var _ref3$children = _ref3.children,
      children = _ref3$children === void 0 ? null : _ref3$children,
      title = _ref3.title;
  return /*#__PURE__*/React.createElement("div", {
    className: "webiny-cognito-view__header"
  }, /*#__PURE__*/React.createElement("h1", null, title), children);
}
export function Row(_ref4) {
  var children = _ref4.children,
      alignRight = _ref4.alignRight,
      alignCenter = _ref4.alignCenter;
  var classes = {
    "webiny-cognito-view__row": true,
    "webiny-cognito-view__row--align-right": alignRight,
    "webiny-cognito-view__row--align-center": alignCenter
  };
  return /*#__PURE__*/React.createElement("div", {
    className: classNames(classes)
  }, children);
}
export function Loading(_ref5) {
  var _ref5$label = _ref5.label,
      label = _ref5$label === void 0 ? null : _ref5$label;
  return /*#__PURE__*/React.createElement("div", {
    className: "webiny-cognito-loading"
  }, /*#__PURE__*/React.createElement("div", {
    className: "webiny-cognito-loading__container"
  }, /*#__PURE__*/React.createElement("div", {
    className: "webiny-cognito-loading__loader"
  }, "Loading..."), label && /*#__PURE__*/React.createElement("div", {
    className: "webiny-cognito-loading__label"
  }, label)));
}
export function Alert(_ref6) {
  var title = _ref6.title,
      _ref6$type = _ref6.type,
      type = _ref6$type === void 0 ? "danger" : _ref6$type,
      children = _ref6.children;
  var classes = {
    "webiny-cognito-alert": true,
    "webiny-cognito-alert--danger": type === "danger",
    "webiny-cognito-alert--info": type === "info"
  };
  return /*#__PURE__*/React.createElement("div", {
    className: classNames(classes)
  }, /*#__PURE__*/React.createElement("p", {
    className: "webiny-cognito-alert__title"
  }, title), /*#__PURE__*/React.createElement("p", {
    className: "webiny-cognito-alert__message"
  }, children));
}
