// @webiny/app-form-builder/admin/plugins/formDetails/formSubmissions/index.js
import _taggedTemplateLiteral from "@babel/runtime/helpers/taggedTemplateLiteral";

function _templateObject() {
  var data = _taggedTemplateLiteral(["Submissions"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

import * as React from "react";
import { renderPlugins } from "@webiny/app/plugins";
import { Tab } from "@webiny/ui/Tabs";
import styled from "@emotion/styled";
import FormSubmissionsOverview from "./FormSubmissionsOverview";
import FormSubmissionsList from "./FormSubmissionsList";
import { CircularProgress } from "@webiny/ui/Progress";
import { i18n } from "@webiny/app/i18n";
var t = i18n.namespace("FormsApp.FormDetails.PreviewContent");
var RenderBlock = /*#__PURE__*/styled("div", {
  target: "e1tsi7oi0",
  label: "RenderBlock"
})({
  position: "relative",
  zIndex: 0,
  backgroundColor: "var(--mdc-theme-background)",
  height: "100%",
  overflow: "scroll",
  padding: 25
});
export default [{
  name: "forms-form-details-revision-content-submissions",
  type: "forms-form-details-revision-content",
  render: function render(_ref) {
    var form = _ref.form,
        loading = _ref.loading;
    return /*#__PURE__*/React.createElement(Tab, {
      label: t(_templateObject()),
      disabled: loading
    }, /*#__PURE__*/React.createElement(RenderBlock, null, /*#__PURE__*/React.createElement("div", {
      style: {
        position: "relative"
      }
    }, loading && /*#__PURE__*/React.createElement(CircularProgress, null), form && renderPlugins("forms-form-details-submissions", {
      form: form
    }))));
  }
}, {
  name: "forms-form-details-submissions-overview",
  type: "forms-form-details-submissions",
  render: function render(_ref2) {
    var form = _ref2.form;
    return /*#__PURE__*/React.createElement(FormSubmissionsOverview, {
      form: form
    });
  }
}, {
  name: "forms-form-details-submissions-list",
  type: "forms-form-details-submissions",
  render: function render(_ref3) {
    var form = _ref3.form;
    return /*#__PURE__*/React.createElement(FormSubmissionsList, {
      form: form
    });
  }
}];
