// @webiny/ui/Accordion/AccordionItem.js
import _objectSpread from "@babel/runtime/helpers/objectSpread2";
import _classCallCheck from "@babel/runtime/helpers/classCallCheck";
import _createClass from "@babel/runtime/helpers/createClass";
import _assertThisInitialized from "@babel/runtime/helpers/assertThisInitialized";
import _inherits from "@babel/runtime/helpers/inherits";
import _createSuper from "@babel/runtime/helpers/createSuper";
import _defineProperty from "@babel/runtime/helpers/defineProperty";
import * as React from "react";
import { ListItem, ListItemGraphic, ListItemMeta } from "@webiny/ui/List";
import Transition from "react-transition-group/Transition";
import { Icon } from "@webiny/ui/Icon";
import styled from "@emotion/styled";
import { css } from "emotion";
import { Typography } from "@webiny/ui/Typography";
import { ReactComponent as UpArrow } from "@svgr/webpack!./icons/round-keyboard_arrow_up-24px.svg";
import { ReactComponent as DownArrow } from "@svgr/webpack!./icons/round-keyboard_arrow_down-24px.svg";
import classNames from "classnames";
var Content = /*#__PURE__*/styled("div", {
  target: "eypjd5u0",
  label: "Content"
})({
  width: "100%",
  borderBottom: "1px solid var(--mdc-theme-background)",
  boxSizing: "border-box"
});
var listItem = /*#__PURE__*/css({
  padding: "15px 20px",
  cursor: "pointer",
  borderBottom: "1px solid var(--mdc-theme-background)",
  "&:last-child": {
    borderBottom: "none"
  },
  ".mdc-list-item__graphic": {
    marginRight: 20
  }
}, "label:listItem;");
var ListItemTitle = /*#__PURE__*/styled("div", {
  target: "eypjd5u1",
  label: "ListItemTitle"
})({
  fontWeight: 600,
  marginBottom: 5
});
var ListItemDescription = /*#__PURE__*/styled("div", {
  target: "eypjd5u2",
  label: "ListItemDescription"
})({});
var TitleContent = /*#__PURE__*/styled("div", {
  target: "eypjd5u3",
  label: "TitleContent"
})({
  display: "flex",
  flexDirection: "column"
});
var openedState = /*#__PURE__*/css({
  backgroundColor: "var(--mdc-theme-on-background)"
}, "label:openedState;");
var duration = 150;
var defaultStyle = {
  transition: "all ".concat(duration, "ms ease-in-out"),
  opacity: 0,
  height: 0
};
var transitionStyles = {
  entering: {
    opacity: 0,
    height: 0,
    padding: "20px 20px 20px 65px"
  },
  entered: {
    opacity: 1,
    height: "auto",
    padding: "20px 20px 20px 65px"
  },
  exiting: {
    height: "auto",
    padding: "20px 20px 20px 65px"
  }
};

var AccordionItem = /*#__PURE__*/function (_React$Component) {
  _inherits(AccordionItem, _React$Component);

  var _super = _createSuper(AccordionItem);

  function AccordionItem() {
    var _this;

    _classCallCheck(this, AccordionItem);

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = _super.call.apply(_super, [this].concat(args));

    _defineProperty(_assertThisInitialized(_this), "state", {
      closed: true
    });

    return _this;
  }

  _createClass(AccordionItem, [{
    key: "render",
    value: function render() {
      var _this2 = this;

      return /*#__PURE__*/React.createElement("div", {
        className: classNames("webiny-ui-accordion-item", this.props.className)
      }, /*#__PURE__*/React.createElement(ListItem, {
        className: classNames(listItem, _defineProperty({}, openedState, !this.state.closed), "webiny-ui-accordion-item__list-item"),
        onClick: function onClick() {
          _this2.setState({
            closed: !_this2.state.closed
          });
        }
      }, this.props.icon && /*#__PURE__*/React.createElement(ListItemGraphic, null, /*#__PURE__*/React.createElement(Icon, {
        icon: this.props.icon
      })), /*#__PURE__*/React.createElement(TitleContent, {
        className: "webiny-ui-accordion-item__title"
      }, /*#__PURE__*/React.createElement(ListItemTitle, null, this.props.title), this.props.description && /*#__PURE__*/React.createElement(ListItemDescription, null, /*#__PURE__*/React.createElement(Typography, {
        use: "subtitle2"
      }, this.props.description))), /*#__PURE__*/React.createElement(ListItemMeta, null, /*#__PURE__*/React.createElement(Icon, {
        icon: this.state.closed ? /*#__PURE__*/React.createElement(DownArrow, null) : /*#__PURE__*/React.createElement(UpArrow, null)
      }))), /*#__PURE__*/React.createElement(Transition, {
        in: !this.state.closed,
        timeout: duration
      }, function (state) {
        return /*#__PURE__*/React.createElement(Content, {
          style: _objectSpread(_objectSpread({}, defaultStyle), transitionStyles[state]),
          className: "webiny-ui-accordion-item__content"
        }, _this2.props.children);
      }));
    }
  }]);

  return AccordionItem;
}(React.Component);

export { AccordionItem };
