// @webiny/app-headless-cms/admin/components/ContentModelEditor/EditorContent.js
import _slicedToArray from "@babel/runtime/helpers/slicedToArray";
import React, { useRef, useState } from "react";
import { css } from "emotion";
import styled from "@emotion/styled";
import { SplitView, LeftPanel, RightPanel } from "@webiny/app-admin/components/SplitView";
import { Typography } from "@webiny/ui/Typography";
import { Tabs, Tab } from "@webiny/ui/Tabs";
import { Icon } from "@webiny/ui/Icon";
import { EditTab } from "./Tabs/EditTab";
import { IndexesTab } from "./Tabs/IndexesTab";
import { PreviewTab } from "./Tabs/PreviewTab";
import { Fields } from "./Fields";
import { ReactComponent as FormIcon } from "@svgr/webpack!./icons/round-assignment-24px.svg";
var ContentContainer = /*#__PURE__*/styled("div", {
  target: "e4hq2j50",
  label: "ContentContainer"
})({
  paddingTop: 65
});
var LeftBarTitle = /*#__PURE__*/styled("div", {
  target: "e4hq2j51",
  label: "LeftBarTitle"
})({
  borderBottom: "1px solid var(--mdc-theme-on-background)",
  display: "flex",
  alignItems: "center",
  padding: 25,
  color: "var(--mdc-theme-on-surface)"
});
var titleIcon = /*#__PURE__*/css({
  height: 24,
  marginRight: 15,
  color: "var(--mdc-theme-primary)"
}, "label:titleIcon;");
var LeftBarFieldList = /*#__PURE__*/styled("div", {
  target: "e4hq2j52",
  label: "LeftBarFieldList"
})({
  padding: 40,
  overflow: "scroll",
  height: "calc(100vh - 250px)"
});
var formTabs = /*#__PURE__*/css({
  "&.webiny-ui-tabs": {
    ".webiny-ui-tabs__tab-bar": {
      backgroundColor: "var(--mdc-theme-surface)"
    }
  }
}, "label:formTabs;");
export default function EditorContent() {
  var tabsRef = useRef();

  var _useState = useState(0),
      _useState2 = _slicedToArray(_useState, 2),
      activeTabIndex = _useState2[0],
      setActiveTabIndex = _useState2[1];

  return /*#__PURE__*/React.createElement(ContentContainer, null, /*#__PURE__*/React.createElement(SplitView, null, /*#__PURE__*/React.createElement(LeftPanel, {
    span: 4
  }, /*#__PURE__*/React.createElement(LeftBarTitle, null, /*#__PURE__*/React.createElement(Icon, {
    className: titleIcon,
    icon: /*#__PURE__*/React.createElement(FormIcon, null)
  }), /*#__PURE__*/React.createElement(Typography, {
    use: "headline6"
  }, "Fields")), /*#__PURE__*/React.createElement(LeftBarFieldList, null, /*#__PURE__*/React.createElement(Fields, {
    onFieldDragStart: function onFieldDragStart() {
      if (tabsRef.current) {
        // @ts-ignore
        tabsRef.current.switchTab(0);
      }
    }
  }))), /*#__PURE__*/React.createElement(RightPanel, {
    span: 8
  }, /*#__PURE__*/React.createElement(Tabs, {
    className: formTabs,
    ref: tabsRef,
    onActivate: function onActivate(e) {
      return setActiveTabIndex(e);
    }
  }, /*#__PURE__*/React.createElement(Tab, {
    label: "Edit"
  }, /*#__PURE__*/React.createElement(EditTab, {
    activeTab: activeTabIndex === 0
  })), /*#__PURE__*/React.createElement(Tab, {
    label: "Preview"
  }, /*#__PURE__*/React.createElement(PreviewTab, {
    activeTab: activeTabIndex === 1
  })), /*#__PURE__*/React.createElement(Tab, {
    label: "Indexes"
  }, /*#__PURE__*/React.createElement(IndexesTab, {
    activeTab: activeTabIndex === 2
  }))))));
}
