// @webiny/app-form-builder/admin/plugins/routes.js
import _objectWithoutProperties from "@babel/runtime/helpers/objectWithoutProperties";
import React, { Suspense, lazy } from "react";
import Helmet from "react-helmet";
import { SecureRoute } from "@webiny/app-security/components";
import { CircularProgress } from "@webiny/ui/Progress";
import { Route } from "@webiny/react-router";
import { AdminLayout } from "@webiny/app-admin/components/AdminLayout";

var Loader = function Loader(_ref) {
  var children = _ref.children,
      props = _objectWithoutProperties(_ref, ["children"]);

  return /*#__PURE__*/React.createElement(Suspense, {
    fallback: /*#__PURE__*/React.createElement(CircularProgress, null)
  }, /*#__PURE__*/React.cloneElement(children, props));
};

var FormEditor = /*#__PURE__*/lazy(function () {
  return import("../views/Editor");
});
var Forms = /*#__PURE__*/lazy(function () {
  return import("../views/Forms/Forms");
});
var ROLE_FORMS_EDITOR = ["forms:form:crud"];
var plugins = [{
  name: "route-form-editors-editor",
  type: "route",
  route: /*#__PURE__*/React.createElement(Route, {
    exact: true,
    path: "/forms/:id",
    render: function render() {
      return /*#__PURE__*/React.createElement(SecureRoute, {
        scopes: ROLE_FORMS_EDITOR
      }, /*#__PURE__*/React.createElement(Helmet, null, /*#__PURE__*/React.createElement("title", null, "Edit form")), /*#__PURE__*/React.createElement(Loader, null, /*#__PURE__*/React.createElement(FormEditor, null)));
    }
  })
}, {
  name: "route-form-editors",
  type: "route",
  route: /*#__PURE__*/React.createElement(Route, {
    exact: true,
    path: "/forms",
    render: function render() {
      return /*#__PURE__*/React.createElement(SecureRoute, {
        scopes: ROLE_FORMS_EDITOR
      }, /*#__PURE__*/React.createElement(AdminLayout, null, /*#__PURE__*/React.createElement(Helmet, {
        title: "Form Builder"
      }), /*#__PURE__*/React.createElement(Loader, null, /*#__PURE__*/React.createElement(Forms, null))));
    }
  })
}];
export default plugins;
