// @webiny/app-form-builder/admin/plugins/scopesList.js
import _taggedTemplateLiteral from "@babel/runtime/helpers/taggedTemplateLiteral";

function _templateObject10() {
  var data = _taggedTemplateLiteral(["Allows creating form submission exports."]);

  _templateObject10 = function _templateObject10() {
    return data;
  };

  return data;
}

function _templateObject9() {
  var data = _taggedTemplateLiteral(["Export form submissions"]);

  _templateObject9 = function _templateObject9() {
    return data;
  };

  return data;
}

function _templateObject8() {
  var data = _taggedTemplateLiteral(["Allows unpublishing form revisions."]);

  _templateObject8 = function _templateObject8() {
    return data;
  };

  return data;
}

function _templateObject7() {
  var data = _taggedTemplateLiteral(["Unpublish form revisions"]);

  _templateObject7 = function _templateObject7() {
    return data;
  };

  return data;
}

function _templateObject6() {
  var data = _taggedTemplateLiteral(["Allows publishing form revision."]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = _taggedTemplateLiteral(["Publish form revisions"]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _taggedTemplateLiteral(["Allows updating Form Builder's settings."]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _taggedTemplateLiteral(["Form Builder Settings"]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteral(["Allows basic CRUD operations on all forms."]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteral(["Forms CRUD"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

import { i18n } from "@webiny/app/i18n";
var t = i18n.ns("app-form-builder/admin/scopesList");
export default [{
  name: "security-scopes-list-form-builder",
  type: "security-scopes-list",
  scopes: [{
    scope: "forms:form:crud",
    title: t(_templateObject()),
    description: t(_templateObject2())
  }, {
    scope: "forms:settings",
    title: t(_templateObject3()),
    description: t(_templateObject4())
  }, {
    scope: "forms:form:revision:publish",
    title: t(_templateObject5()),
    description: t(_templateObject6())
  }, {
    scope: "forms:form:revision:unpublish",
    title: t(_templateObject7()),
    description: t(_templateObject8())
  }, {
    scope: "forms:form:submissions:export",
    title: t(_templateObject9()),
    description: t(_templateObject10())
  }]
}];
