// @webiny/app-headless-cms/admin/plugins/contentDetails/footer/Footer.js
import React from "react";
import { css } from "emotion";
import { renderPlugins } from "@webiny/app/plugins";
import { Grid, Cell } from "@webiny/ui/Grid";
import classNames from "classnames";
var toolbarGrid = /*#__PURE__*/css({
  borderTop: "1px solid var(--mdc-theme-on-background)"
}, "label:toolbarGrid;");
var footerActions = /*#__PURE__*/css({
  display: "flex",
  alignItems: "center"
}, "label:footerActions;");
var footerActionsLeft = /*#__PURE__*/css({
  justifyContent: "flex-end"
}, "label:footerActionsLeft;");
var footerActionsRight = /*#__PURE__*/css({
  justifyContent: "flex-start"
}, "label:footerActionsRight;");

var Footer = function Footer(props) {
  return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(Grid, {
    className: toolbarGrid
  }, /*#__PURE__*/React.createElement(Cell, {
    span: 6,
    className: classNames(footerActions, footerActionsRight)
  }, renderPlugins("cms-content-details-footer-left", props)), /*#__PURE__*/React.createElement(Cell, {
    span: 6,
    className: classNames(footerActions, footerActionsLeft)
  }, renderPlugins("cms-content-details-footer-right", props))));
};

export default Footer;
