// @webiny/app-page-builder/editor/plugins/elements/imagesList/ImagesListDesignSettings.js
import * as React from "react";
import { Grid, Cell } from "@webiny/ui/Grid";
import { Select } from "@webiny/ui/Select";
import ImagesList from "./ImagesList";
import { getPlugins } from "@webiny/plugins";

var ImagesListDesignSettings = function ImagesListDesignSettings(_ref) {
  var Bind = _ref.Bind,
      data = _ref.data;
  var components = getPlugins("pb-editor-page-element-images-list-component");
  return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(Grid, null, /*#__PURE__*/React.createElement(Cell, {
    span: 12
  }, /*#__PURE__*/React.createElement(Bind, {
    name: "component",
    defaultValue: components[0] ? components[0].componentName : null
  }, /*#__PURE__*/React.createElement(Select, {
    label: "Design",
    description: "Select a component to render the list"
  }, components.map(function (cmp) {
    return /*#__PURE__*/React.createElement("option", {
      key: cmp.name,
      value: cmp.componentName
    }, cmp.title);
  }))))), /*#__PURE__*/React.createElement(Grid, null, /*#__PURE__*/React.createElement(Cell, {
    span: 12,
    style: {
      overflowY: "scroll"
    }
  }, /*#__PURE__*/React.createElement(ImagesList, {
    data: data
  }))));
};

export default ImagesListDesignSettings;
