// @webiny/app-page-builder/editor/plugins/elements/imagesList/index.js
import _objectSpread from "@babel/runtime/helpers/objectSpread2";
import React from "react";
import { Tab } from "@webiny/ui/Tabs";
import ImagesList from "./ImagesList";
import ImagesListImagesSettings from "./ImagesListImagesSettings";
import ImagesListDesignSettings from "./ImagesListDesignSettings";
import styled from "@emotion/styled";
import { ReactComponent as DesignIcon } from "@svgr/webpack!./icons/round-style-24px.svg";
import { ReactComponent as ImageGalleryIcon } from "@svgr/webpack!./icons/round-photo_library-24px.svg";
import { ReactComponent as ImagesIcon } from "@svgr/webpack!@webiny/app-page-builder/admin/assets/round-photo_library-24px.svg";
export default (function () {
  var PreviewBox = /*#__PURE__*/styled("div", {
    target: "e18t4fdc0",
    label: "PreviewBox"
  })({
    textAlign: "center",
    margin: "0 auto",
    width: 50,
    svg: {
      width: 50
    }
  });
  return [{
    name: "pb-editor-page-element-images-list",
    type: "pb-editor-page-element",
    elementType: "images-list",
    toolbar: {
      title: "Image Gallery",
      group: "pb-editor-element-group-basic",
      preview: function preview() {
        return /*#__PURE__*/React.createElement(PreviewBox, null, /*#__PURE__*/React.createElement(ImageGalleryIcon, null));
      }
    },
    settings: ["pb-editor-page-element-settings-delete"],
    target: ["row", "column"],
    onCreate: "open-settings",
    create: function create() {
      var options = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {};
      return _objectSpread({
        type: "images-list",
        data: {
          component: "mosaic",
          settings: {
            margin: {
              desktop: {
                all: 0
              },
              mobile: {
                all: 0
              }
            },
            padding: {
              desktop: {
                all: 0
              },
              mobile: {
                all: 0
              }
            }
          }
        }
      }, options);
    },
    render: function render(_ref) {
      var element = _ref.element;
      return /*#__PURE__*/React.createElement(ImagesList, {
        data: element.data
      });
    }
  }, {
    name: "pb-editor-page-element-advanced-settings-images-list-filter",
    type: "pb-editor-page-element-advanced-settings",
    elementType: "images-list",
    render: function render(props) {
      return /*#__PURE__*/React.createElement(Tab, {
        icon: /*#__PURE__*/React.createElement(ImagesIcon, null),
        label: "Images"
      }, /*#__PURE__*/React.createElement(ImagesListImagesSettings, Object.assign({}, props, {
        filter: true
      })));
    }
  }, {
    name: "pb-editor-page-element-advanced-settings-images-list-design",
    type: "pb-editor-page-element-advanced-settings",
    elementType: "images-list",
    render: function render(props) {
      return /*#__PURE__*/React.createElement(Tab, {
        icon: /*#__PURE__*/React.createElement(DesignIcon, null),
        label: "Design"
      }, /*#__PURE__*/React.createElement(ImagesListDesignSettings, props));
    }
  }];
});
