// @webiny/app-admin/plugins/UserMenu/plugins/DarkMode/DarkMode.js
import React from "react";
import { ListItem, ListItemGraphic, ListItemMeta } from "@webiny/ui/List";
import { Switch } from "@webiny/ui/Switch";
import { Icon } from "@webiny/ui/Icon";
import { ReactComponent as DarkModeIcon } from "@svgr/webpack!@webiny/app-admin/assets/icons/round-invert_colors-24px.svg";
import { useTheme } from "@webiny/app-admin/hooks/useTheme";
import { css } from "emotion";
import noop from "lodash/noop";
var darkModeSwitch = /*#__PURE__*/css({
  margin: "5px 0 0 5px"
}, "label:darkModeSwitch;");

var DarkMode = function DarkMode() {
  var _useTheme = useTheme(),
      toggleDarkMode = _useTheme.toggleDarkMode,
      theme = _useTheme.theme;

  return /*#__PURE__*/React.createElement(ListItem, {
    ripple: false,
    onClick: toggleDarkMode
  }, /*#__PURE__*/React.createElement(ListItemGraphic, null, /*#__PURE__*/React.createElement(Icon, {
    icon: /*#__PURE__*/React.createElement(DarkModeIcon, null)
  })), "Dark mode:", /*#__PURE__*/React.createElement(ListItemMeta, null, /*#__PURE__*/React.createElement(Switch, {
    className: darkModeSwitch,
    value: theme.dark,
    onChange: noop
  })));
};

export default DarkMode;
