// @webiny/app-headless-cms/admin/views/ContentModels/ContentModels.js
import _slicedToArray from "@babel/runtime/helpers/slicedToArray";
import React from "react";
import { SplitView, LeftPanel } from "@webiny/app-admin/components/SplitView";
import { FloatingActionButton } from "@webiny/app-admin/components/FloatingActionButton";
import ContentModelsDataList from "./ContentModelsDataList";
import NewContentModelDialog from "./NewContentModelDialog";
import { LIST_CONTENT_MODELS } from "../../viewsGraphql";
import { useDataList } from "@webiny/app/hooks/useDataList";
import { useApolloClient } from "@webiny/app-headless-cms/admin/hooks";

function ContentModels() {
  var _React$useState = React.useState(false),
      _React$useState2 = _slicedToArray(_React$useState, 2),
      newContentModelDialogOpened = _React$useState2[0],
      openNewContentModelDialog = _React$useState2[1];

  var apolloClient = useApolloClient();
  var dataList = useDataList({
    client: apolloClient,
    query: LIST_CONTENT_MODELS,
    variables: {
      sort: {
        savedOn: -1
      }
    }
  });
  return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(NewContentModelDialog, {
    open: newContentModelDialogOpened,
    onClose: function onClose() {
      return openNewContentModelDialog(false);
    },
    contentModelsDataList: dataList
  }), /*#__PURE__*/React.createElement(SplitView, null, /*#__PURE__*/React.createElement(LeftPanel, {
    span: 12
  }, /*#__PURE__*/React.createElement(ContentModelsDataList, {
    dataList: dataList
  }))), /*#__PURE__*/React.createElement(FloatingActionButton, {
    "data-testid": "new-record-button",
    onClick: function onClick() {
      return openNewContentModelDialog(true);
    }
  }));
}

export default ContentModels;
