// @webiny/app-headless-cms/admin/views/ContentModels/NewContentModelDialog.js
import _taggedTemplateLiteral from "@babel/runtime/helpers/taggedTemplateLiteral";
import _regeneratorRuntime from "@babel/runtime/regenerator";
import _asyncToGenerator from "@babel/runtime/helpers/asyncToGenerator";
import _slicedToArray from "@babel/runtime/helpers/slicedToArray";

function _templateObject7() {
  var data = _taggedTemplateLiteral(["Create"]);

  _templateObject7 = function _templateObject7() {
    return data;
  };

  return data;
}

function _templateObject6() {
  var data = _taggedTemplateLiteral(["Description"]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = _taggedTemplateLiteral(["Content model group"]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _taggedTemplateLiteral(["Choose a content model group"]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _taggedTemplateLiteral(["The name of the content model"]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteral(["Name"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteral(["New Content Model"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

import React from "react";
import { css } from "emotion";
import { useRouter } from "@webiny/react-router";
import { Form } from "@webiny/form";
import { Input } from "@webiny/ui/Input";
import { Select } from "@webiny/ui/Select";
import { CREATE_CONTENT_MODEL, LIST_MENU_CONTENT_GROUPS_MODELS } from "../../viewsGraphql";
import get from "lodash.get";
import { useSnackbar } from "@webiny/app-admin/hooks/useSnackbar";
import { CircularProgress } from "@webiny/ui/Progress";
import { validation } from "@webiny/validation";
import { useQuery, useMutation } from "@webiny/app-headless-cms/admin/hooks";
import { i18n } from "@webiny/app/i18n";
import { ButtonDefault } from "@webiny/ui/Button";
import { Dialog, DialogTitle, DialogContent, DialogActions } from "@webiny/ui/Dialog";
import { Grid, Cell } from "@webiny/ui/Grid";
var t = i18n.ns("app-headless-cms/admin/views/content-models/new-content-model-dialog");
var narrowDialog = /*#__PURE__*/css({
  ".mdc-dialog__surface": {
    width: 600,
    minWidth: 600
  }
}, "label:narrowDialog;");
var noPadding = /*#__PURE__*/css({
  padding: "5px !important"
}, "label:noPadding;");

var NewContentModelDialog = function NewContentModelDialog(_ref) {
  var open = _ref.open,
      onClose = _ref.onClose,
      contentModelsDataList = _ref.contentModelsDataList;

  var _React$useState = React.useState(false),
      _React$useState2 = _slicedToArray(_React$useState, 2),
      loading = _React$useState2[0],
      setLoading = _React$useState2[1];

  var _useSnackbar = useSnackbar(),
      showSnackbar = _useSnackbar.showSnackbar;

  var _useRouter = useRouter(),
      history = _useRouter.history;

  var _useMutation = useMutation(CREATE_CONTENT_MODEL),
      _useMutation2 = _slicedToArray(_useMutation, 1),
      createContentModel = _useMutation2[0];

  var _useQuery = useQuery(LIST_MENU_CONTENT_GROUPS_MODELS, {
    skip: !open
  }),
      data = _useQuery.data;

  var contentModelGroups = get(data, "listContentModelGroups.data", []).map(function (item) {
    return {
      value: item.id,
      label: item.name
    };
  });
  return /*#__PURE__*/React.createElement(Dialog, {
    open: open,
    onClose: onClose,
    className: narrowDialog,
    "data-testid": "cms-new-content-model-modal"
  }, open && /*#__PURE__*/React.createElement(Form, {
    data: {
      group: get(contentModelGroups, "0.value")
    },
    onSubmit: /*#__PURE__*/function () {
      var _ref2 = _asyncToGenerator( /*#__PURE__*/_regeneratorRuntime.mark(function _callee(data) {
        var response;
        return _regeneratorRuntime.wrap(function _callee$(_context) {
          while (1) {
            switch (_context.prev = _context.next) {
              case 0:
                setLoading(true);
                _context.t0 = get;
                _context.next = 4;
                return createContentModel({
                  variables: {
                    data: data
                  },
                  awaitRefetchQueries: true,
                  refetchQueries: ["HeadlessCmsListContentModels", "HeadlessCmsListMenuContentGroupsModels"]
                });

              case 4:
                _context.t1 = _context.sent;
                response = (0, _context.t0)(_context.t1, "data.createContentModel");

                if (!response.error) {
                  _context.next = 9;
                  break;
                }

                setLoading(false);
                return _context.abrupt("return", showSnackbar(response.error.message));

              case 9:
                _context.next = 11;
                return contentModelsDataList.refresh();

              case 11:
                history.push("/cms/content-models/" + response.data.id);

              case 12:
              case "end":
                return _context.stop();
            }
          }
        }, _callee);
      }));

      return function (_x) {
        return _ref2.apply(this, arguments);
      };
    }()
  }, function (_ref3) {
    var Bind = _ref3.Bind,
        submit = _ref3.submit;
    return /*#__PURE__*/React.createElement(React.Fragment, null, loading && /*#__PURE__*/React.createElement(CircularProgress, null), /*#__PURE__*/React.createElement(DialogTitle, null, t(_templateObject())), /*#__PURE__*/React.createElement(DialogContent, null, /*#__PURE__*/React.createElement(Grid, {
      className: noPadding
    }, /*#__PURE__*/React.createElement(Cell, {
      span: 12
    }, /*#__PURE__*/React.createElement(Bind, {
      name: "name",
      validators: validation.create("required,maxLength:100")
    }, /*#__PURE__*/React.createElement(Input, {
      label: t(_templateObject2()),
      description: t(_templateObject3())
    }))), /*#__PURE__*/React.createElement(Cell, {
      span: 12
    }, /*#__PURE__*/React.createElement(Bind, {
      name: "group",
      validators: validation.create("required")
    }, /*#__PURE__*/React.createElement(Select, {
      description: t(_templateObject4()),
      label: t(_templateObject5()),
      options: contentModelGroups
    }))), /*#__PURE__*/React.createElement(Cell, {
      span: 12
    }, /*#__PURE__*/React.createElement(Bind, {
      name: "description"
    }, function (props) {
      return /*#__PURE__*/React.createElement(Input, Object.assign({}, props, {
        rows: 4,
        maxLength: 200,
        characterCount: true,
        label: t(_templateObject6())
      }));
    })))), /*#__PURE__*/React.createElement(DialogActions, null, /*#__PURE__*/React.createElement(ButtonDefault, {
      onClick: submit
    }, "+ ", t(_templateObject7()))));
  }));
};

export default NewContentModelDialog;
