// @webiny/app-form-builder-theme/fields/Checkbox.js
import _toConsumableArray from "@babel/runtime/helpers/toConsumableArray";
import React from "react";
import { I18NValue } from "@webiny/app-i18n/components";
import HelperMessage from "../components/HelperMessage";

var change = function change(_ref) {
  var option = _ref.option,
      value = _ref.value,
      onChange = _ref.onChange;
  var newValues = Array.isArray(value) ? _toConsumableArray(value) : [];

  if (newValues.includes(option.value)) {
    newValues.splice(newValues.indexOf(option.value), 1);
  } else {
    newValues.push(option.value);
  }

  onChange(newValues);
};

var checked = function checked(_ref2) {
  var option = _ref2.option,
      value = _ref2.value;
  return Array.isArray(value) && value.includes(option.value);
};

var Checkbox = function Checkbox(props) {
  var _props$bind = props.bind,
      _onChange = _props$bind.onChange,
      value = _props$bind.value,
      validation = _props$bind.validation;
  var fieldId = props.field.fieldId;
  return /*#__PURE__*/React.createElement("div", {
    className: "webiny-fb-form-field webiny-fb-form-field--checkbox"
  }, /*#__PURE__*/React.createElement("label", {
    className: "webiny-fb-form-field__label webiny-pb-typography-body"
  }, /*#__PURE__*/React.createElement(I18NValue, {
    value: props.field.label
  })), /*#__PURE__*/React.createElement("div", {
    className: "webiny-fb-form-field__checkbox-group"
  }, props.field.options.map(function (option) {
    return /*#__PURE__*/React.createElement("div", {
      className: "webiny-fb-form-field__checkbox",
      key: option.value
    }, /*#__PURE__*/React.createElement("input", {
      name: fieldId,
      className: "webiny-fb-form-field__checkbox-input",
      type: "checkbox",
      id: "checkbox-" + fieldId + option.value,
      checked: checked({
        option: option,
        value: value
      }),
      onChange: function onChange() {
        return change({
          option: option,
          value: value,
          onChange: _onChange
        });
      }
    }), /*#__PURE__*/React.createElement("label", {
      htmlFor: "checkbox-" + fieldId + option.value,
      className: "webiny-fb-form-field__checkbox-label"
    }, /*#__PURE__*/React.createElement(I18NValue, {
      value: option.label
    })));
  })), /*#__PURE__*/React.createElement(HelperMessage, {
    isValid: validation.isValid,
    errorMessage: validation.message,
    helperMessage: /*#__PURE__*/React.createElement(I18NValue, {
      value: props.field.helpText
    })
  }));
};

export default Checkbox;
