// @webiny/app-form-builder-theme/fields/Input.js
import * as React from "react";
import { I18NValue } from "@webiny/app-i18n/components";
import HelperMessage from "../components/HelperMessage";

var Input = function Input(props) {
  var _props$bind = props.bind,
      _onChange = _props$bind.onChange,
      value = _props$bind.value,
      validation = _props$bind.validation,
      validate = _props$bind.validate; // @ts-ignore

  var onBlur = function onBlur(e) {
    if (validate) {
      // Since we are accessing event in an async operation, we need to persist it.
      // See https://reactjs.org/docs/events.html#event-pooling.
      e.persist();
      validate();
    }
  };

  return /*#__PURE__*/React.createElement("div", {
    className: "webiny-fb-form-field webiny-fb-form-field--input"
  }, /*#__PURE__*/React.createElement("label", {
    className: "webiny-pb-typography-body webiny-fb-form-field__label"
  }, /*#__PURE__*/React.createElement(I18NValue, {
    value: props.field.label
  })), /*#__PURE__*/React.createElement("input", {
    onBlur: onBlur,
    onChange: function onChange(e) {
      return _onChange(e.target.value);
    },
    value: value,
    placeholder: I18NValue(props.field.placeholderText),
    type: props.type,
    name: props.field.fieldId,
    id: props.field.fieldId,
    className: "webiny-fb-form-field__input"
  }), /*#__PURE__*/React.createElement(HelperMessage, {
    isValid: validation.isValid,
    errorMessage: validation.message,
    helperMessage: /*#__PURE__*/React.createElement(I18NValue, {
      value: props.field.helpText
    })
  }));
};

export default Input;
