// @webiny/app-form-builder-theme/fields/Radio.js
import * as React from "react";
import { I18NValue } from "@webiny/app-i18n/components";
import HelperMessage from "../components/HelperMessage";

var Radio = function Radio(props) {
  var _props$bind = props.bind,
      _onChange = _props$bind.onChange,
      value = _props$bind.value,
      validation = _props$bind.validation;
  var fieldId = props.field.fieldId;
  return /*#__PURE__*/React.createElement("div", {
    className: "webiny-fb-form-field webiny-fb-form-field--radio"
  }, /*#__PURE__*/React.createElement("label", {
    className: "webiny-fb-form-field__label webiny-pb-typography-body"
  }, /*#__PURE__*/React.createElement(I18NValue, {
    value: props.field.label
  })), /*#__PURE__*/React.createElement("div", {
    className: "webiny-fb-form-field__radio-group"
  }, props.field.options.map(function (option) {
    return /*#__PURE__*/React.createElement("div", {
      className: "webiny-fb-form-field__radio",
      key: option.value
    }, /*#__PURE__*/React.createElement("input", {
      checked: value === option.value,
      onChange: function onChange() {
        return _onChange(option.value);
      },
      name: fieldId,
      className: "webiny-fb-form-field__radio-input",
      type: "radio",
      id: "radio-" + fieldId + option.value,
      value: option.value
    }), /*#__PURE__*/React.createElement("label", {
      htmlFor: "radio-" + fieldId + option.value,
      className: "webiny-fb-form-field__radio-label"
    }, I18NValue({
      value: option.label
    })));
  })), /*#__PURE__*/React.createElement(HelperMessage, {
    isValid: validation.isValid,
    errorMessage: validation.message,
    helperMessage: /*#__PURE__*/React.createElement(I18NValue, {
      value: props.field.helpText
    })
  }));
};

export default Radio;
