// @webiny/app-form-builder-theme/fields/Select.js
import * as React from "react";
import { I18NValue } from "@webiny/app-i18n/components";
import HelperMessage from "../components/HelperMessage";

var Select = function Select(props) {
  var _props$bind = props.bind,
      _onChange = _props$bind.onChange,
      value = _props$bind.value,
      validation = _props$bind.validation;
  return /*#__PURE__*/React.createElement("div", {
    className: "webiny-fb-form-field webiny-fb-form-field--select"
  }, /*#__PURE__*/React.createElement("label", {
    className: "webiny-fb-form-field__label webiny-pb-typography-body"
  }, /*#__PURE__*/React.createElement(I18NValue, {
    value: props.field.label
  })), /*#__PURE__*/React.createElement("select", {
    value: value,
    onChange: function onChange(e) {
      return _onChange(e.target.value);
    },
    id: props.field.fieldId,
    name: props.field.fieldId,
    className: "webiny-fb-form-field__select"
  }, /*#__PURE__*/React.createElement("option", {
    disabled: true,
    value: ""
  }, I18NValue({
    value: props.field.placeholderText
  })), props.field.options.map(function (option) {
    return /*#__PURE__*/React.createElement("option", {
      key: option.value,
      value: option.value
    }, I18NValue({
      value: option.label
    }));
  })), /*#__PURE__*/React.createElement(HelperMessage, {
    isValid: validation.isValid,
    errorMessage: validation.message,
    helperMessage: /*#__PURE__*/React.createElement(I18NValue, {
      value: props.field.helpText
    })
  }));
};

export default Select;
