// @webiny/app-form-builder-theme/fields/Textarea.js
import * as React from "react";
import { I18NValue } from "@webiny/app-i18n/components";
import HelperMessage from "../components/HelperMessage";

var Textarea = function Textarea(props) {
  var _props$bind = props.bind,
      _onChange = _props$bind.onChange,
      value = _props$bind.value,
      validation = _props$bind.validation;
  return /*#__PURE__*/React.createElement("div", {
    className: "webiny-fb-form-field webiny-fb-form-field--textarea"
  }, /*#__PURE__*/React.createElement("label", {
    className: "webiny-fb-form-field__label webiny-pb-typography-body"
  }, /*#__PURE__*/React.createElement(I18NValue, {
    value: props.field.label
  })), /*#__PURE__*/React.createElement("textarea", {
    onChange: function onChange(e) {
      return _onChange(e.target.value);
    },
    value: value,
    placeholder: I18NValue({
      value: props.field.placeholderText
    }),
    rows: props.field.settings.rows ? props.field.settings.rows : 4,
    name: props.field.fieldId,
    id: props.field.fieldId,
    className: "webiny-fb-form-field__textarea"
  }), /*#__PURE__*/React.createElement(HelperMessage, {
    isValid: validation.isValid,
    errorMessage: validation.message,
    helperMessage: /*#__PURE__*/React.createElement(I18NValue, {
      value: props.field.helpText
    })
  }));
};

export default Textarea;
