// @webiny/app-page-builder/editor/plugins/toolbar/undoRedo/index.js
import platform from "platform";
import React from "react";
import { dispatch } from "@webiny/app-page-builder/editor/redux";
import { ActionCreators } from "redux-undo";
import { ReactComponent as UndoIcon } from "@svgr/webpack!@webiny/app-page-builder/editor/assets/icons/undo-icon.svg";
import { ReactComponent as RedoIcon } from "@svgr/webpack!@webiny/app-page-builder/editor/assets/icons/redo-icon.svg";
import Action from "../Action";
var metaKey = platform.os.family === "OS X" ? "CMD" : "CTRL";
export var undo = {
  name: "pb-editor-toolbar-undo",
  type: "pb-editor-toolbar-bottom",
  renderAction: function renderAction() {
    return /*#__PURE__*/React.createElement(Action, {
      tooltip: "Undo (".concat(metaKey, "+Z)"),
      onClick: function onClick() {
        return dispatch(ActionCreators.undo());
      },
      icon: /*#__PURE__*/React.createElement(UndoIcon, null)
    });
  }
};
export var redo = {
  name: "pb-editor-toolbar-redo",
  type: "pb-editor-toolbar-bottom",
  renderAction: function renderAction() {
    return /*#__PURE__*/React.createElement(Action, {
      tooltip: "Redo (".concat(metaKey, "+SHIFT+Z)"),
      onClick: function onClick() {
        return dispatch(ActionCreators.redo());
      },
      icon: /*#__PURE__*/React.createElement(RedoIcon, null)
    });
  }
};
