// @webiny/app-page-builder/editor/plugins/elements/media/youtube/index.js
import React from "react";
import styled from "@emotion/styled";
import { Tab } from "@webiny/ui/Tabs";
import { Input } from "@webiny/ui/Input";
import { Grid, Cell } from "@webiny/ui/Grid";
import { createEmbedSettingsPlugin, createEmbedPlugin } from "./../../utils/oembed";
import { ReactComponent as MediaIcon } from "@svgr/webpack!./../../../elementGroups/media/round-music_video-24px.svg";
import YoutubeEmbed from "./YoutubeEmbed";
import placeholder from "./placeholder.png";
import { validation } from "@webiny/validation";
import { ReactComponent as LogoIcon } from "@svgr/webpack!./youtube-brands.svg";
var PreviewBox = /*#__PURE__*/styled("div", {
  target: "ewlif860",
  label: "PreviewBox"
})({
  textAlign: "center",
  height: 50,
  svg: {
    height: 50,
    width: 50
  }
});
export default (function () {
  return [createEmbedPlugin({
    type: "youtube",
    toolbar: {
      title: "Youtube",
      group: "pb-editor-element-group-media",
      preview: function preview() {
        return /*#__PURE__*/React.createElement(PreviewBox, null, /*#__PURE__*/React.createElement(LogoIcon, null));
      }
    },
    onCreate: "open-settings",
    oembed: {
      renderEmbed: function renderEmbed(props) {
        return /*#__PURE__*/React.createElement(YoutubeEmbed, props);
      }
    },
    renderElementPreview: function renderElementPreview(_ref) {
      var width = _ref.width,
          height = _ref.height;
      return /*#__PURE__*/React.createElement("img", {
        style: {
          width: width,
          height: height
        },
        src: placeholder,
        alt: "Youtube"
      });
    }
  }), createEmbedSettingsPlugin({
    type: "youtube",
    render: function render(_ref2) {
      var Bind = _ref2.Bind;
      return /*#__PURE__*/React.createElement(Tab, {
        icon: /*#__PURE__*/React.createElement(MediaIcon, null),
        label: "YouTube"
      }, /*#__PURE__*/React.createElement(Grid, null, /*#__PURE__*/React.createElement(Cell, {
        span: 12
      }, /*#__PURE__*/React.createElement(Bind, {
        name: "source.url",
        validators: validation.create("required,url")
      }, /*#__PURE__*/React.createElement(Input, {
        label: "Video URL",
        description: "Enter a video URL"
      })))));
    }
  })];
});
