// @webiny/app-page-builder/editor/plugins/pageSettings/components/SeoSettings.js
import * as React from "react";
import { DelayedOnChange } from "@webiny/app-page-builder/editor/components/DelayedOnChange";
import { Grid, Cell } from "@webiny/ui/Grid";
import { Input } from "@webiny/ui/Input";
import MetaTags from "./MetaTags";

var SocialSettings = function SocialSettings(_ref) {
  var Bind = _ref.Bind;
  return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(Grid, null, /*#__PURE__*/React.createElement(Cell, {
    span: 12
  }, /*#__PURE__*/React.createElement(Bind, {
    name: "settings.seo.title"
  }, /*#__PURE__*/React.createElement(DelayedOnChange, null, /*#__PURE__*/React.createElement(Input, {
    label: "Title",
    description: "SEO title"
  }))))), /*#__PURE__*/React.createElement(Grid, null, /*#__PURE__*/React.createElement(Cell, {
    span: 12
  }, /*#__PURE__*/React.createElement(Bind, {
    name: "settings.seo.description"
  }, /*#__PURE__*/React.createElement(DelayedOnChange, null, /*#__PURE__*/React.createElement(Input, {
    rows: 4,
    label: "Description",
    description: "SEO description"
  }))))), /*#__PURE__*/React.createElement(Bind, {
    name: "settings.seo.meta",
    defaultValue: []
  }, function (props) {
    return /*#__PURE__*/React.createElement(MetaTags, Object.assign({
      prefix: "settings.seo.meta",
      Bind: Bind
    }, props));
  }));
};

export default SocialSettings;
