// @webiny/app-page-builder/editor/plugins/pageSettings/components/SocialSettings.js
import * as React from "react";
import { DelayedOnChange } from "@webiny/app-page-builder/editor/components/DelayedOnChange";
import { Grid, Cell } from "@webiny/ui/Grid";
import { Input } from "@webiny/ui/Input";
import OpenGraphTags from "./OpenGraphTags";
import PageImage from "./PageImage";
import appendOgImageDimensions from "./appendOgImageDimensions";

var SocialSettings = function SocialSettings(_ref) {
  var Bind = _ref.Bind,
      form = _ref.form;
  return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(Grid, null, /*#__PURE__*/React.createElement(Cell, {
    span: 12
  }, /*#__PURE__*/React.createElement(Bind, {
    name: "settings.social.title"
  }, /*#__PURE__*/React.createElement(DelayedOnChange, null, /*#__PURE__*/React.createElement(Input, {
    label: "Title (leave blank to use your page title)",
    description: "Social media title (og:title)."
  }))))), /*#__PURE__*/React.createElement(Grid, null, /*#__PURE__*/React.createElement(Cell, {
    span: 12
  }, /*#__PURE__*/React.createElement(Bind, {
    name: "settings.social.description"
  }, /*#__PURE__*/React.createElement(DelayedOnChange, null, /*#__PURE__*/React.createElement(Input, {
    label: "Description (leave blank to use your page snippet)",
    description: "Social media description (og:description)."
  }))))), /*#__PURE__*/React.createElement(Grid, null, /*#__PURE__*/React.createElement(Cell, {
    span: 12
  }, /*#__PURE__*/React.createElement(Bind, {
    name: "settings.social.image",
    afterChange: function afterChange(value) {
      return appendOgImageDimensions({
        value: value,
        form: form
      });
    }
  }, /*#__PURE__*/React.createElement(PageImage, {
    label: "Social Image",
    description: "Linked via \"og:image\" tag. Recommended resolution 1596x545." // @ts-ignore // TODO: @adrian what's this prop name ?
    ,
    imageEditor: {
      crop: {
        autoEnable: true,
        aspectRatio: 1596 / 545
      }
    }
  })))), /*#__PURE__*/React.createElement(Bind, {
    name: "settings.social.meta",
    defaultValue: []
  }, function (props) {
    return /*#__PURE__*/React.createElement(OpenGraphTags, Object.assign({
      prefix: "settings.social.meta",
      Bind: Bind
    }, props));
  }));
};

export default SocialSettings;
