// @webiny/app-i18n/admin/plugins/richTextEditor/link/index.js
import React from "react";
import { ReactComponent as LinkIcon } from "@svgr/webpack!@webiny/app-i18n/admin/assets/icons/link.svg";
import { LinkDialog } from "./LinkDialog";
import { LinkTooltip } from "./LinkTooltip";
var plugin = {
  name: "i18n-input-rich-text-editor-link",
  type: "i18n-input-rich-text-editor",
  plugin: {
    name: "link",
    menu: {
      render: function render(_ref) {
        var MenuButton = _ref.MenuButton,
            editor = _ref.editor,
            activatePlugin = _ref.activatePlugin;
        return /*#__PURE__*/React.createElement(MenuButton, {
          onMouseDown: function onMouseDown() {
            if (!editor.selection) {
              return;
            }

            activatePlugin("link");
          }
        }, /*#__PURE__*/React.createElement(LinkIcon, null));
      },
      renderDialog: function renderDialog(props) {
        return /*#__PURE__*/React.createElement(LinkDialog, props);
      }
    },
    editor: {
      renderElement: function renderElement(props, next) {
        var attributes = props.attributes,
            children = props.children,
            element = props.element;

        if (element.type === "link") {
          var href = element.href,
              noFollow = element.noFollow;
          return /*#__PURE__*/React.createElement("a", Object.assign({}, attributes, {
            href: href,
            rel: noFollow ? "nofollow" : null
          }), children);
        }

        return next();
      },
      renderEditor: function renderEditor(_ref2, next) {
        var activatePlugin = _ref2.activatePlugin;
        return /*#__PURE__*/React.createElement(React.Fragment, null, next(), /*#__PURE__*/React.createElement(LinkTooltip, {
          activatePlugin: activatePlugin
        }));
      }
    }
  }
};
export default plugin;
