// @webiny/app-headless-cms/admin/plugins/contentDetails/header/revisionSelector/RevisionSelector.js
import _get from "lodash/get";
import React from "react";
import { css } from "emotion";
import { useRouter } from "@webiny/react-router";
import { ButtonDefault } from "@webiny/ui/Button";
import { Icon } from "@webiny/ui/Icon";
import { ReactComponent as DownButton } from "@svgr/webpack!@webiny/app-headless-cms/admin/icons/round-arrow_drop_down-24px.svg";
import { Typography } from "@webiny/ui/Typography";
import { Menu, MenuItem } from "@webiny/ui/Menu";
var buttonStyle = /*#__PURE__*/css({
  "&.mdc-button": {
    color: "var(--mdc-theme-text-primary-on-background) !important"
  }
}, "label:buttonStyle;");
var menuStyles = /*#__PURE__*/css({
  width: 150,
  right: 0,
  ".mdc-list-item": {
    flexDirection: "column",
    justifyContent: "center",
    alignItems: "baseline",
    textAlign: "left"
  }
}, "label:menuStyles;");

var RevisionSelector = function RevisionSelector(_ref) {
  var content = _ref.content,
      getLoading = _ref.getLoading,
      revisionsList = _ref.revisionsList;

  var _useRouter = useRouter(),
      location = _useRouter.location,
      history = _useRouter.history;

  var query = new URLSearchParams(location.search);
  var currentRevision = {
    version: _get(content, "meta.version", 1),
    status: _get(content, "meta.status", "draft")
  };

  var allRevisions = _get(revisionsList, "data.content.data.meta.revisions", [{
    id: "new",
    meta: {
      version: 1,
      status: "draft"
    }
  }]);

  return /*#__PURE__*/React.createElement(Menu, {
    className: menuStyles,
    handle: /*#__PURE__*/React.createElement(ButtonDefault, {
      className: buttonStyle,
      disabled: getLoading()
    }, "v", currentRevision.version, " (", currentRevision.status, ")", " ", /*#__PURE__*/React.createElement(Icon, {
      icon: /*#__PURE__*/React.createElement(DownButton, null)
    }))
  }, allRevisions.map(function (revision) {
    return /*#__PURE__*/React.createElement(MenuItem, {
      key: revision.id,
      onClick: function onClick() {
        query.set("id", revision.id);
        history.push({
          search: query.toString()
        });
      }
    }, /*#__PURE__*/React.createElement(Typography, {
      use: "body2"
    }, "v", revision.meta.version), /*#__PURE__*/React.createElement(Typography, {
      use: "caption"
    }, "(", revision.meta.status, ")"));
  }));
};

export default RevisionSelector;
