// @webiny/app-page-builder/editor/plugins/slate/link/index.js
import React, { Fragment } from "react";
import { ReactComponent as LinkIcon } from "@svgr/webpack!@webiny/app-page-builder/editor/assets/icons/link.svg";
import LinkDialog from "./LinkDialog";
import LinkTooltip from "./LinkTooltip";
export default (function () {
  return {
    menu: [{
      name: "pb-editor-slate-menu-item-link",
      type: "pb-editor-slate-menu-item",
      render: function render(props) {
        var _this = this;

        var MenuButton = props.MenuButton;
        return /*#__PURE__*/React.createElement(MenuButton, {
          onMouseDown: function onMouseDown() {
            return props.activatePlugin(_this.name);
          }
        }, /*#__PURE__*/React.createElement(LinkIcon, null));
      },
      renderDialog: function renderDialog(props) {
        return /*#__PURE__*/React.createElement(LinkDialog, props);
      }
    }],
    editor: [{
      name: "pb-editor-slate-editor-link",
      type: "pb-editor-slate-editor",
      slate: {
        renderNode: function renderNode(props, next) {
          var attributes = props.attributes,
              children = props.children,
              node = props.node; // @ts-ignore

          if (node.type === "link") {
            // @ts-ignore
            var data = node.data;
            var href = data.get("href");
            var noFollow = data.get("noFollow");
            return /*#__PURE__*/React.createElement("a", Object.assign({}, attributes, {
              href: href,
              rel: noFollow ? "nofollow" : null
            }), children);
          }

          return next();
        },
        renderEditor: function renderEditor(_ref, next) {
          var editor = _ref.editor,
              onChange = _ref.onChange,
              activatePlugin = _ref.activatePlugin;
          // @ts-ignore
          var children = next();
          return /*#__PURE__*/React.createElement(Fragment, null, children, /*#__PURE__*/React.createElement(LinkTooltip, {
            editor: editor,
            onChange: onChange,
            activatePlugin: activatePlugin
          }));
        }
      }
    }]
  };
});
