// @webiny/app-page-builder/editor/plugins/defaultBar/components/PageOptionsMenu.js
import React from "react";
import { Menu } from "@webiny/ui/Menu";
import { IconButton } from "@webiny/ui/Button";
import { ReactComponent as MoreVerticalIcon } from "@svgr/webpack!@webiny/app-page-builder/editor/assets/icons/more_vert.svg";
import { getPlugins } from "@webiny/plugins";
import { css } from "emotion";
var menuStyles = /*#__PURE__*/css({
  width: 220,
  right: -105,
  left: "auto !important",
  ".disabled": {
    opacity: 0.5,
    pointerEvents: "none"
  }
}, "label:menuStyles;");
export default function PageOptionsMenu() {
  var plugins = getPlugins("pb-editor-default-bar-right-page-options");
  return /*#__PURE__*/React.createElement(Menu, {
    "data-testid": "pb-editor-page-options-menu",
    className: menuStyles,
    handle: /*#__PURE__*/React.createElement(IconButton, {
      icon: /*#__PURE__*/React.createElement(MoreVerticalIcon, null)
    })
  }, plugins.map(function (pl) {
    return /*#__PURE__*/React.cloneElement(pl.render(), {
      key: pl.name
    });
  }));
}
