// @webiny/app-headless-cms/admin/views/Environments/ConfirmationDialogWithInput.js
import _taggedTemplateLiteral from "@babel/runtime/helpers/taggedTemplateLiteral";
import _regeneratorRuntime from "@babel/runtime/regenerator";
import _asyncToGenerator from "@babel/runtime/helpers/asyncToGenerator";
import _classCallCheck from "@babel/runtime/helpers/classCallCheck";
import _createClass from "@babel/runtime/helpers/createClass";
import _assertThisInitialized from "@babel/runtime/helpers/assertThisInitialized";
import _inherits from "@babel/runtime/helpers/inherits";
import _createSuper from "@babel/runtime/helpers/createSuper";
import _defineProperty from "@babel/runtime/helpers/defineProperty";

function _templateObject() {
  var data = _taggedTemplateLiteral(["Please type {resourceName} to confirm."]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

import React from "react";
import { css } from "emotion";
import noop from "lodash/noop";
import { Dialog, DialogButton, DialogCancel, DialogActions, DialogTitle, DialogContent } from "@webiny/ui/Dialog";
import { Typography } from "@webiny/ui/Typography";
import { Input } from "@webiny/ui/Input";
import { CircularProgress } from "@webiny/ui/Progress";
import { Alert } from "@webiny/ui/Alert";
import { i18n } from "@webiny/app/i18n";
var t = i18n.ns("app-headless-cms/admin/environments/data-list-utils");
var messageStyle = /*#__PURE__*/css({
  "& .mdc-dialog__surface": {
    maxWidth: "540px !important"
  },
  "& .mdc-dialog__content": {
    span: {
      display: "inline-block",
      marginBottom: 16
    },
    b: {
      fontWeight: "bold"
    }
  },
  "& .mdc-dialog__actions": {
    alignItems: "center !important"
  },
  "& .mdc-dialog__button": {
    width: "100%",
    "&:nth-child(2)": {
      marginLeft: "0px !important"
    }
  }
}, "label:messageStyle;");
var confirmButtonStyles = /*#__PURE__*/css({
  backgroundColor: "#FED7D7 !important",
  color: "#9B2C2C !important",
  "&:disabled": {
    opacity: 0.4
  }
}, "label:confirmButtonStyles;");
var inputStyles = /*#__PURE__*/css({
  margin: "16px 0px"
}, "label:inputStyles;");

/**
 * Use ConfirmationDialog component to display a list of choices, once the handler is triggered.
 */
var ConfirmationDialogWithInput = /*#__PURE__*/function (_React$Component) {
  _inherits(ConfirmationDialogWithInput, _React$Component);

  var _super = _createSuper(ConfirmationDialogWithInput);

  function ConfirmationDialogWithInput() {
    var _this;

    _classCallCheck(this, ConfirmationDialogWithInput);

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = _super.call.apply(_super, [this].concat(args));

    _defineProperty(_assertThisInitialized(_this), "__isMounted", false);

    _defineProperty(_assertThisInitialized(_this), "callbacks", {
      onAccept: noop,
      onCancel: noop
    });

    _defineProperty(_assertThisInitialized(_this), "state", {
      show: false,
      loading: false,
      name: "",
      disableConfirm: true
    });

    _defineProperty(_assertThisInitialized(_this), "showConfirmation", function (onAccept, onCancel) {
      _this.callbacks = {
        onAccept: onAccept,
        onCancel: onCancel
      };

      _this.setState({
        show: true
      });
    });

    _defineProperty(_assertThisInitialized(_this), "hideConfirmation", function () {
      _this.setState({
        show: false
      });
    });

    _defineProperty(_assertThisInitialized(_this), "onAccept", /*#__PURE__*/_asyncToGenerator( /*#__PURE__*/_regeneratorRuntime.mark(function _callee() {
      var onAccept;
      return _regeneratorRuntime.wrap(function _callee$(_context) {
        while (1) {
          switch (_context.prev = _context.next) {
            case 0:
              onAccept = _this.callbacks.onAccept;

              if (!(typeof onAccept === "function")) {
                _context.next = 6;
                break;
              }

              _this.setState({
                loading: true
              });

              _context.next = 5;
              return onAccept();

            case 5:
              if (_this.__isMounted) {
                _this.setState({
                  loading: false,
                  show: false
                });
              }

            case 6:
            case "end":
              return _context.stop();
          }
        }
      }, _callee);
    })));

    _defineProperty(_assertThisInitialized(_this), "onCancel", /*#__PURE__*/_asyncToGenerator( /*#__PURE__*/_regeneratorRuntime.mark(function _callee2() {
      var onCancel;
      return _regeneratorRuntime.wrap(function _callee2$(_context2) {
        while (1) {
          switch (_context2.prev = _context2.next) {
            case 0:
              onCancel = _this.callbacks.onCancel;

              if (!(typeof onCancel === "function")) {
                _context2.next = 4;
                break;
              }

              _context2.next = 4;
              return onCancel();

            case 4:
            case "end":
              return _context2.stop();
          }
        }
      }, _callee2);
    })));

    _defineProperty(_assertThisInitialized(_this), "handleOnchange", function (value) {
      _this.setState({
        name: value
      });
    });

    return _this;
  }

  _createClass(ConfirmationDialogWithInput, [{
    key: "componentDidMount",
    value: function componentDidMount() {
      this.__isMounted = true;
    }
  }, {
    key: "componentWillUnmount",
    value: function componentWillUnmount() {
      this.__isMounted = false;
    }
  }, {
    key: "componentDidUpdate",
    value: function componentDidUpdate(prevProps, prevState) {
      if (prevState.name === this.state.name) {
        return;
      }

      if (this.state.disableConfirm && this.state.name === this.props.resourceName) {
        this.setState({
          disableConfirm: false
        });
      } else {
        this.setState({
          disableConfirm: true
        });
      }
    }
  }, {
    key: "render",
    value: function render() {
      return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(Dialog, {
        className: messageStyle,
        open: this.state.show,
        onClose: this.hideConfirmation,
        "data-testid": this.props["data-testid"]
      }, this.state.loading ? this.props.loading : null, /*#__PURE__*/React.createElement(DialogTitle, null, this.props.title), /*#__PURE__*/React.createElement(DialogContent, null, /*#__PURE__*/React.createElement(Alert, {
        title: "Wait! Let's review what you're about to do.",
        type: "warning"
      }), /*#__PURE__*/React.createElement(Typography, {
        use: "body1"
      }, this.props.message), " ", /*#__PURE__*/React.createElement("br", null), /*#__PURE__*/React.createElement(Typography, {
        use: "body1"
      }, t(_templateObject())({
        resourceName: /*#__PURE__*/React.createElement("b", null, this.props.resourceName)
      })), /*#__PURE__*/React.createElement(Input, {
        className: inputStyles,
        value: this.state.name,
        onChange: this.handleOnchange
      })), /*#__PURE__*/React.createElement(DialogActions, null, /*#__PURE__*/React.createElement(DialogCancel, {
        onClick: this.onCancel
      }, "Cancel"), /*#__PURE__*/React.createElement(DialogButton, {
        className: confirmButtonStyles,
        "data-testid": "confirmationdialog-confirm-action",
        onClick: this.onAccept,
        disabled: this.state.disableConfirm
      }, "I understand the consequences, delete this environment"))), this.props.children({
        showConfirmation: this.showConfirmation
      }));
    }
  }]);

  return ConfirmationDialogWithInput;
}(React.Component);

_defineProperty(ConfirmationDialogWithInput, "defaultProps", {
  title: "Confirmation",
  message: "Are you sure you want to continue?",
  loading: /*#__PURE__*/React.createElement(CircularProgress, null),
  resourceName: null
});

export { ConfirmationDialogWithInput };
