// @webiny/app-headless-cms/admin/views/Environments/EnvironmentsForm.js
import _taggedTemplateLiteral from "@babel/runtime/helpers/taggedTemplateLiteral";

function _templateObject8() {
  var data = _taggedTemplateLiteral(["Save environment"]);

  _templateObject8 = function _templateObject8() {
    return data;
  };

  return data;
}

function _templateObject7() {
  var data = _taggedTemplateLiteral(["A base environment from which the new one will be created."]);

  _templateObject7 = function _templateObject7() {
    return data;
  };

  return data;
}

function _templateObject6() {
  var data = _taggedTemplateLiteral(["A base environment from which this one was created."]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = _taggedTemplateLiteral(["Base environment"]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _taggedTemplateLiteral(["Created from"]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _taggedTemplateLiteral(["Description"]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteral(["Name"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteral(["New environment"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

import React from "react";
import { Form } from "@webiny/form";
import { Grid, Cell } from "@webiny/ui/Grid";
import { Input } from "@webiny/ui/Input";
import { ButtonPrimary } from "@webiny/ui/Button";
import { CircularProgress } from "@webiny/ui/Progress";
import { useCrud } from "@webiny/app-admin/hooks/useCrud";
import { i18n } from "@webiny/app/i18n";
import { validation } from "@webiny/validation";
import { SimpleForm, SimpleFormFooter, SimpleFormContent, SimpleFormHeader } from "@webiny/app-admin/components/SimpleForm";
import { AutoComplete } from "@webiny/ui/AutoComplete";
var t = i18n.ns("app-headless-cms/admin/environments/form");
import { useCms } from "@webiny/app-headless-cms/admin/hooks";

function EnvironmentsForm() {
  var _useCrud = useCrud(),
      crudForm = _useCrud.form;

  var _useCms = useCms(),
      createdFromOptions = _useCms.environments.environments;

  return /*#__PURE__*/React.createElement(Form, crudForm, function (_ref) {
    var form = _ref.form,
        data = _ref.data,
        Bind = _ref.Bind;
    return /*#__PURE__*/React.createElement(SimpleForm, {
      "data-testid": "pb-environments-form"
    }, crudForm.loading && /*#__PURE__*/React.createElement(CircularProgress, null), /*#__PURE__*/React.createElement(SimpleFormHeader, {
      title: data.name ? data.name : t(_templateObject())
    }), /*#__PURE__*/React.createElement(SimpleFormContent, null, /*#__PURE__*/React.createElement(Grid, null, /*#__PURE__*/React.createElement(Cell, {
      span: 12
    }, /*#__PURE__*/React.createElement(Bind, {
      name: "name",
      validators: validation.create("required,maxLength:100")
    }, /*#__PURE__*/React.createElement(Input, {
      label: t(_templateObject2())
    }))), /*#__PURE__*/React.createElement(Cell, {
      span: 12
    }, /*#__PURE__*/React.createElement(Bind, {
      name: "description"
    }, function (props) {
      return /*#__PURE__*/React.createElement(Input, Object.assign({}, props, {
        rows: 4,
        maxLength: 200,
        characterCount: true,
        label: t(_templateObject3())
      }));
    })), /*#__PURE__*/React.createElement(Cell, {
      span: 12
    }, /*#__PURE__*/React.createElement(Bind, {
      name: "createdFrom",
      validators: !data.id && validation.create("required")
    }, function (props) {
      return /*#__PURE__*/React.createElement(AutoComplete, Object.assign({}, props, {
        disabled: !!data.id,
        label: data.id ? t(_templateObject4()) : t(_templateObject5()),
        options: createdFromOptions,
        description: data.id ? t(_templateObject6()) : t(_templateObject7())
      }));
    })))), /*#__PURE__*/React.createElement(SimpleFormFooter, null, /*#__PURE__*/React.createElement(ButtonPrimary, {
      onClick: form.submit
    }, t(_templateObject8()))));
  });
}

export default EnvironmentsForm;
