// @webiny/app-headless-cms/admin/views/Environments/EnvironmentsDataList.js
import _regeneratorRuntime from "@babel/runtime/regenerator";
import _asyncToGenerator from "@babel/runtime/helpers/asyncToGenerator";
import _taggedTemplateLiteral from "@babel/runtime/helpers/taggedTemplateLiteral";

function _templateObject10() {
  var data = _taggedTemplateLiteral(["This action {verb} be undone. This will permanently delete the {name} environment and all of the created content."]);

  _templateObject10 = function _templateObject10() {
    return data;
  };

  return data;
}

function _templateObject9() {
  var data = _taggedTemplateLiteral(["Not linked with an alias."]);

  _templateObject9 = function _templateObject9() {
    return data;
  };

  return data;
}

function _templateObject8() {
  var data = _taggedTemplateLiteral(["This environment is linked with the \"{environmentAlias}\" alias."]);

  _templateObject8 = function _templateObject8() {
    return data;
  };

  return data;
}

function _templateObject7() {
  var data = _taggedTemplateLiteral(["Assigned to: {environmentAlias}"]);

  _templateObject7 = function _templateObject7() {
    return data;
  };

  return data;
}

function _templateObject6() {
  var data = _taggedTemplateLiteral(["(default)"]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = _taggedTemplateLiteral(["Name Z-A"]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _taggedTemplateLiteral(["Name A-Z"]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _taggedTemplateLiteral(["Oldest to newest"]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteral(["Newest to oldest"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteral(["Environments"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

import React from "react";
import { i18n } from "@webiny/app/i18n";
import { DeleteIcon } from "@webiny/ui/List/DataList/icons";
import { useCrud } from "@webiny/app-admin/hooks/useCrud";
import { Typography } from "@webiny/ui/Typography";
import { useCms } from "@webiny/app-headless-cms/admin/hooks";
import { DataList, List, ListItem, ListItemText, ListItemTextSecondary, ListItemMeta, ListActions } from "@webiny/ui/List";
import { Link } from "@webiny/react-router";
import { ConfirmationDialogWithInput } from "./ConfirmationDialogWithInput";
var t = i18n.ns("app-headless-cms/admin/environments/data-list");

var EnvironmentsDataList = function EnvironmentsDataList() {
  var _useCrud = useCrud(),
      actions = _useCrud.actions,
      list = _useCrud.list;

  var _useCms = useCms(),
      _useCms$environments = _useCms.environments,
      refreshEnvironments = _useCms$environments.refreshEnvironments,
      selectAvailableEnvironment = _useCms$environments.selectAvailableEnvironment,
      isSelectedEnvironment = _useCms$environments.isSelectedEnvironment;

  return /*#__PURE__*/React.createElement(DataList, Object.assign({}, list, {
    title: t(_templateObject()),
    sorters: [{
      label: t(_templateObject2()),
      sorters: {
        createdOn: -1
      }
    }, {
      label: t(_templateObject3()),
      sorters: {
        createdOn: 1
      }
    }, {
      label: t(_templateObject4()),
      sorters: {
        name: 1
      }
    }, {
      label: t(_templateObject5()),
      sorters: {
        name: -1
      }
    }]
  }), function (_ref) {
    var data = _ref.data,
        isSelected = _ref.isSelected,
        select = _ref.select;
    return /*#__PURE__*/React.createElement(List, {
      "data-testid": "default-data-list"
    }, data.map(function (item) {
      return /*#__PURE__*/React.createElement(ListItem, {
        key: item.id,
        selected: isSelected(item)
      }, /*#__PURE__*/React.createElement(ListItemText, {
        onClick: function onClick() {
          return select(item);
        }
      }, item.name, " ", item.default && /*#__PURE__*/React.createElement(Typography, {
        use: "overline"
      }, t(_templateObject6())), /*#__PURE__*/React.createElement(ListItemTextSecondary, null, item.environmentAlias ? t(_templateObject7())({
        environmentAlias: /*#__PURE__*/React.createElement(Link, {
          onClick: function onClick(e) {
            return e.stopPropagation();
          },
          to: "/settings/cms/environments/aliases?id=".concat(item.environmentAlias.id),
          title: t(_templateObject8())({
            environmentAlias: item.environmentAlias.name
          })
        }, item.environmentAlias.name)
      }) : t(_templateObject9()))), /*#__PURE__*/React.createElement(ListItemMeta, null, /*#__PURE__*/React.createElement(ListActions, null, /*#__PURE__*/React.createElement(ConfirmationDialogWithInput, {
        title: "Remove environment",
        message: t(_templateObject10())({
          name: /*#__PURE__*/React.createElement("b", null, item.name),
          verb: /*#__PURE__*/React.createElement("b", null, "cannot")
        }),
        resourceName: item.name
      }, function (_ref2) {
        var showConfirmation = _ref2.showConfirmation;
        return /*#__PURE__*/React.createElement(DeleteIcon, {
          onClick: function onClick() {
            showConfirmation( /*#__PURE__*/_asyncToGenerator( /*#__PURE__*/_regeneratorRuntime.mark(function _callee() {
              return _regeneratorRuntime.wrap(function _callee$(_context) {
                while (1) {
                  switch (_context.prev = _context.next) {
                    case 0:
                      _context.next = 2;
                      return actions.delete(item);

                    case 2:
                      if (isSelectedEnvironment(item)) {
                        selectAvailableEnvironment([item]);
                      }

                      refreshEnvironments();

                    case 4:
                    case "end":
                      return _context.stop();
                  }
                }
              }, _callee);
            })));
          }
        });
      }))));
    }));
  });
};

export default EnvironmentsDataList;
