// @webiny/app-security/admin/views/Users/UsersForm.js
import _taggedTemplateLiteral from "@babel/runtime/helpers/taggedTemplateLiteral";
import _slicedToArray from "@babel/runtime/helpers/slicedToArray";

function _templateObject7() {
  var data = _taggedTemplateLiteral(["Save user"]);

  _templateObject7 = function _templateObject7() {
    return data;
  };

  return data;
}

function _templateObject6() {
  var data = _taggedTemplateLiteral(["Roles"]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = _taggedTemplateLiteral(["Groups"]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _taggedTemplateLiteral(["E-mail"]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _taggedTemplateLiteral(["Last name"]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteral(["First Name"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteral(["New User"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

import React, { useState } from "react";
import { getPlugin } from "@webiny/plugins";
import { i18n } from "@webiny/app/i18n";
import { Form } from "@webiny/form";
import { Input } from "@webiny/ui/Input";
import { CircularProgress } from "@webiny/ui/Progress";
import { ButtonPrimary } from "@webiny/ui/Button";
import GroupsAutocomplete from "./../Components/GroupsAutocomplete";
import RolesAutocomplete from "./../Components/RolesAutocomplete";
import AvatarImage from "./../Components/AvatarImage";
import { useCrud } from "@webiny/app-admin/hooks/useCrud";
import { validation } from "@webiny/validation";
import { SimpleForm, SimpleFormHeader, SimpleFormFooter, SimpleFormContent } from "@webiny/app-admin/components/SimpleForm";
import AccountTokens from "@webiny/app-security/admin/views/AccountTokens";
var t = i18n.ns("app-security/admin/users/form");

var UsersForm = function UsersForm() {
  var _useState = useState(false),
      _useState2 = _slicedToArray(_useState, 2),
      formIsLoading = _useState2[0],
      setFormIsLoading = _useState2[1];

  var _useCrud = useCrud(),
      crudForm = _useCrud.form;

  var auth = getPlugin("security-view-user-form");

  if (!auth) {
    throw Error("You must register a \"security-view-user-form\" plugin to render User form!");
  }

  return /*#__PURE__*/React.createElement(Form, crudForm, function (_ref) {
    var data = _ref.data,
        form = _ref.form,
        Bind = _ref.Bind,
        setValue = _ref.setValue;
    return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement("div", {
      style: {
        marginBottom: "32px",
        marginTop: "24px"
      }
    }, /*#__PURE__*/React.createElement(Bind, {
      name: "avatar"
    }, /*#__PURE__*/React.createElement(AvatarImage, {
      round: true
    }))), /*#__PURE__*/React.createElement(SimpleForm, null, (formIsLoading || crudForm.loading) && /*#__PURE__*/React.createElement(CircularProgress, null), /*#__PURE__*/React.createElement(SimpleFormHeader, {
      title: data.fullName || t(_templateObject())
    }), /*#__PURE__*/React.createElement(SimpleFormContent, null, /*#__PURE__*/React.createElement(auth.view, {
      Bind: Bind,
      data: data,
      fields: {
        firstName: /*#__PURE__*/React.createElement(Bind, {
          name: "firstName",
          validators: validation.create("required")
        }, /*#__PURE__*/React.createElement(Input, {
          label: t(_templateObject2())
        })),
        lastName: /*#__PURE__*/React.createElement(Bind, {
          name: "lastName",
          validators: validation.create("required")
        }, /*#__PURE__*/React.createElement(Input, {
          label: t(_templateObject3())
        })),
        email: /*#__PURE__*/React.createElement(Bind, {
          name: "email",
          validators: validation.create("required,email")
        }, /*#__PURE__*/React.createElement(Input, {
          label: t(_templateObject4())
        })),
        groups: /*#__PURE__*/React.createElement(Bind, {
          name: "groups"
        }, /*#__PURE__*/React.createElement(GroupsAutocomplete, {
          label: t(_templateObject5())
        })),
        roles: /*#__PURE__*/React.createElement(Bind, {
          name: "roles"
        }, /*#__PURE__*/React.createElement(RolesAutocomplete, {
          label: t(_templateObject6())
        })),
        personalAccessTokens: /*#__PURE__*/React.createElement(Bind, {
          name: "personalAccessTokens"
        }, /*#__PURE__*/React.createElement(AccountTokens, {
          data: data,
          setValue: setValue,
          setFormIsLoading: setFormIsLoading
        }))
      }
    })), /*#__PURE__*/React.createElement(SimpleFormFooter, null, /*#__PURE__*/React.createElement(ButtonPrimary, {
      onClick: form.submit
    }, t(_templateObject7())))));
  });
};

export default UsersForm;
