// @webiny/app-security/admin/views/Users/UsersDataList.js
import _taggedTemplateLiteral from "@babel/runtime/helpers/taggedTemplateLiteral";

function _templateObject7() {
  var data = _taggedTemplateLiteral(["You can't delete your own user account."]);

  _templateObject7 = function _templateObject7() {
    return data;
  };

  return data;
}

function _templateObject6() {
  var data = _taggedTemplateLiteral(["User's avatar."]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = _taggedTemplateLiteral(["Name Z-A"]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _taggedTemplateLiteral(["Name A-Z"]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _taggedTemplateLiteral(["Oldest to newest"]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteral(["Newest to oldest"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteral(["Security Users"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

import React from "react";
import { i18n } from "@webiny/app/i18n";
import { useSecurity } from "@webiny/app-security/hooks/useSecurity";
import { ConfirmationDialog } from "@webiny/ui/ConfirmationDialog";
import { Tooltip } from "@webiny/ui/Tooltip";
import { Image } from "@webiny/app/components";
import { useCrud } from "@webiny/app-admin/hooks/useCrud";
import { DataList, ScrollList, ListItem, ListItemText, ListItemTextSecondary, ListItemMeta, ListActions, ListItemGraphic } from "@webiny/ui/List";
import { DeleteIcon } from "@webiny/ui/List/DataList/icons";
import { Avatar } from "@webiny/ui/Avatar";
var t = i18n.ns("app-security/admin/users/data-list");

var UsersDataList = function UsersDataList() {
  var security = useSecurity();

  var _useCrud = useCrud(),
      actions = _useCrud.actions,
      list = _useCrud.list;

  return /*#__PURE__*/React.createElement(DataList, Object.assign({}, list, {
    title: t(_templateObject()),
    sorters: [{
      label: t(_templateObject2()),
      sorters: {
        savedOn: -1
      }
    }, {
      label: t(_templateObject3()),
      sorters: {
        savedOn: 1
      }
    }, {
      label: t(_templateObject4()),
      sorters: {
        lastName: 1
      }
    }, {
      label: t(_templateObject5()),
      sorters: {
        lastName: -1
      }
    }]
  }), function (_ref) {
    var data = _ref.data,
        isSelected = _ref.isSelected,
        select = _ref.select;
    return /*#__PURE__*/React.createElement(ScrollList, {
      twoLine: true,
      avatarList: true
    }, data.map(function (item) {
      return /*#__PURE__*/React.createElement(ListItem, {
        key: item.id,
        selected: isSelected(item)
      }, /*#__PURE__*/React.createElement(ListItemGraphic, null, /*#__PURE__*/React.createElement(Avatar, {
        renderImage: function renderImage(props) {
          return /*#__PURE__*/React.createElement(Image, Object.assign({}, props, {
            transform: {
              width: 100
            }
          }));
        },
        src: item.avatar && item.avatar.src,
        fallbackText: item.fullName,
        alt: t(_templateObject6())
      })), /*#__PURE__*/React.createElement(ListItemText, {
        onClick: function onClick() {
          return select(item);
        }
      }, item.fullName, /*#__PURE__*/React.createElement(ListItemTextSecondary, null, item.email)), /*#__PURE__*/React.createElement(ListItemMeta, null, /*#__PURE__*/React.createElement(ListActions, null, security.user && security.user.id !== item.id ? /*#__PURE__*/React.createElement(ConfirmationDialog, null, function (_ref2) {
        var showConfirmation = _ref2.showConfirmation;
        return /*#__PURE__*/React.createElement(DeleteIcon, {
          onClick: function onClick() {
            return showConfirmation(function () {
              return actions.delete(item);
            });
          }
        });
      }) : /*#__PURE__*/React.createElement(Tooltip, {
        placement: "bottom",
        content: /*#__PURE__*/React.createElement("span", null, t(_templateObject7()))
      }, /*#__PURE__*/React.createElement(DeleteIcon, {
        disabled: true
      })))));
    }));
  });
};

export default UsersDataList;
