// @webiny/app-security/admin/views/Users/Users.js
import _objectSpread from "@babel/runtime/helpers/objectSpread2";
import _pick from "lodash/pick";
import React from "react";
import { SplitView, LeftPanel, RightPanel } from "@webiny/app-admin/components/SplitView";
import { FloatingActionButton } from "@webiny/app-admin/components/FloatingActionButton";
import UsersDataList from "./UsersDataList";
import UsersForm from "./UsersForm";
import { CREATE_USER, DELETE_USER, READ_USER, LIST_USERS, UPDATE_USER } from "./graphql";
import { CrudProvider } from "@webiny/app-admin/contexts/Crud";

var variables = function variables(data) {
  return {
    data: _objectSpread(_objectSpread({}, _pick(data, ["email", "password", "firstName", "lastName", "avatar", "enabled"])), {}, {
      roles: (data.roles || []).map(function (x) {
        return x.id;
      }),
      groups: (data.groups || []).map(function (x) {
        return x.id;
      })
    })
  };
};

var Users = function Users() {
  return /*#__PURE__*/React.createElement(CrudProvider, {
    delete: DELETE_USER,
    read: READ_USER,
    create: {
      mutation: CREATE_USER,
      variables: variables
    },
    update: {
      mutation: UPDATE_USER,
      variables: variables
    },
    list: {
      query: LIST_USERS,
      variables: {
        sort: {
          savedOn: -1
        }
      }
    }
  }, function (_ref) {
    var actions = _ref.actions;
    return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(SplitView, null, /*#__PURE__*/React.createElement(LeftPanel, null, /*#__PURE__*/React.createElement(UsersDataList, null)), /*#__PURE__*/React.createElement(RightPanel, {
      style: {
        marginLeft: "100px",
        marginRight: "100px",
        overflow: "hidden"
      }
    }, /*#__PURE__*/React.createElement(UsersForm, null))), /*#__PURE__*/React.createElement(FloatingActionButton, {
      onClick: actions.resetForm
    }));
  });
};

export default Users;
