// @webiny/app-page-builder/admin/plugins/pageDetails/pageRevisions/RevisionsList.js
import React from "react";
import { css } from "emotion";
import { List } from "@webiny/ui/List";
import { Elevation } from "@webiny/ui/Elevation";
import { CircularProgress } from "@webiny/ui/Progress";
import Revision from "./Revision";
var listWrapper = /*#__PURE__*/css({
  margin: 25,
  display: "flex",
  flexDirection: "column",
  overflow: "scroll",
  maxHeight: "calc(100vh - 160px)",
  ".mdc-list .mdc-list-item": {
    borderBottom: "1px solid var(--mdc-theme-on-background)"
  },
  ".mdc-list .mdc-list-item:last-child": {
    borderBottom: "none"
  }
}, "label:listWrapper;");

var RevisionsList = function RevisionsList(_ref) {
  var page = _ref.pageDetails.page,
      loading = _ref.loading;
  return /*#__PURE__*/React.createElement(Elevation, {
    className: listWrapper,
    z: 2
  }, /*#__PURE__*/React.createElement("div", {
    style: {
      position: "relative"
    }
  }, loading && /*#__PURE__*/React.createElement(CircularProgress, null), /*#__PURE__*/React.createElement(List, {
    nonInteractive: true,
    twoLine: true
  }, Array.isArray(page.revisions) && page.revisions.map(function (rev) {
    return /*#__PURE__*/React.createElement(Revision, {
      rev: rev,
      key: rev.id
    });
  }))));
};

export default RevisionsList;
