// @webiny/app-plugin-security-cognito-theme/admin/views/ForgotPassword.js
import * as React from "react";
import { Form } from "@webiny/form";
import { validation } from "@webiny/validation";
import { ButtonPrimary, ButtonDefault } from "@webiny/ui/Button";
import { Input } from "@webiny/ui/Input";
import { Grid, Cell } from "@webiny/ui/Grid";
import { Typography } from "@webiny/ui/Typography";
import { Alert } from "@webiny/ui/Alert";
import { Elevation } from "@webiny/ui/Elevation";
import StateContainer from "./StateContainer";
import { alignRight, InnerContent, Title } from "./StyledComponents";
import { CircularProgress } from "@webiny/ui/Progress";

var ForgotPassword = function ForgotPassword(_ref) {
  var requestCode = _ref.requestCode,
      setPassword = _ref.setPassword,
      codeSent = _ref.codeSent,
      error = _ref.error,
      loading = _ref.loading,
      authProps = _ref.authProps;

  var _ref2 = authProps.authData || {},
      _ref2$username = _ref2.username,
      username = _ref2$username === void 0 ? "" : _ref2$username;

  return /*#__PURE__*/React.createElement(StateContainer, null, /*#__PURE__*/React.createElement(Form, {
    data: {
      username: username
    },
    onSubmit: function onSubmit(_ref3) {
      var username = _ref3.username;
      return codeSent ? setPassword({
        username: username
      }) : requestCode({
        username: username
      });
    },
    submitOnEnter: true
  }, function (_ref4) {
    var Bind = _ref4.Bind,
        submit = _ref4.submit,
        data = _ref4.data;
    return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(Elevation, {
      z: 2
    }, /*#__PURE__*/React.createElement(InnerContent, null, loading && /*#__PURE__*/React.createElement(CircularProgress, null), /*#__PURE__*/React.createElement(Title, null, /*#__PURE__*/React.createElement("h1", null, /*#__PURE__*/React.createElement(Typography, {
      use: "headline4"
    }, "Password recovery")), /*#__PURE__*/React.createElement("p", null, /*#__PURE__*/React.createElement(Typography, {
      use: "subtitle2"
    }, "request a password reset code"))), error && /*#__PURE__*/React.createElement(Grid, null, /*#__PURE__*/React.createElement(Cell, {
      span: 12
    }, /*#__PURE__*/React.createElement(Alert, {
      title: "Something went wrong",
      type: "danger"
    }, error))), !codeSent ? /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(Grid, null, /*#__PURE__*/React.createElement(Cell, {
      span: 12
    }, /*#__PURE__*/React.createElement(Bind, {
      name: "username",
      validators: validation.create("required"),
      beforeChange: function beforeChange(val, cb) {
        return cb(val.toLowerCase());
      }
    }, /*#__PURE__*/React.createElement(Input, {
      label: "Email",
      outlined: true,
      description: "A reset code will be sent to your email."
    })))), /*#__PURE__*/React.createElement(Grid, null, /*#__PURE__*/React.createElement(Cell, {
      span: 12,
      className: alignRight
    }, /*#__PURE__*/React.createElement(ButtonPrimary, {
      onClick: submit
    }, "Send me the code")))) : /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(Grid, null, /*#__PURE__*/React.createElement(Cell, {
      span: 12
    }, /*#__PURE__*/React.createElement(Typography, {
      use: "body1"
    }, "We have sent you a code to reset your password!", /*#__PURE__*/React.createElement("br", null), /*#__PURE__*/React.createElement("br", null), "Click the \"Resend code\" button below to resend the code in case you haven't received it the first time."))), /*#__PURE__*/React.createElement(Grid, null, /*#__PURE__*/React.createElement(Cell, {
      span: 12,
      className: alignRight
    }, /*#__PURE__*/React.createElement(ButtonDefault, {
      onClick: function onClick() {
        return requestCode({
          username: data.username
        });
      }
    }, "Resend code"), /*#__PURE__*/React.createElement(ButtonPrimary, {
      onClick: submit
    }, "I got the code!")))))), /*#__PURE__*/React.createElement(Grid, null, /*#__PURE__*/React.createElement(Cell, {
      span: 12,
      style: {
        textAlign: "center"
      }
    }, "Want to sign in? ", /*#__PURE__*/React.createElement("a", {
      href: "#",
      onClick: function onClick() {
        return authProps.changeState("signIn");
      }
    }, "Sign in"))));
  }));
};

export default ForgotPassword;
