// @webiny/app-plugin-security-cognito-theme/admin/views/InstallForm.js
import _taggedTemplateLiteral from "@babel/runtime/helpers/taggedTemplateLiteral";

function _templateObject() {
  var data = _taggedTemplateLiteral(["Password"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

import React from "react";
import { Cell, Grid } from "@webiny/ui/Grid";
import { validation } from "@webiny/validation";
import { Input } from "@webiny/ui/Input";
import { Alert } from "@webiny/ui/Alert";
import { i18n } from "@webiny/app/i18n";
var t = i18n.ns("app-plugin-security-cognito-theme/install-form");

var InstallForm = function InstallForm(_ref) {
  var Bind = _ref.Bind,
      fields = _ref.fields,
      error = _ref.error;
  return /*#__PURE__*/React.createElement(Grid, null, error && /*#__PURE__*/React.createElement(Cell, {
    span: 12
  }, /*#__PURE__*/React.createElement(Alert, {
    title: "Something went wrong",
    type: "danger"
  }, error.message)), /*#__PURE__*/React.createElement(Cell, {
    span: 12
  }, "Let's create your admin user:"), /*#__PURE__*/React.createElement(Cell, {
    span: 12
  }, fields.firstName), /*#__PURE__*/React.createElement(Cell, {
    span: 12
  }, fields.lastName), /*#__PURE__*/React.createElement(Cell, {
    span: 12
  }, fields.email), /*#__PURE__*/React.createElement(Cell, {
    span: 12
  }, /*#__PURE__*/React.createElement(Bind, {
    name: "password",
    validators: validation.create("required")
  }, /*#__PURE__*/React.createElement(Input, {
    autoComplete: "off",
    type: "password",
    label: t(_templateObject())
  }))));
};

export default InstallForm;
