// @webiny/app-plugin-security-cognito-theme/admin/views/UserForm.js
import _taggedTemplateLiteral from "@babel/runtime/helpers/taggedTemplateLiteral";

function _templateObject2() {
  var data = _taggedTemplateLiteral(["Password"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteral(["Type a new password to reset it."]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

import React from "react";
import { Cell, Grid } from "@webiny/ui/Grid";
import { validation } from "@webiny/validation";
import { Input } from "@webiny/ui/Input";
import { i18n } from "@webiny/app/i18n";
import { Accordion, AccordionItem } from "@webiny/ui/Accordion";
import { ReactComponent as SettingsIcon } from "@svgr/webpack!../assets/icons/settings-24px.svg";
import { ReactComponent as SecurityIcon } from "@svgr/webpack!../assets/icons/security-24px.svg";
var t = i18n.ns("app-plugin-security-cognito-theme/users/form");

var UserForm = function UserForm(_ref) {
  var Bind = _ref.Bind,
      data = _ref.data,
      fields = _ref.fields;
  return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(Accordion, {
    elevation: 0
  }, /*#__PURE__*/React.createElement(AccordionItem, {
    description: "Set Account bio information",
    title: "Bio",
    icon: /*#__PURE__*/React.createElement(SettingsIcon, null)
  }, /*#__PURE__*/React.createElement(Grid, null, /*#__PURE__*/React.createElement(Cell, {
    span: 12
  }, fields.email), /*#__PURE__*/React.createElement(Cell, {
    span: 12
  }, /*#__PURE__*/React.createElement(Bind, {
    name: "password",
    validators: data.id ? null : validation.create("required")
  }, /*#__PURE__*/React.createElement(Input, {
    autoComplete: "off",
    description: data.id && t(_templateObject()),
    type: "password",
    label: t(_templateObject2())
  }))), /*#__PURE__*/React.createElement(Cell, {
    span: 12
  }, fields.firstName), /*#__PURE__*/React.createElement(Cell, {
    span: 12
  }, fields.lastName))), /*#__PURE__*/React.createElement(AccordionItem, {
    description: "Set Groups & Roles",
    title: "Groups & Roles",
    icon: /*#__PURE__*/React.createElement(SecurityIcon, null)
  }, /*#__PURE__*/React.createElement(Cell, {
    span: 12,
    style: {
      marginBottom: "8px"
    }
  }, fields.groups), /*#__PURE__*/React.createElement(Cell, {
    span: 12
  }, fields.roles)), /*#__PURE__*/React.createElement(AccordionItem, {
    description: "Set Personal Access Tokens",
    title: "Personal Access Tokens",
    icon: /*#__PURE__*/React.createElement(SecurityIcon, null)
  }, fields.personalAccessTokens)));
};

export default UserForm;
