// @webiny/app-page-builder/site/plugins/index.js
import _taggedTemplateLiteral from "@babel/runtime/helpers/taggedTemplateLiteral";
import _toConsumableArray from "@babel/runtime/helpers/toConsumableArray";
import _get from "lodash/get";

function _templateObject() {
  var data = _taggedTemplateLiteral(["\n                    query PbGetSettings {\n                        pageBuilder {\n                            getSettings {\n                                data {\n                                    name\n                                    favicon {\n                                        src\n                                    }\n                                    social {\n                                        image {\n                                            src\n                                        }\n                                    }\n                                }\n                            }\n                        }\n                    }\n                "]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

import * as React from "react";
import { Route } from "@webiny/react-router";
import renderPlugins from "@webiny/app-page-builder/render/presets/default";
import Helmet from "react-helmet";
import { Query } from "react-apollo";
import gql from "graphql-tag";
import linkPreloadPlugin from "./linkPreload";
import { PageRoute } from "./../components/Page";
export default (function () {
  return [{
    name: "pb-route",
    type: "route",
    route: /*#__PURE__*/React.createElement(Route, {
      component: PageRoute
    })
  }, linkPreloadPlugin(), {
    type: "addon-render",
    name: "addon-render-site-head",
    component: /*#__PURE__*/React.createElement(Query, {
      query: gql(_templateObject())
    }, function (_ref) {
      var response = _ref.data;

      var _ref2 = _get(response, "pageBuilder.getSettings.data") || {},
          favicon = _ref2.favicon,
          name = _ref2.name,
          social = _ref2.social; // Manually added "?width=128" to the favicon URL. In the future, fix this.
      // See "packages/webiny-app/src/plugins/imagePlugin.js:54"


      if (favicon && favicon.src) {
        var _social$image;

        return /*#__PURE__*/React.createElement(Helmet, null, /*#__PURE__*/React.createElement("link", {
          rel: "icon",
          type: "image/png",
          href: favicon.src + "?width=128",
          sizes: "16x16"
        }), name && /*#__PURE__*/React.createElement("meta", {
          name: "title",
          content: name
        }), name && /*#__PURE__*/React.createElement("meta", {
          property: "og:title",
          content: name
        }), (social === null || social === void 0 ? void 0 : (_social$image = social.image) === null || _social$image === void 0 ? void 0 : _social$image.src) && /*#__PURE__*/React.createElement("meta", {
          property: "og:image",
          content: social.image.src
        }));
      }

      return null;
    })
  }].concat(_toConsumableArray(renderPlugins()));
});
