// @webiny/app-headless-cms/admin/views/EnvironmentAliases/EnvironmentAliases.js
import React from "react";
import { SplitView, LeftPanel, RightPanel } from "@webiny/app-admin/components/SplitView";
import { FloatingActionButton } from "@webiny/app-admin/components/FloatingActionButton";
import EnvironmentAliasesDataList from "./EnvironmentAliasesDataList";
import EnvironmentAliasesForm from "./EnvironmentAliasesForm";
import { CrudProvider } from "@webiny/app-admin/contexts/Crud";
import { READ_ENVIRONMENT_ALIAS, LIST_ENVIRONMENT_ALIASES, CREATE_ENVIRONMENT_ALIAS, UPDATE_ENVIRONMENT_ALIAS, DELETE_ENVIRONMENT_ALIAS } from "./graphql";

function EnvironmentAliases() {
  return /*#__PURE__*/React.createElement(CrudProvider, {
    delete: {
      mutation: DELETE_ENVIRONMENT_ALIAS
    },
    read: READ_ENVIRONMENT_ALIAS,
    create: {
      mutation: CREATE_ENVIRONMENT_ALIAS
    },
    update: {
      mutation: UPDATE_ENVIRONMENT_ALIAS
    },
    list: {
      query: LIST_ENVIRONMENT_ALIASES,
      variables: {
        sort: {
          savedOn: -1
        }
      }
    }
  }, function (_ref) {
    var actions = _ref.actions;
    return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(SplitView, null, /*#__PURE__*/React.createElement(LeftPanel, {
      span: 4
    }, /*#__PURE__*/React.createElement(EnvironmentAliasesDataList, null)), /*#__PURE__*/React.createElement(RightPanel, {
      span: 8
    }, /*#__PURE__*/React.createElement(EnvironmentAliasesForm, null))), /*#__PURE__*/React.createElement(FloatingActionButton, {
      "data-testid": "new-record-button",
      onClick: actions.resetForm
    }));
  });
}

export default EnvironmentAliases;
