// @webiny/app-headless-cms/admin/views/EnvironmentAliases/graphql.js
import _taggedTemplateLiteral from "@babel/runtime/helpers/taggedTemplateLiteral";

function _templateObject6() {
  var data = _taggedTemplateLiteral(["\n    mutation deleteEnvironmentAlias($id: ID!) {\n        cms {\n            deleteEnvironmentAlias(id: $id) {\n                data\n                error {\n                    code\n                    message\n                }\n            }\n        }\n    }\n"]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = _taggedTemplateLiteral(["\n    mutation updateEnvironmentAlias($id: ID!, $data: CmsEnvironmentAliasInput!){\n        cms {\n            environmentAlias: updateEnvironmentAlias(id: $id, data: $data) {\n                data {\n                    ", "\n                }\n                error {\n                    code\n                    message\n                    data\n                }\n            }\n        }\n    }\n"]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _taggedTemplateLiteral(["\n    mutation createEnvironmentAlias($data: CmsEnvironmentAliasInput!){\n        cms {\n            environmentAlias: createEnvironmentAlias(data: $data) {\n                data {\n                    ", "\n                }\n                error {\n                    code\n                    message\n                    data\n                }\n            }\n        }\n    }\n"]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _taggedTemplateLiteral(["\n    query getEnvironmentAlias($id: ID!) {\n        cms {\n            environmentAlias: getEnvironmentAlias(id: $id){\n                data {\n                    ", "\n                }\n                error {\n                    code\n                    message\n                }\n            }\n        }\n    }\n"]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteral(["\n    query getEnvironmentAliasBySlug($slug: String) {\n        cms {\n            getEnvironmentAlias(where: { slug: $slug }) {\n                data {\n                    name\n                    id\n                }\n            }\n        }\n    }\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteral(["\n    query listEnvironmentAliases(\n        $where: JSON\n        $sort: JSON\n        $search: CmsSearchInput\n        $limit: Int\n        $after: String\n        $before: String\n    ) {\n        cms {\n            environmentAliases: listEnvironmentAliases(\n                where: $where\n                sort: $sort\n                search: $search\n                limit: $limit\n                after: $after\n                before: $before\n            ) {\n                data {\n                    id\n                    name\n                    slug\n                    createdOn\n                    environment {\n                        id\n                        name\n                    }\n                }\n                meta {\n                    cursors {\n                        next\n                        previous\n                    }\n                    hasNextPage\n                    hasPreviousPage\n                    totalCount\n                }\n            }\n        }\n    }\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

import gql from "graphql-tag";
var fields = "\n    id\n    name\n    slug\n    description\n    environment {\n        id\n        name\n    }\n";
export var LIST_ENVIRONMENT_ALIASES = gql(_templateObject());
export var GET_ENVIRONMENT_ALIAS_BY_SLUG = gql(_templateObject2());
export var READ_ENVIRONMENT_ALIAS = gql(_templateObject3(), fields);
export var CREATE_ENVIRONMENT_ALIAS = gql(_templateObject4(), fields);
export var UPDATE_ENVIRONMENT_ALIAS = gql(_templateObject5(), fields);
export var DELETE_ENVIRONMENT_ALIAS = gql(_templateObject6());
