// @webiny/app-admin/plugins/init.js
import _taggedTemplateLiteral from "@babel/runtime/helpers/taggedTemplateLiteral";

function _templateObject() {
  var data = _taggedTemplateLiteral(["Settings"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

import React from "react";
import { i18n } from "@webiny/app/i18n";
import { registerPlugins, getPlugins } from "@webiny/plugins";
import { ReactComponent as SettingsIcon } from "@svgr/webpack!@webiny/app-admin/assets/icons/round-settings-24px.svg";
var t = i18n.namespace("app-admin/menus");
var plugin = {
  type: "webiny-init",
  name: "webiny-init-settings",
  init: function init() {
    // Settings
    // Apps / integrations can register settings plugins and add menu items like the following.
    var settingsPlugins = getPlugins("admin-menu-settings");
    registerPlugins({
      name: "menu-settings",
      type: "admin-menu",
      order: 100,
      render: function render(_ref) {
        var Menu = _ref.Menu,
            Section = _ref.Section,
            Item = _ref.Item;
        return /*#__PURE__*/React.createElement(Menu, {
          name: "settings",
          label: t(_templateObject()),
          icon: /*#__PURE__*/React.createElement(SettingsIcon, null)
        }, settingsPlugins.map(function (plugin) {
          return /*#__PURE__*/React.createElement(React.Fragment, {
            key: plugin.name
          }, plugin.render({
            Section: Section,
            Item: Item
          }));
        }));
      }
    });
  }
};
export default plugin;
