// @webiny/app-page-builder/admin/plugins/settings/components/PageBuilderSettings.js
import _regeneratorRuntime from "@babel/runtime/regenerator";
import _asyncToGenerator from "@babel/runtime/helpers/asyncToGenerator";
import _get from "lodash/get";
import React from "react";
import { Form } from "@webiny/form";
import { Grid, Cell } from "@webiny/ui/Grid";
import { ButtonPrimary } from "@webiny/ui/Button";
import { Query, Mutation } from "react-apollo";
import { useSnackbar } from "@webiny/app-admin/hooks/useSnackbar";
import graphql from "./graphql";
import { PagesAutocomplete } from "@webiny/app-page-builder/admin/components/PagesAutocomplete";
import { CircularProgress } from "@webiny/ui/Progress";
import SingleImageUpload from "@webiny/app-admin/components/SingleImageUpload";
import { SimpleForm, SimpleFormFooter, SimpleFormContent, SimpleFormHeader } from "@webiny/app-admin/components/SimpleForm";

var PageBuilderSettings = function PageBuilderSettings() {
  var _useSnackbar = useSnackbar(),
      showSnackbar = _useSnackbar.showSnackbar;

  return /*#__PURE__*/React.createElement(Query, {
    query: graphql.query
  }, function (_ref) {
    var data = _ref.data,
        queryInProgress = _ref.loading;
    var settings = _get(data, "pageBuilder.getSettings.data") || {};
    return /*#__PURE__*/React.createElement(Mutation, {
      mutation: graphql.mutation
    }, function (update, _ref2) {
      var mutationInProgress = _ref2.loading;
      return /*#__PURE__*/React.createElement(Form, {
        data: settings,
        onSubmit: /*#__PURE__*/function () {
          var _ref3 = _asyncToGenerator( /*#__PURE__*/_regeneratorRuntime.mark(function _callee(data) {
            return _regeneratorRuntime.wrap(function _callee$(_context) {
              while (1) {
                switch (_context.prev = _context.next) {
                  case 0:
                    _context.next = 2;
                    return update({
                      variables: {
                        data: data
                      }
                    });

                  case 2:
                    showSnackbar("Settings updated successfully.");

                  case 3:
                  case "end":
                    return _context.stop();
                }
              }
            }, _callee);
          }));

          return function (_x) {
            return _ref3.apply(this, arguments);
          };
        }()
      }, function (_ref4) {
        var Bind = _ref4.Bind,
            form = _ref4.form;
        return /*#__PURE__*/React.createElement(SimpleForm, null, (queryInProgress || mutationInProgress) && /*#__PURE__*/React.createElement(CircularProgress, null), /*#__PURE__*/React.createElement(SimpleFormHeader, {
          title: "Website Settings"
        }), /*#__PURE__*/React.createElement(SimpleFormContent, null, /*#__PURE__*/React.createElement(Grid, null, /*#__PURE__*/React.createElement(Cell, {
          span: 6
        }, /*#__PURE__*/React.createElement(Grid, null, /*#__PURE__*/React.createElement(Cell, {
          span: 12
        }, /*#__PURE__*/React.createElement(Bind, {
          name: "pages.home"
        }, /*#__PURE__*/React.createElement(PagesAutocomplete, {
          label: "Homepage",
          description: "This is the homepage of your website."
        }))), /*#__PURE__*/React.createElement(Cell, {
          span: 12
        }, /*#__PURE__*/React.createElement(Bind, {
          name: "pages.error"
        }, /*#__PURE__*/React.createElement(PagesAutocomplete, {
          label: "Error page",
          description: "Shown when an error occurs during a page load."
        }))), /*#__PURE__*/React.createElement(Cell, {
          span: 12
        }, /*#__PURE__*/React.createElement(Bind, {
          name: "pages.notFound"
        }, /*#__PURE__*/React.createElement(PagesAutocomplete, {
          label: "Not found (404) page",
          description: "Shown when the requested page is not found."
        }))))), /*#__PURE__*/React.createElement(Cell, {
          span: 6
        }, /*#__PURE__*/React.createElement(Grid, null, /*#__PURE__*/React.createElement(Cell, {
          span: 12
        }, /*#__PURE__*/React.createElement(Bind, {
          name: "social.image"
        }, /*#__PURE__*/React.createElement(SingleImageUpload, {
          label: "Default Open Graph image",
          description: "The default OG image for all pages. Recommended resolution 1596x545." // TODO: @adrian
          // imageEditor={{
          //     crop: {
          //         autoEnable: true,
          //         aspectRatio: 1596 / 545
          //     }
          // }}

        }))))))), /*#__PURE__*/React.createElement(SimpleFormFooter, null, /*#__PURE__*/React.createElement(ButtonPrimary, {
          onClick: form.submit
        }, "Save")));
      });
    });
  });
};

export default PageBuilderSettings;
