// @webiny/app-plugin-security-cognito-theme/site/views/ForgotPassword.js
import React from "react";
import { Form } from "@webiny/form";
import { validation } from "@webiny/validation";
import Input from "../components/Input";
import ButtonPrimary from "../components/ButtonPrimary";
import ButtonDefault from "../components/ButtonDefault";
import { Content, Header, Row, Loading, Alert } from "../components/Layout";

var ForgotPassword = function ForgotPassword(_ref) {
  var requestCode = _ref.requestCode,
      setPassword = _ref.setPassword,
      codeSent = _ref.codeSent,
      error = _ref.error,
      loading = _ref.loading,
      authProps = _ref.authProps;

  var _ref2 = authProps.authData || {},
      _ref2$username = _ref2.username,
      username = _ref2$username === void 0 ? "" : _ref2$username;

  return /*#__PURE__*/React.createElement(Form, {
    data: {
      username: username
    },
    onSubmit: function onSubmit(data) {
      return codeSent ? setPassword(data) : requestCode(data);
    },
    submitOnEnter: true
  }, function (_ref3) {
    var Bind = _ref3.Bind,
        submit = _ref3.submit,
        data = _ref3.data;
    return /*#__PURE__*/React.createElement(Content, null, loading && /*#__PURE__*/React.createElement(Loading, null), /*#__PURE__*/React.createElement(Header, {
      title: "Password recovery"
    }, "To reset your password please provide your email address."), error && /*#__PURE__*/React.createElement(Row, null, /*#__PURE__*/React.createElement(Alert, {
      title: "Something went wrong"
    }, error)), !codeSent ? /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(Row, null, /*#__PURE__*/React.createElement(Bind, {
      name: "username",
      validators: validation.create("required"),
      beforeChange: function beforeChange(val, cb) {
        return cb(val.toLowerCase());
      }
    }, /*#__PURE__*/React.createElement(Input, {
      label: "Email",
      description: "A reset code will be sent to your email."
    }))), /*#__PURE__*/React.createElement(Row, {
      alignRight: true
    }, /*#__PURE__*/React.createElement(ButtonPrimary, {
      onClick: submit
    }, "Send me the code"))) : /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(Row, null, "We have sent you a code to reset your password!", /*#__PURE__*/React.createElement("br", null), /*#__PURE__*/React.createElement("br", null), "Click the \"Resend code\" button below to resend the code in case you haven't received it the first time."), /*#__PURE__*/React.createElement(Row, {
      alignRight: true
    }, /*#__PURE__*/React.createElement(ButtonDefault, {
      onClick: function onClick() {
        return requestCode(data);
      }
    }, "Resend code"), /*#__PURE__*/React.createElement(ButtonPrimary, {
      onClick: submit
    }, "I got the code!"))), /*#__PURE__*/React.createElement(Row, {
      alignCenter: true
    }, "Want to sign in? ", /*#__PURE__*/React.createElement("a", {
      href: "#",
      onClick: function onClick() {
        return authProps.changeState("signIn");
      }
    }, "Sign in")));
  });
};

export default ForgotPassword;
