// @webiny/app-plugin-security-cognito-theme/site/views/RequireNewPassword.js
import _lowerCase from "lodash/lowerCase";
import React from "react";
import { Form } from "@webiny/form";
import { validation } from "@webiny/validation";
import Input from "../components/Input";
import ButtonPrimary from "../components/ButtonPrimary";
import { Content, Header, Row } from "../components/Layout";

var sentenceCase = function sentenceCase(str) {
  var lower = _lowerCase(str);

  return lower[0].toUpperCase() + lower.substring(1);
};

var RequireNewPassword = function RequireNewPassword(_ref) {
  var confirm = _ref.confirm,
      requiredAttributes = _ref.requiredAttributes;
  return /*#__PURE__*/React.createElement(Form, {
    onSubmit: function onSubmit(data) {
      return confirm(data);
    },
    submitOnEnter: true
  }, function (_ref2) {
    var Bind = _ref2.Bind,
        submit = _ref2.submit;
    return /*#__PURE__*/React.createElement(Content, null, /*#__PURE__*/React.createElement(Header, {
      title: "Set new password"
    }), /*#__PURE__*/React.createElement(Row, null, /*#__PURE__*/React.createElement(Bind, {
      name: "password",
      validators: validation.create("required")
    }, /*#__PURE__*/React.createElement(Input, {
      type: "password",
      label: "New password"
    }))), requiredAttributes.length > 0 && /*#__PURE__*/React.createElement(Header, {
      title: "Please enter additional information"
    }), requiredAttributes.map(function (name) {
      return /*#__PURE__*/React.createElement(Row, {
        key: name
      }, /*#__PURE__*/React.createElement(Bind, {
        name: name,
        validators: validation.create("required")
      }, /*#__PURE__*/React.createElement(Input, {
        label: sentenceCase(name),
        box: true
      })));
    }), /*#__PURE__*/React.createElement(Row, {
      alignRight: true
    }, /*#__PURE__*/React.createElement(ButtonPrimary, {
      onClick: submit
    }, "Set password")));
  });
};

export default RequireNewPassword;
