// @webiny/app-plugin-security-cognito-theme/site/views/SetNewPassword.js
import React from "react";
import { Form } from "@webiny/form";
import { validation } from "@webiny/validation";
import Input from "../components/Input";
import ButtonPrimary from "../components/ButtonPrimary";
import { Content, Header, Row, Loading, Alert } from "../components/Layout";

var SetNewPassword = function SetNewPassword(_ref) {
  var setPassword = _ref.setPassword,
      error = _ref.error,
      loading = _ref.loading,
      authProps = _ref.authProps;
  return /*#__PURE__*/React.createElement(Form, {
    onSubmit: setPassword,
    submitOnEnter: true
  }, function (_ref2) {
    var Bind = _ref2.Bind,
        submit = _ref2.submit,
        data = _ref2.data;

    var retypePasswordValidator = function retypePasswordValidator(value) {
      if (value !== data.password) {
        throw Error("Passwords do not match!");
      }
    };

    return /*#__PURE__*/React.createElement(Content, null, loading && /*#__PURE__*/React.createElement(Loading, null), /*#__PURE__*/React.createElement(Header, {
      title: "Set new password"
    }), error && /*#__PURE__*/React.createElement(Alert, {
      title: "Something went wrong"
    }, error), /*#__PURE__*/React.createElement(Row, null, /*#__PURE__*/React.createElement(Bind, {
      name: "code",
      validators: validation.create("required")
    }, /*#__PURE__*/React.createElement(Input, {
      autoComplete: false,
      label: "Password reset code",
      description: "Enter the code we sent to your email."
    }))), /*#__PURE__*/React.createElement(Row, null, /*#__PURE__*/React.createElement(Bind, {
      name: "password",
      validators: validation.create("required")
    }, /*#__PURE__*/React.createElement(Input, {
      autoComplete: false,
      type: "password",
      label: "New password",
      description: "Enter your new password."
    }))), /*#__PURE__*/React.createElement(Row, null, /*#__PURE__*/React.createElement(Bind, {
      name: "retypePassword",
      validators: [validation.create("required"), retypePasswordValidator]
    }, /*#__PURE__*/React.createElement(Input, {
      type: "password",
      label: "Retype password",
      description: "Enter your new password once more."
    }))), /*#__PURE__*/React.createElement(Row, {
      alignRight: true
    }, /*#__PURE__*/React.createElement(ButtonPrimary, {
      onClick: submit
    }, "Set new password")), /*#__PURE__*/React.createElement(Row, {
      alignCenter: true
    }, "Want to sign in? ", /*#__PURE__*/React.createElement("a", {
      href: "#",
      onClick: function onClick() {
        return authProps.changeState("signIn");
      }
    }, "Sign in")));
  });
};

export default SetNewPassword;
