// @webiny/app-plugin-security-cognito-theme/site/views/SignIn.js
import * as React from "react";
import { Form } from "@webiny/form";
import { validation } from "@webiny/validation";
import Input from "../components/Input";
import ButtonPrimary from "../components/ButtonPrimary";
import { Content, Header, Row, Loading, Alert } from "../components/Layout";

var SignIn = function SignIn(_ref) {
  var signIn = _ref.signIn,
      error = _ref.error,
      authProps = _ref.authProps,
      loading = _ref.loading;
  return /*#__PURE__*/React.createElement(Form, {
    onSubmit: signIn,
    submitOnEnter: true
  }, function (_ref2) {
    var Bind = _ref2.Bind,
        submit = _ref2.submit;
    return /*#__PURE__*/React.createElement(Content, null, loading && /*#__PURE__*/React.createElement(Loading, null), /*#__PURE__*/React.createElement(Header, {
      title: "Sign In"
    }, /*#__PURE__*/React.createElement("p", null, "Welcome to Webiny.", /*#__PURE__*/React.createElement("br", null), "To view the page please sign in using your credentials.")), authProps.message && /*#__PURE__*/React.createElement(Row, null, /*#__PURE__*/React.createElement(Alert, {
      title: authProps.message.title,
      type: authProps.message.type
    }, authProps.message.text)), error && /*#__PURE__*/React.createElement(Row, null, /*#__PURE__*/React.createElement(Alert, {
      title: "Authentication error",
      type: "danger"
    }, error.message)), /*#__PURE__*/React.createElement(Row, null, /*#__PURE__*/React.createElement(Bind, {
      name: "username",
      validators: validation.create("required,email"),
      beforeChange: function beforeChange(val, cb) {
        return cb(val.toLowerCase());
      }
    }, /*#__PURE__*/React.createElement(Input, {
      label: "Your e-mail",
      outlined: true
    }))), /*#__PURE__*/React.createElement(Row, null, /*#__PURE__*/React.createElement(Bind, {
      name: "password",
      validators: validation.create("required")
    }, /*#__PURE__*/React.createElement(Input, {
      type: "password",
      label: "Your password",
      outlined: true
    }))), /*#__PURE__*/React.createElement(Row, {
      alignRight: true
    }, /*#__PURE__*/React.createElement(ButtonPrimary, {
      onClick: submit
    }, "Submit")), /*#__PURE__*/React.createElement(Row, {
      alignCenter: true
    }, /*#__PURE__*/React.createElement("a", {
      href: "#",
      onClick: function onClick() {
        return authProps.changeState("forgotPassword");
      }
    }, "Forgot password?")));
  });
};

export default SignIn;
