// @webiny/app-page-builder/editor/plugins/background/Background.js
import React from "react";
import { connect } from "@webiny/app-page-builder/editor/redux";
import { withActiveElement } from "@webiny/app-page-builder/editor/components";
import { highlightElement, deactivateElement } from "@webiny/app-page-builder/editor/actions";
import { css } from "emotion";
var backgroundStyle = /*#__PURE__*/css({
  position: "fixed",
  top: 0,
  left: 0,
  width: "100%",
  minHeight: "100%"
}, "label:backgroundStyle;");

var Background = function Background(_ref) {
  var element = _ref.element,
      deactivateElement = _ref.deactivateElement,
      highlightElement = _ref.highlightElement;
  return /*#__PURE__*/React.createElement("div", {
    className: backgroundStyle,
    onMouseOver: function onMouseOver() {
      return highlightElement({
        element: null
      });
    },
    onClick: function onClick() {
      return element && deactivateElement();
    }
  });
};

export default connect(null, {
  deactivateElement: deactivateElement,
  highlightElement: highlightElement
})(withActiveElement()(Background));
