// @webiny/app-admin/hooks/useConfirmationDialog.js
import _taggedTemplateLiteral from "@babel/runtime/helpers/taggedTemplateLiteral";
import _objectSpread from "@babel/runtime/helpers/objectSpread2";

function _templateObject2() {
  var data = _taggedTemplateLiteral(["Cancel"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteral(["Confirm"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

import { useUi } from "@webiny/app/hooks/useUi";
import { i18n } from "@webiny/app/i18n";
var t = i18n.ns("app-admin/hooks/use-confirmation-dialog");

var useConfirmationDialog = function useConfirmationDialog(_ref) {
  var title = _ref.title,
      message = _ref.message;
  var ui = useUi();
  return {
    showConfirmation: function showConfirmation(onAccept) {
      var onCancel = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : null;
      ui.setState(function (ui) {
        return _objectSpread(_objectSpread({}, ui), {}, {
          dialog: {
            message: message,
            options: {
              title: title,
              actions: {
                accept: {
                  label: t(_templateObject()),
                  onClick: onAccept
                },
                cancel: {
                  label: t(_templateObject2()),
                  onClick: onCancel
                }
              }
            }
          }
        });
      });
    }
  };
};

export { useConfirmationDialog };
